description: fix gcc bool-compare warnings
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/navigation_predictor/navigation_predictor.cc
+++ b/chrome/browser/navigation_predictor/navigation_predictor.cc
@@ -929,18 +929,6 @@ double NavigationPredictor::CalculateAnc
   DCHECK_LE(0, metrics.ratio_visible_area);
   DCHECK_GE(1, metrics.ratio_visible_area);
 
-  DCHECK_LE(0, metrics.is_in_iframe);
-  DCHECK_GE(1, metrics.is_in_iframe);
-
-  DCHECK_LE(0, metrics.is_same_host);
-  DCHECK_GE(1, metrics.is_same_host);
-
-  DCHECK_LE(0, metrics.contains_image);
-  DCHECK_GE(1, metrics.contains_image);
-
-  DCHECK_LE(0, metrics.is_url_incremented_by_one);
-  DCHECK_GE(1, metrics.is_url_incremented_by_one);
-
   DCHECK_LE(0, document_engagement_score);
   DCHECK_GE(1, document_engagement_score);
 
