description: fix gcc warnings about multi-line comments
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/browser/frame_host/render_frame_host_impl.h
+++ b/content/browser/frame_host/render_frame_host_impl.h
@@ -1084,11 +1084,13 @@ class CONTENT_EXPORT RenderFrameHostImpl
   // local roots which are under this frame and their first ancestor which is a
   // local root is either this frame or this frame's local root. For instance,
   // in a frame tree such as:
+  /*
   //                    A0
   //                 /  |   \
   //                B   A1   E
   //               /   /  \   \
   //              D  A2    C   F
+  */
   // RFHs at nodes B, E, D, C, and F are all local roots in the given frame tree
   // under the root at A0, but only B, C, and E are considered immediate local
   // roots of A0. Note that this will exclude any speculative or pending RFHs.
--- a/media/video/video_encode_accelerator.h
+++ b/media/video/video_encode_accelerator.h
@@ -161,7 +161,7 @@ class MEDIA_EXPORT VideoEncodeAccelerato
   // handed back to the client, and afterwards the |flush_callback| will be
   // called. The FlushCallback takes a boolean argument: |true| indicates the
   // flush is complete; |false| indicates the flush is cancelled due to errors
-  // or destruction. The client should not invoke Flush() or Encode() while the\
+  // or destruction. The client should not invoke Flush() or Encode() while the
   // previous Flush() is not finished yet.
   virtual void Flush(FlushCallback flush_callback);
 
--- a/third_party/WebKit/Source/core/css/cssom/CSSNumericValue.cpp
+++ b/third_party/WebKit/Source/core/css/cssom/CSSNumericValue.cpp
@@ -127,6 +127,7 @@ CSSNumericValue* CalcToNumericValue(cons
   DCHECK_EQ(root.GetType(), CSSCalcExpressionNode::kCssCalcBinaryOperation);
   CSSNumericValueVector values;
 
+  /*
   // For cases like calc(1 + 2 + 3), the calc expression tree looks like:
   //       +
   //      / \
@@ -139,6 +140,7 @@ CSSNumericValue* CalcToNumericValue(cons
   //      /|\
   //     1 2 3
   //
+  */
   // So when the left child has the same operator as its parent, we can combine
   // the two nodes. We keep moving down the left side of the tree as long as the
   // current node and the root can be combined, collecting the right child of
