description: fix gcc warnings about multi-line comments
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/browser/frame_host/render_frame_host_impl.h
+++ b/content/browser/frame_host/render_frame_host_impl.h
@@ -1107,11 +1107,13 @@ class CONTENT_EXPORT RenderFrameHostImpl
   // local roots which are under this frame and their first ancestor which is a
   // local root is either this frame or this frame's local root. For instance,
   // in a frame tree such as:
+  /*
   //                    A0
   //                 /  |   \
   //                B   A1   E
   //               /   /  \   \
   //              D  A2    C   F
+  */
   // RFHs at nodes B, E, D, C, and F are all local roots in the given frame tree
   // under the root at A0, but only B, C, and E are considered immediate local
   // roots of A0. Note that this will exclude any speculative or pending RFHs.
--- a/third_party/blink/renderer/core/css/cssom/css_numeric_value.cc
+++ b/third_party/blink/renderer/core/css/cssom/css_numeric_value.cc
@@ -127,6 +127,7 @@ CSSNumericValue* CalcToNumericValue(cons
   DCHECK_EQ(root.GetType(), CSSCalcExpressionNode::kCssCalcBinaryOperation);
   CSSNumericValueVector values;
 
+  /*
   // For cases like calc(1 + 2 + 3), the calc expression tree looks like:
   //       +
   //      / \
@@ -139,6 +140,7 @@ CSSNumericValue* CalcToNumericValue(cons
   //      /|\
   //     1 2 3
   //
+  */
   // So when the left child has the same operator as its parent, we can combine
   // the two nodes. We keep moving down the left side of the tree as long as the
   // current node and the root can be combined, collecting the right child of
