description: ignore deprecated bison directive warnings
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/build/scripts/rule_bison.py
+++ b/third_party/blink/renderer/build/scripts/rule_bison.py
@@ -82,7 +82,7 @@ def main():
     original_output_h = os.path.join(output_dir,
                                      output_basename + BISON_HEADER_EXT)
 
-    return_code = subprocess.call([bison_exe, '-d', input_file,
+    return_code = subprocess.call([bison_exe, '-d', input_file, '-Wno-deprecated',
                                    '-o', output_cc])
     assert return_code == 0
     # If the file doesn't exist, this raise an OSError.
