description: unfriend classes that friend themselves
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/apps/platform_apps/audio_focus_web_contents_observer.h
+++ b/chrome/browser/apps/platform_apps/audio_focus_web_contents_observer.h
@@ -22,7 +22,6 @@ class AudioFocusWebContentsObserver
 
  private:
   friend class content::WebContentsUserData<AudioFocusWebContentsObserver>;
-  friend class AudioFocusWebContentsObserver;
   friend class AudioFocusWebContentsObserverBrowserTest;
 
   explicit AudioFocusWebContentsObserver(content::WebContents*);
