description: avoid clang warnings about gcc specific compiler flag
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/platform/heap/impl/finalizer_traits.h
+++ b/third_party/blink/renderer/platform/heap/impl/finalizer_traits.h
@@ -43,10 +43,7 @@
 // an object's base class has a virtual destructor. In case there is no virtual
 // destructor present, the object is always finalized through its leaf type. In
 // other words: there is no finalization through a base pointer.
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdelete-non-virtual-dtor"
       static_cast<T*>(obj)->~T();
-#pragma GCC diagnostic pop
     }
   };
   using FinalizeImpl =
