description: fix gcc sign-compare warnings
author: Michael Gilbert <mgilbert@debian.org>

--- a/storage/browser/file_system/obfuscated_file_util_memory_delegate.cc
+++ b/storage/browser/file_system/obfuscated_file_util_memory_delegate.cc
@@ -582,10 +582,15 @@ base::File::Error ObfuscatedFileUtilMemo
   if (source_info.is_directory)
     return base::File::FILE_ERROR_NOT_A_FILE;
 
+  // Negative file size.
+  if (source_info.size < 0)
+    return base::File::FILE_ERROR_FAILED;
+
   // |size_t| limits the maximum size that the memory file can keep and |int|
   // limits the maximum size that base::ReadFile function reads.
-  if (source_info.size > std::numeric_limits<size_t>::max() ||
-      source_info.size > std::numeric_limits<int>::max()) {
+  uint64_t size = source_info.size;
+  if (size > std::numeric_limits<size_t>::max() ||
+      size > std::numeric_limits<int>::max()) {
     return base::File::FILE_ERROR_NO_SPACE;
   }
 
--- a/device/fido/credential_management.cc
+++ b/device/fido/credential_management.cc
@@ -130,7 +130,7 @@ base::Optional<CredentialsMetadataRespon
   if (it == response_map.end() || !it->second.is_unsigned()) {
     return base::nullopt;
   }
-  const int64_t existing_count = it->second.GetUnsigned();
+  const uint64_t existing_count = it->second.GetUnsigned();
   if (existing_count > std::numeric_limits<size_t>::max()) {
     return base::nullopt;
   }
@@ -142,7 +142,7 @@ base::Optional<CredentialsMetadataRespon
   if (it == response_map.end() || !it->second.is_unsigned()) {
     return base::nullopt;
   }
-  const int64_t remaining_count = it->second.GetUnsigned();
+  const uint64_t remaining_count = it->second.GetUnsigned();
   if (remaining_count > std::numeric_limits<size_t>::max()) {
     return base::nullopt;
   }
@@ -173,8 +173,11 @@ base::Optional<EnumerateRPsResponse> Enu
     return base::nullopt;
   }
   if (expect_rp_count) {
-    if (it == response_map.end() || !it->second.is_unsigned() ||
-        it->second.GetUnsigned() > std::numeric_limits<size_t>::max()) {
+    if (it == response_map.end() || !it->second.is_unsigned()) {
+      return base::nullopt;
+    }
+    const uint64_t count = it->second.GetUnsigned();
+    if (count > std::numeric_limits<size_t>::max()) {
       return base::nullopt;
     }
     rp_count = static_cast<size_t>(it->second.GetUnsigned());
@@ -290,8 +293,11 @@ EnumerateCredentialsResponse::Parse(
   } else {
     it = response_map.find(cbor::Value(
         static_cast<int>(CredentialManagementResponseKey::kTotalCredentials)));
-    if (it == response_map.end() || !it->second.is_unsigned() ||
-        it->second.GetUnsigned() > std::numeric_limits<size_t>::max()) {
+    if (it == response_map.end() || !it->second.is_unsigned()) {
+      return base::nullopt;
+    }
+    const uint64_t count = it->second.GetUnsigned();
+    if (count > std::numeric_limits<size_t>::max()) {
       return base::nullopt;
     }
     credential_count = static_cast<size_t>(it->second.GetUnsigned());
--- a/third_party/webrtc/modules/rtp_rtcp/source/rtp_packet_history.cc
+++ b/third_party/webrtc/modules/rtp_rtcp/source/rtp_packet_history.cc
@@ -134,7 +134,7 @@ void RtpPacketHistory::PutRtpPacket(std:
   // Store packet.
   const uint16_t rtp_seq_no = packet->SequenceNumber();
   int packet_index = GetPacketIndex(rtp_seq_no);
-  if (packet_index >= 0u &&
+  if (packet_index >= 0 &&
       static_cast<size_t>(packet_index) < packet_history_.size() &&
       packet_history_[packet_index].packet_ != nullptr) {
     RTC_LOG(LS_WARNING) << "Duplicate packet inserted: " << rtp_seq_no;
--- a/chrome/browser/chromeos/input_method/input_method_engine_base.cc
+++ b/chrome/browser/chromeos/input_method/input_method_engine_base.cc
@@ -39,7 +39,7 @@ const char kErrorWrongContext[] = "Context is not active.";
 const char kErrorInvalidValue[] = "Argument '%s' with value '%d' is not valid.";
 
 bool IsUint32Value(int i) {
-  return 0 <= i && i <= std::numeric_limits<uint32_t>::max();
+  return 0 <= i && static_cast<uint32_t>(i) <= std::numeric_limits<uint32_t>::max();
 }
 
 }  // namespace
