description: variable is unset because of a typo
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/ui/webui/signin/inline_login_handler_impl.h
+++ b/chrome/browser/ui/webui/signin/inline_login_handler_impl.h
@@ -13,6 +13,7 @@
 #include "base/macros.h"
 #include "base/memory/ref_counted.h"
 #include "base/memory/weak_ptr.h"
+#include "chrome/browser/signin/signin_util.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/ui/webui/signin/inline_login_handler.h"
 #include "chrome/browser/ui/webui/signin/signin_email_confirmation_dialog.h"
@@ -183,7 +184,9 @@ class InlineSigninHelper : public GaiaAu
   std::string gaia_id_;
   std::string password_;
   std::string auth_code_;
+#if BUILDFLAG(ENABLE_ONE_CLICK_SIGNIN)
   bool confirm_untrusted_signin_;
+#endif
   bool is_force_sign_in_with_usermanager_;
 
   DISALLOW_COPY_AND_ASSIGN(InlineSigninHelper);
--- a/chrome/browser/ui/webui/signin/inline_login_handler_impl.cc
+++ b/chrome/browser/ui/webui/signin/inline_login_handler_impl.cc
@@ -293,7 +293,6 @@ InlineSigninHelper::InlineSigninHelper(
       gaia_id_(gaia_id),
       password_(password),
       auth_code_(auth_code),
-      confirm_untrusted_signin_(confirm_untrusted_signin),
       is_force_sign_in_with_usermanager_(is_force_sign_in_with_usermanager) {
   DCHECK(profile_);
   DCHECK(!email_.empty());
