description: remove functions that are unused [-Wunused-function]
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/nasm/asm/nasm.c
+++ b/third_party/nasm/asm/nasm.c
@@ -358,26 +358,6 @@ static void emit_dependencies(StrList *l
         fclose(deps);
 }
 
-/* Convert a struct tm to a POSIX-style time constant */
-static int64_t make_posix_time(const struct tm *tm)
-{
-    int64_t t;
-    int64_t y = tm->tm_year;
-
-    /* See IEEE 1003.1:2004, section 4.14 */
-
-    t = (y-70)*365 + (y-69)/4 - (y-1)/100 + (y+299)/400;
-    t += tm->tm_yday;
-    t *= 24;
-    t += tm->tm_hour;
-    t *= 60;
-    t += tm->tm_min;
-    t *= 60;
-    t += tm->tm_sec;
-
-    return t;
-}
-
 static void timestamp(void)
 {
     struct compile_time * const oct = &official_compile_time;
