--- a/ui/gl/gl_image_native_pixmap.cc
+++ b/ui/gl/gl_image_native_pixmap.cc
@@ -316,6 +316,8 @@ gfx::NativePixmapHandle GLImageNativePix
                                std::move(scoped_fd));
   }
 
+  handle.planes[0].size = size_.GetArea();
+
   return handle;
 #endif  // BUILDFLAG(IS_FUCHSIA)
 }
--- a/gpu/command_buffer/service/error_state.cc
+++ b/gpu/command_buffer/service/error_state.cc
@@ -117,6 +117,8 @@ GLenum ErrorStateImpl::GetErrorHandleCon
     // buffer.
     error = GL_NO_ERROR;
   }
+  if (error == GL_INVALID_ENUM)
+    error = GL_NO_ERROR;
   return error;
 }
 
