--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1045,7 +1045,6 @@ if (!is_ios) {
       "//services/device/public/mojom:generic_sensor_js_data_deps",
       "//services/device/public/mojom:usb_js_data_deps",
       "//skia/public/mojom:mojom_js_data_deps",
-      "//testing/buildbot/filters:blink_web_tests_filter",
       "//third_party/blink/public:blink_devtools_frontend_resources_files",
       "//third_party/blink/public/mojom:mojom_platform_js_data_deps",
       "//third_party/blink/renderer/core/html:js_files_for_form_controls_web_tests",
--- a/services/BUILD.gn
+++ b/services/BUILD.gn
@@ -30,10 +30,6 @@ test("services_unittests") {
     "//services/test:run_all_unittests",
   ]
 
-  data_deps = [
-    "//testing/buildbot/filters:services_unittests_filters",
-  ]
-
   if (!is_ios) {
     deps += [
       "//services/audio:tests",
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -527,10 +527,6 @@ if (is_android) {
       "//testing/android/native_test:native_test_support",
     ]
 
-    data_deps = [
-      "//testing/buildbot/filters:android_browsertests_filters",
-    ]
-
     sources = [
       "../browser/engagement/important_sites_util_browsertest.cc",
       "../browser/payments/empty_parameters_browsertest.cc",
@@ -771,8 +767,6 @@ if (!is_android) {
       "//ppapi:copy_test_files2",
       "//ppapi:ppapi_tests",
       "//ppapi:power_saver_test_plugin",
-      "//testing/buildbot/filters:browser_tests_filters",
-      "//testing/buildbot/filters:linux_trusty_rel_browser_tests_filters",
       "//third_party/mesa_headers",
       "//third_party/widevine/cdm:widevine_test_license_server",
       "//tools/media_engagement_preload:generator",
@@ -1443,7 +1437,6 @@ if (!is_android) {
       data_deps += [
         "//chrome",
         "//ash/keyboard/ui:resources",
-        "//testing/buildbot/filters:chromeos_filters",
         "//ui/file_manager:unit_test_data",
       ]
 
@@ -2561,8 +2554,6 @@ if (!is_android) {
         "../browser/process_singleton_browsertest.cc",
       ]
 
-      data += [ "//testing/buildbot/filters/mac_window_server_killers.browser_tests.filter" ]
-
       if (safe_browsing_mode == 1) {
         sources -= [
           # single-process mode hangs on Mac sometimes because of multiple UI
@@ -3498,7 +3489,6 @@ test("unit_tests") {
   data_deps = [
     "//chrome/test/data/media/engagement/preload:generate_preload_list",
     "//chrome/test/data/media/engagement/preload:test_data",
-    "//testing/buildbot/filters:unit_tests_filters",
   ]
 
   data = [
@@ -6165,7 +6155,6 @@ if (!is_android && !is_fuchsia) {
 
     data_deps = [
       "//third_party/mesa_headers",
-      "//testing/buildbot/filters:e2e_sync_integration_tests_filters",
     ]
 
     if (is_mac) {
--- a/third_party/breakpad/BUILD.gn
+++ b/third_party/breakpad/BUILD.gn
@@ -721,7 +721,6 @@ if (is_linux || is_android) {
 
     data_deps = [
       ":linux_dumper_unittest_helper",
-      "//testing/buildbot/filters:breakpad_unittests_filters",
     ]
 
     include_dirs = [
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -1186,7 +1186,6 @@ test("content_browsertests") {
   data_deps = [
     ":web_ui_test_mojo_bindings_js_data_deps",
     "//content/shell:pak",
-    "//testing/buildbot/filters:content_browsertests_filters",
     "//third_party/mesa_headers",
   ]
 
@@ -2054,7 +2053,6 @@ test("content_unittests") {
   ]
 
   data_deps = [
-    "//testing/buildbot/filters:content_unittests_filters",
     "//third_party/mesa_headers",
   ]
 
