--- a/third_party/devtools-frontend/src/front_end/component_docs/computed_style_property/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/computed_style_property/BUILD.gn
@@ -14,7 +14,6 @@ ts_library("ts") {
   ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../elements:bundle",
   ]
 }
--- a/third_party/devtools-frontend/src/front_end/component_docs/data_grid/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/data_grid/BUILD.gn
@@ -18,7 +18,6 @@ ts_library("ts") {
   ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../ui/components:bundle",
   ]
--- a/third_party/devtools-frontend/src/front_end/component_docs/data_grid_controller/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/data_grid_controller/BUILD.gn
@@ -14,7 +14,6 @@ ts_library("ts") {
   ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../text_utils:bundle",
     "../../ui/components:bundle",
--- a/third_party/devtools-frontend/src/front_end/component_docs/flexbox_editor/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/flexbox_editor/BUILD.gn
@@ -10,7 +10,6 @@ ts_library("ts") {
   sources = [ "basic.ts" ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../elements:bundle",
   ]
--- a/third_party/devtools-frontend/src/front_end/component_docs/icon_button/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/icon_button/BUILD.gn
@@ -10,7 +10,6 @@ ts_library("ts") {
   sources = [ "basic.ts" ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../ui/components:bundle",
   ]
--- a/third_party/devtools-frontend/src/front_end/component_docs/markdown_image/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/markdown_image/BUILD.gn
@@ -6,7 +6,6 @@ ts_library("ts") {
   sources = [ "basic.ts" ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../issues:bundle",
   ]
--- a/third_party/devtools-frontend/src/front_end/component_docs/markdown_link/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/markdown_link/BUILD.gn
@@ -6,7 +6,6 @@ ts_library("ts") {
   sources = [ "basic.ts" ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../issues:bundle",
   ]
--- a/third_party/devtools-frontend/src/front_end/component_docs/markdown_view/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/markdown_view/BUILD.gn
@@ -6,7 +6,6 @@ ts_library("ts") {
   sources = [ "basic.ts" ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../issues:bundle",
     "../../third_party/marked:bundle",
--- a/third_party/devtools-frontend/src/front_end/component_docs/survey_link/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/survey_link/BUILD.gn
@@ -10,7 +10,6 @@ ts_library("ts") {
   sources = [ "basic.ts" ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../common:bundle",
     "../../component_helpers:bundle",
     "../../ui/components:bundle",
--- a/third_party/devtools-frontend/src/front_end/component_docs/tree_outline/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/tree_outline/BUILD.gn
@@ -15,7 +15,6 @@ ts_library("ts") {
   ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../third_party/lit-html:bundle",
     "../../ui/components:bundle",
--- a/third_party/devtools-frontend/src/front_end/component_docs/webvitals/BUILD.gn
+++ b/third_party/devtools-frontend/src/front_end/component_docs/webvitals/BUILD.gn
@@ -10,7 +10,6 @@ ts_library("ts") {
   sources = [ "webvitals.ts" ]
 
   deps = [
-    "../../../test/unittests/front_end/helpers",
     "../../component_helpers:bundle",
     "../../timeline:bundle",
     "../../ui:bundle",
