description: Disable usage of google-chrome in driver
author: Michel Le Bihan <michel@lebihan.pl>

--- a/chrome/test/chromedriver/chrome/chrome_finder.cc
+++ b/chrome/test/chromedriver/chrome/chrome_finder.cc
@@ -63,9 +63,6 @@ void GetApplicationDirs(std::vector<base
   locations->push_back(base::FilePath("/usr/bin"));
   locations->push_back(base::FilePath("/sbin"));
   locations->push_back(base::FilePath("/bin"));
-  // Lastly, try the default installation location.
-  locations->push_back(base::FilePath("/opt/google/chrome"));
-  locations->push_back(base::FilePath("/opt/chromium.org/chromium"));
 }
 #elif defined(OS_ANDROID)
 void GetApplicationDirs(std::vector<base::FilePath>* locations) {
@@ -139,8 +139,6 @@ bool FindChrome(base::FilePath* browser_exe) {
 #if defined(OS_WIN) || defined(OS_MAC)
     base::FilePath(chrome::kBrowserProcessExecutablePath),
 #elif defined(OS_LINUX) || defined(OS_CHROMEOS)
-    base::FilePath("google-chrome"),
-    base::FilePath(chrome::kBrowserProcessExecutablePath),
     base::FilePath("chromium"),
     base::FilePath("chromium-browser")
 #else
