description: disable third-party cookies by default
author: Chad Miller <chad.miller@canonical.com>

--- a/chrome/browser/prefs/browser_prefs.cc
+++ b/chrome/browser/prefs/browser_prefs.cc
@@ -549,7 +549,7 @@ void RegisterProfilePrefsForMigration(
 
   registry->RegisterDictionaryPref(kObservedSessionTime);
 
-  registry->RegisterBooleanPref(kBlockThirdPartyCookies, false);
+  registry->RegisterBooleanPref(kBlockThirdPartyCookies, true);
 
   registry->RegisterTimePref(kPluginsDeprecationInfobarLastShown, base::Time());
 
