description: avoid chromium's embedded C++ library when bootstrapping gn
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -111,6 +111,7 @@ def main(argv):
   cmd = [
       sys.executable,
       os.path.join(GN_ROOT, 'build', 'gen.py'),
+      '--no-static-libstdc++',
       '--no-last-commit-position',
       '--out-path=' + gn_build_dir,
   ]
