--- a/base/i18n/icu_util.cc
+++ b/base/i18n/icu_util.cc
@@ -350,134 +350,32 @@
 }
 
 const char kICUDataFile[] = "ICU.DataFile";
-const char kICUCreateInstance[] = "ICU.CreateInstance";
 
-enum class ICUCreateInstance {
-  kCharacterBreakIterator = 0,
-  kWordBreakIterator = 1,
-  kLineBreakIterator = 2,
-  kLineBreakIteratorTypeLoose = 3,
-  kLineBreakIteratorTypeNormal = 4,
-  kLineBreakIteratorTypeStrict = 5,
-  kSentenceBreakIterator = 6,
-  kTitleBreakIterator = 7,
-  kThaiBreakEngine = 8,
-  kLaoBreakEngine = 9,
-  kBurmeseBreakEngine = 10,
-  kKhmerBreakEngine = 11,
-  kChineseJapaneseBreakEngine = 12,
-
-  kMaxValue = kChineseJapaneseBreakEngine
-};
-
-// Callback functions to report the opening of ICU Data File, and creation of
-// key objects to UMA. This help us to understand what built-in ICU data files
-// are rarely used in the user's machines and the distribution of ICU usage.
-static void U_CALLCONV TraceICUEntry(const void*, int32_t fn_number) {
-  switch (fn_number) {
-    case UTRACE_UBRK_CREATE_CHARACTER:
-      base::UmaHistogramEnumeration(kICUCreateInstance,
-                                    ICUCreateInstance::kCharacterBreakIterator);
-      break;
-    case UTRACE_UBRK_CREATE_SENTENCE:
-      base::UmaHistogramEnumeration(kICUCreateInstance,
-                                    ICUCreateInstance::kSentenceBreakIterator);
-      break;
-    case UTRACE_UBRK_CREATE_TITLE:
-      base::UmaHistogramEnumeration(kICUCreateInstance,
-                                    ICUCreateInstance::kTitleBreakIterator);
-      break;
-    case UTRACE_UBRK_CREATE_WORD:
-      base::UmaHistogramEnumeration(kICUCreateInstance,
-                                    ICUCreateInstance::kWordBreakIterator);
-      break;
-    default:
-      return;
-  }
-}
-
-static void U_CALLCONV TraceICUData(const void* context,
-                                    int32_t fn_number,
-                                    int32_t level,
-                                    const char* fmt,
-                                    va_list args) {
-  switch (fn_number) {
-    case UTRACE_UDATA_DATA_FILE: {
-      std::string icu_data_file_name(va_arg(args, const char*));
-      va_end(args);
-      // Skip icu version specified prefix if exist.
-      // path is prefixed with icu version prefix such as "icudt65l-".
-      // Histogram only the part after the -.
-      if (icu_data_file_name.find("icudt") == 0) {
-        size_t dash = icu_data_file_name.find("-");
-        if (dash != std::string::npos) {
-          icu_data_file_name = icu_data_file_name.substr(dash + 1);
-        }
-      }
-      // UmaHistogramSparse should track less than 100 values.
-      // We currently have about total 55 built-in data files inside ICU
-      // so it fit the UmaHistogramSparse usage.
-      int hash = base::HashMetricName(icu_data_file_name);
-      base::UmaHistogramSparse(kICUDataFile, hash);
-      return;
-    }
-    case UTRACE_UBRK_CREATE_LINE: {
-      const char* lb_type = va_arg(args, const char*);
-      va_end(args);
-      ICUCreateInstance value;
-      switch (lb_type[0]) {
-        case '\0':
-          value = ICUCreateInstance::kLineBreakIterator;
-          break;
-        case 'l':
-          DCHECK(strcmp(lb_type, "loose") == 0);
-          value = ICUCreateInstance::kLineBreakIteratorTypeLoose;
-          break;
-        case 'n':
-          DCHECK(strcmp(lb_type, "normal") == 0);
-          value = ICUCreateInstance::kLineBreakIteratorTypeNormal;
-          break;
-        case 's':
-          DCHECK(strcmp(lb_type, "strict") == 0);
-          value = ICUCreateInstance::kLineBreakIteratorTypeStrict;
-          break;
-        default:
-          return;
-      }
-      base::UmaHistogramEnumeration(kICUCreateInstance, value);
-      return;
-    }
-    case UTRACE_UBRK_CREATE_BREAK_ENGINE: {
-      const char* script = va_arg(args, const char*);
-      va_end(args);
-      ICUCreateInstance value;
-      switch (script[0]) {
-        case 'H':
-          DCHECK(strcmp(script, "Hani") == 0);
-          value = ICUCreateInstance::kChineseJapaneseBreakEngine;
-          break;
-        case 'K':
-          DCHECK(strcmp(script, "Khmr") == 0);
-          value = ICUCreateInstance::kKhmerBreakEngine;
-          break;
-        case 'L':
-          DCHECK(strcmp(script, "Laoo") == 0);
-          value = ICUCreateInstance::kLaoBreakEngine;
-          break;
-        case 'M':
-          DCHECK(strcmp(script, "Mymr") == 0);
-          value = ICUCreateInstance::kBurmeseBreakEngine;
-          break;
-        case 'T':
-          DCHECK(strcmp(script, "Thai") == 0);
-          value = ICUCreateInstance::kThaiBreakEngine;
-          break;
-        default:
-          return;
+// Callback function to report the opening of ICU Data File to UMA.
+// This help us to understand what built-in ICU data files are rarely used
+// in the user's machines.
+static void U_CALLCONV TraceICUDataFile(const void* context,
+                                        int32_t fn_number,
+                                        int32_t level,
+                                        const char* fmt,
+                                        va_list args) {
+  if (fn_number == UTRACE_UDATA_DATA_FILE) {
+    std::string icu_data_file_name(va_arg(args, const char*));
+    va_end(args);
+    // Skip icu version specified prefix if exist.
+    // path is prefixed with icu version prefix such as "icudt65l-".
+    // Histogram only the part after the -.
+    if (icu_data_file_name.find("icudt") == 0) {
+      size_t dash = icu_data_file_name.find("-");
+      if (dash != std::string::npos) {
+        icu_data_file_name = icu_data_file_name.substr(dash + 1);
       }
-      base::UmaHistogramEnumeration(kICUCreateInstance, value);
-      return;
     }
+    // UmaHistogramSparse should track less than 100 values.
+    // We currently have about total 55 built-in data files inside ICU
+    // so it fit the UmaHistogramSparse usage.
+    int hash = base::HashMetricName(icu_data_file_name);
+    base::UmaHistogramSparse(kICUDataFile, hash);
   }
 }
 
@@ -491,7 +389,7 @@
   InitializeIcuTimeZone();
 
   const void* context = nullptr;
-  utrace_setFunctions(context, TraceICUEntry, nullptr, TraceICUData);
+  utrace_setFunctions(context, nullptr, nullptr, TraceICUDataFile);
   utrace_setLevel(UTRACE_VERBOSE);
   return true;
 }
