# Additional changes for disabling SafeBrowsing

--- a/content/browser/native_file_system/mock_native_file_system_permission_context.cc
+++ b/content/browser/native_file_system/mock_native_file_system_permission_context.cc
@@ -31,12 +31,4 @@ void MockNativeFileSystemPermissionConte
                                    frame_id, callback);
 }
 
-void MockNativeFileSystemPermissionContext::PerformSafeBrowsingChecks(
-    std::unique_ptr<NativeFileSystemWriteItem> item,
-    int process_id,
-    int frame_id,
-    base::OnceCallback<void(SafeBrowsingResult)> callback) {
-  PerformSafeBrowsingChecks_(item.get(), process_id, frame_id, callback);
-}
-
 }  // namespace content
--- a/content/browser/native_file_system/mock_native_file_system_permission_context.h
+++ b/content/browser/native_file_system/mock_native_file_system_permission_context.h
@@ -62,17 +62,6 @@ class MockNativeFileSystemPermissionCont
            int frame_id,
            base::OnceCallback<void(SensitiveDirectoryResult)>& callback));
 
-  void PerformSafeBrowsingChecks(
-      std::unique_ptr<NativeFileSystemWriteItem> item,
-      int process_id,
-      int frame_id,
-      base::OnceCallback<void(SafeBrowsingResult)> callback) override;
-  MOCK_METHOD4(PerformSafeBrowsingChecks_,
-               void(NativeFileSystemWriteItem* item,
-                    int process_id,
-                    int frame_id,
-                    base::OnceCallback<void(SafeBrowsingResult)>& callback));
-
   MOCK_METHOD1(CanRequestWritePermission, bool(const url::Origin& origin));
 };
 
