--- a/src/chrome/browser/autofill/autofill_cc_infobar_delegate.cc
+++ b/src/chrome/browser/autofill/autofill_cc_infobar_delegate.cc
@@ -29,8 +29,6 @@ AutoFillCCInfoBarDelegate::AutoFillCCInf
     // This is NULL for TestTabContents.
     if (tab_contents->delegate())
       browser_ = tab_contents->delegate()->GetBrowser();
-
-    tab_contents->AddInfoBar(this);
   }
 }
 
@@ -50,9 +48,7 @@ void AutoFillCCInfoBarDelegate::InfoBarC
     host_->OnInfoBarClosed(false);
     host_ = NULL;
   }
-
-  // This will delete us.
-  ConfirmInfoBarDelegate::InfoBarClosed();
+  delete this;
 }
 
 std::wstring AutoFillCCInfoBarDelegate::GetMessageText() const {
--- a/src/chrome/browser/autofill/autofill_manager.cc
+++ b/src/chrome/browser/autofill/autofill_manager.cc
@@ -442,10 +442,15 @@ void AutoFillManager::HandleSubmit() {
   CreditCard* credit_card;
   personal_data_->GetImportedFormData(&profile, &credit_card);
 
-  if (credit_card) {
-    cc_infobar_.reset(new AutoFillCCInfoBarDelegate(tab_contents_, this));
-  } else {
+  if (!credit_card) {
     UploadFormData();
+    return;
+  }
+
+  // Show an infobar to offer to save the credit card info.
+  if (tab_contents_) {
+    tab_contents_->AddInfoBar(new AutoFillCCInfoBarDelegate(tab_contents_,
+                                                            this));
   }
 }
 
--- a/src/chrome/browser/autofill/autofill_manager.h
+++ b/src/chrome/browser/autofill/autofill_manager.h
@@ -202,7 +202,8 @@ class AutoFillManager : public RenderVie
   scoped_ptr<FormStructure> upload_form_structure_;
 
   // The InfoBar that asks for permission to store credit card information.
-  scoped_ptr<AutoFillCCInfoBarDelegate> cc_infobar_;
+  // Deletes itself when closed.
+  AutoFillCCInfoBarDelegate* cc_infobar_;
 
   friend class TestAutoFillManager;
   FRIEND_TEST_ALL_PREFIXES(AutoFillManagerTest, FillCreditCardForm);
