--- a/src/third_party/WebKit/WebCore/editing/CompositeEditCommand.cpp
+++ b/src/third_party/WebKit/WebCore/editing/CompositeEditCommand.cpp
@@ -1183,6 +1183,8 @@ PassRefPtr<Node> CompositeEditCommand::s
 
     RefPtr<Node> node;
     for (node = start; node && node->parent() != end; node = node->parent()) {
+        if (!node->parent()->isElementNode())
+            break;
         VisiblePosition positionInParent(Position(node->parent(), 0), DOWNSTREAM);
         VisiblePosition positionInNode(Position(node, 0), DOWNSTREAM);
         if (positionInParent != positionInNode)
--- a/src/third_party/WebKit/WebCore/editing/IndentOutdentCommand.cpp
+++ b/src/third_party/WebKit/WebCore/editing/IndentOutdentCommand.cpp
@@ -130,6 +130,9 @@ void IndentOutdentCommand::indentIntoBlo
     else
         nodeToSplitTo = editableRootForPosition(start);
 
+    if (!nodeToSplitTo)
+        return;
+
     RefPtr<Node> outerBlock = (start.node() == nodeToSplitTo) ? start.node() : splitTreeToNode(start.node(), nodeToSplitTo);
 
     if (!targetBlockquote) {
