--- a/src/chrome/browser/download/download_exe.cc
+++ b/src/chrome/browser/download/download_exe.cc
@@ -56,6 +56,30 @@ namespace download_util {
  * ***** END LICENSE BLOCK ***** */
 
 static const char* const g_executables[] = {
+  "class",
+  "htm",
+  "html",
+  "jar",
+  "pdf",
+  "pdfxml",
+  "mars",
+  "fdf",
+  "xfdf",
+  "xdp",
+  "xfd",
+  "pl",
+  "py",
+  "rb",
+  "shtm",
+  "shtml",
+  "svg",
+  "swf",
+  "xht",
+  "xhtm",
+  "xhtml",
+  "xml",
+  "xsl",
+  "xslt",
 #if defined(OS_WIN)
   "ad",
   "ade",
@@ -76,13 +100,10 @@ static const char* const g_executables[]
   "fxp",
   "hlp",
   "hta",
-  "htm",
-  "html",
   "htt",
   "inf",
   "ins",
   "isp",
-  "jar",
   "js",
   "jse",
   "lnk",
@@ -125,9 +146,6 @@ static const char* const g_executables[]
   "sct",
   "shb",
   "shs",
-  "shtm",
-  "shtml",
-  "svg",
   "url",
   "vb",
   "vbe",
@@ -142,14 +160,9 @@ static const char* const g_executables[]
   "wsf",
   "wsh",
   "xbap",
-  "xht",
-  "xhtm",
-  "xhtml",
-  "xml",
-  "xsl",
-  "xslt",
 #elif defined(OS_MACOSX)
   // TODO(thakis): Figure out what makes sense here -- crbug.com/19096
+  "app",
   "dmg",
 #elif defined(OS_POSIX)
   // TODO(estade): lengthen this list.
@@ -158,9 +171,6 @@ static const char* const g_executables[]
   "deb",
   "exe",
   "ksh",
-  "pl",
-  "py",
-  "rb",
   "rpm",
   "sh",
   "tcsh",
--- a/src/chrome/browser/download/download_manager_unittest.cc
+++ b/src/chrome/browser/download/download_manager_unittest.cc
@@ -665,11 +665,11 @@ const struct {
     true,
     false,
     true, },
-  { "http://www.foo.com/auto-open.pdf",
+  { "http://www.foo.com/always_prompt.pdf",
     "application/pdf",
     false,
     true,
-    false, },
+    true, },
 };
 
 }  // namespace
