--- a/src/third_party/WebKit/WebCore/dom/Node.cpp
+++ b/src/third_party/WebKit/WebCore/dom/Node.cpp
@@ -3044,7 +3044,7 @@ void Node::defaultEventHandler(Event* ev
             if (Frame* frame = document()->frame())
                 frame->eventHandler()->defaultTextInputEventHandler(static_cast<TextEvent*>(event));
 #if ENABLE(PAN_SCROLLING)
-    } else if (eventType == eventNames().mousedownEvent) {
+    } else if (eventType == eventNames().mousedownEvent && event->isMouseEvent()) {
         MouseEvent* mouseEvent = static_cast<MouseEvent*>(event);
         if (mouseEvent->button() == MiddleButton) {
             if (enclosingLinkEventParentOrSelf())
