--- a/src/media/filters/ffmpeg_demuxer.cc
+++ b/src/media/filters/ffmpeg_demuxer.cc
@@ -532,7 +532,12 @@ void FFmpegDemuxer::DemuxTask() {
   // thread.
   DCHECK_GE(packet->stream_index, 0);
   DCHECK_LT(packet->stream_index, static_cast<int>(packet_streams_.size()));
-  FFmpegDemuxerStream* demuxer_stream = packet_streams_[packet->stream_index];
+  FFmpegDemuxerStream* demuxer_stream = NULL;
+  size_t i = packet->stream_index;
+  // Defend against ffmpeg giving us a bad stream index.
+  if (i < packet_streams_.size()) {
+    demuxer_stream = packet_streams_[i];
+  }
   if (demuxer_stream) {
     // Queue the packet with the appropriate stream.  The stream takes
     // ownership of the AVPacket.
