--- a/src/third_party/WebKit/WebCore/page/EventHandler.cpp
+++ b/src/third_party/WebKit/WebCore/page/EventHandler.cpp
@@ -595,13 +595,13 @@ void EventHandler::updateSelectionForMou
     if (!targetNode)
         return;
 
+    if (!canMouseDragExtendSelect(targetNode))
+	return;
+
     RenderObject* targetRenderer = targetNode->renderer();
     if (!targetRenderer)
         return;
         
-    if (!canMouseDragExtendSelect(targetNode))
-        return;
-
     VisiblePosition targetPosition(targetRenderer->positionForPoint(localPoint));
 
     // Don't modify the selection if we're not on a node.
