--- a/src/third_party/WebKit/WebCore/css/CSSParser.cpp
+++ b/src/third_party/WebKit/WebCore/css/CSSParser.cpp
@@ -3561,7 +3561,7 @@ bool CSSParser::parseFontFaceSrc()
             // There are two allowed functions: local() and format().
             CSSParserValueList* args = val->function->args;
             if (args && args->size() == 1) {
-                if (equalIgnoringCase(val->function->name, "local(") && !expectComma) {
+                if (equalIgnoringCase(val->function->name, "local(") && !expectComma && (args->current()->unit == CSSPrimitiveValue::CSS_STRING || args->current()->unit == CSSPrimitiveValue::CSS_IDENT)) {
                     expectComma = true;
                     allowFormat = false;
                     CSSParserValue* a = args->current();
