--- a/src/third_party/WebKit/WebCore/dom/Document.cpp
+++ b/src/third_party/WebKit/WebCore/dom/Document.cpp
@@ -3069,6 +3069,18 @@ void Document::detachNodeIterator(NodeIt
     m_nodeIterators.remove(ni);
 }
 
+void Document::moveNodeIteratorsToNewDocument(Node* node, Document* newDocument)
+{
+    HashSet<NodeIterator*> nodeIteratorsList = m_nodeIterators;
+    HashSet<NodeIterator*>::const_iterator nodeIteratorsEnd = nodeIteratorsList.end();
+    for (HashSet<NodeIterator*>::const_iterator it = nodeIteratorsList.begin(); it != nodeIteratorsEnd; ++it) {
+        if ((*it)->root() == node) {
+            detachNodeIterator(*it);
+            newDocument->attachNodeIterator(*it);
+        }
+    }
+}
+
 void Document::nodeChildrenChanged(ContainerNode* container)
 {
     if (!disableRangeMutation(page())) {
--- a/src/third_party/WebKit/WebCore/dom/Document.h
+++ b/src/third_party/WebKit/WebCore/dom/Document.h
@@ -632,6 +632,7 @@ public:
 
     void attachNodeIterator(NodeIterator*);
     void detachNodeIterator(NodeIterator*);
+    void moveNodeIteratorsToNewDocument(Node*, Document*);
 
     void attachRange(Range*);
     void detachRange(Range*);
--- a/src/third_party/WebKit/WebCore/dom/Node.cpp
+++ b/src/third_party/WebKit/WebCore/dom/Node.cpp
@@ -439,8 +439,10 @@ void Node::setDocument(Document* documen
         document->addNodeListCache();
     }
 
-    if (m_document)
+    if (m_document) {
+        m_document->moveNodeIteratorsToNewDocument(this, document);
         m_document->selfOnlyDeref();
+    }
 
     m_document = document;
 
