--- a/src/third_party/WebKit/WebCore/css/CSSCursorImageValue.cpp
+++ b/src/third_party/WebKit/WebCore/css/CSSCursorImageValue.cpp
@@ -70,7 +70,7 @@ CSSCursorImageValue::~CSSCursorImageValu
 
     for (; it != end; ++it) {
         SVGElement* referencedElement = *it;
-        referencedElement->cursorElementRemoved();
+        referencedElement->cursorImageValueRemoved();
         if (SVGCursorElement* cursorElement = resourceReferencedByCursorElement(url, referencedElement->document()))
             cursorElement->removeClient(referencedElement);
     }
--- a/src/third_party/WebKit/WebCore/svg/SVGElement.cpp
+++ b/src/third_party/WebKit/WebCore/svg/SVGElement.cpp
@@ -207,7 +207,7 @@ void SVGElement::setCursorImageValue(CSS
     rareData->setCursorImageValue(cursorImageValue);
 }
 
-void SVGElement::cursorImageElementRemoved()
+void SVGElement::cursorImageValueRemoved()
 {
     ASSERT(hasRareSVGData());
     rareSVGData()->setCursorImageValue(0);
--- a/src/third_party/WebKit/WebCore/svg/SVGElement.h
+++ b/src/third_party/WebKit/WebCore/svg/SVGElement.h
@@ -79,7 +79,7 @@ namespace WebCore {
         void setCursorElement(SVGCursorElement*);
         void cursorElementRemoved();
         void setCursorImageValue(CSSCursorImageValue*);
-        void cursorImageElementRemoved();
+        void cursorImageValueRemoved();
 
         virtual void updateAnimatedSVGAttribute(const QualifiedName&) const;
 
