---
 build/common.gypi |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/src/build/common.gypi
+++ b/src/build/common.gypi
@@ -86,6 +86,10 @@
       # Python version.
       'python_ver%': '2.5',
 
+      # Set to 1 to build with -sse2. As it causes SIGILL on older CPUs, it must
+      # be set to 0 when a broader audience is targetted.
+      'enable_sse2%': 1,
+
       # Set ARM-v7 compilation flags
       'armv7%': 0,
 
@@ -108,6 +112,7 @@
     'fastbuild%': '<(fastbuild)',
     'linux_fpic%': '<(linux_fpic)',
     'python_ver%': '<(python_ver)',
+    'enable_sse2%': '<(enable_sse2)',
     'armv7%': '<(armv7)',
     'arm_neon%': '<(arm_neon)',
     'sysroot%': '<(sysroot)',
@@ -882,7 +887,7 @@
             # compiler optimized the code, since the value is always kept
             # in its specified precision.
             'conditions': [
-              ['branding=="Chromium"', {
+              ['branding=="Chromium" and enable_sse2==1', {
                 'cflags': [
                   '-march=pentium4',
                   '-msse2',
