Index: sid/src/build/common.gypi
===================================================================
--- sid.orig/src/build/common.gypi	2013-01-11 13:03:22.250737845 +0100
+++ sid/src/build/common.gypi	2013-01-11 13:03:57.050738219 +0100
@@ -815,7 +815,7 @@
     'arm_fpu%': 'vfpv3',
 
     # Set ARM float abi compilation flag.
-    'arm_float_abi%': 'softfp',
+    #'arm_float_abi%': 'softfp',
 
     # Enable new NPDevice API.
     'enable_new_npdevice_api%': 0,
