Index: sid/src/build/common.gypi
===================================================================
--- sid.orig/src/build/common.gypi	2013-02-22 13:16:14.915093970 +0100
+++ sid/src/build/common.gypi	2013-02-22 13:16:37.555093429 +0100
@@ -868,7 +868,7 @@
     'arm_fpu%': 'vfpv3',
 
     # Set ARM float abi compilation flag.
-    'arm_float_abi%': 'softfp',
+    #'arm_float_abi%': 'softfp',
 
     # Enable new NPDevice API.
     'enable_new_npdevice_api%': 0,
