Index: sid/src/build/common.gypi
===================================================================
--- sid.orig/src/build/common.gypi	2012-05-16 15:53:49.278125377 +0200
+++ sid/src/build/common.gypi	2012-05-17 15:01:32.214355670 +0200
@@ -1962,7 +1962,6 @@
                     'cflags': [
                       '-march=armv7-a',
                       '-mtune=cortex-a8',
-                      '-mfloat-abi=softfp',
                     ],
                     'conditions': [
                       ['arm_neon==1', {
