Index: sid/src/build/common.gypi
===================================================================
--- sid.orig/src/build/common.gypi	2012-08-06 12:15:02.377549576 +0200
+++ sid/src/build/common.gypi	2012-08-06 12:15:13.853549209 +0200
@@ -732,7 +732,7 @@
     'arm_fpu%': 'vfpv3',
 
     # Set ARM float abi compilation flag.
-    'arm_float_abi%': 'softfp',
+    #'arm_float_abi%': 'softfp',
 
     # Enable new NPDevice API.
     'enable_new_npdevice_api%': 0,
