Index: sid/src/build/common.gypi
===================================================================
--- sid.orig/src/build/common.gypi	2012-09-26 11:06:40.322706593 +0200
+++ sid/src/build/common.gypi	2012-09-26 11:06:54.338706295 +0200
@@ -779,7 +779,7 @@
     'arm_fpu%': 'vfpv3',
 
     # Set ARM float abi compilation flag.
-    'arm_float_abi%': 'softfp',
+    #'arm_float_abi%': 'softfp',
 
     # Enable new NPDevice API.
     'enable_new_npdevice_api%': 0,
