Index: exp/src/build/common.gypi
===================================================================
--- exp.orig/src/build/common.gypi	2012-05-19 19:23:03.000000000 +0200
+++ exp/src/build/common.gypi	2012-05-19 19:40:07.198682266 +0200
@@ -723,7 +723,7 @@
     'arm_fpu%': 'vfpv3',
 
     # Set ARM float abi compilation flag.
-    'arm_float_abi%': 'softfp',
+    #'arm_float_abi%': 'softfp',
 
     # Enable new NPDevice API.
     'enable_new_npdevice_api%': 0,
