Do not use armv4 incopatible code
--- a/src/third_party/skia/include/core/SkMath.h
+++ b/src/third_party/skia/include/core/SkMath.h
@@ -170,7 +170,7 @@ static inline int SkNextLog2(uint32_t va
 #if defined(__arm__) \
   && !defined(__thumb__) \
   && !defined(__ARM_ARCH_4T__) \
-  && !defined(__ARM_ARCH_5T__)
+  && 0
     static inline int32_t SkMulS16(S16CPU x, S16CPU y) {
         SkASSERT((int16_t)x == x);
         SkASSERT((int16_t)y == y);
--- a/src/build/common.gypi
+++ b/src/build/common.gypi
@@ -1096,6 +1096,7 @@
                   # silences those warnings, as they are not helpful and
                   # clutter legitimate warnings.
                   '-Wno-abi',
+                  '-fno-tree-sink',
                 ],
                 'conditions': [
                   ['arm_thumb == 1', {
--- a/src/base/debug/debugger_posix.cc
+++ b/src/base/debug/debugger_posix.cc
@@ -152,7 +152,7 @@ bool DebugUtil::BeingDebugged() {
 #if defined(NDEBUG) && !defined(OS_MACOSX)
 #define DEBUG_BREAK() abort()
 #elif defined(ARCH_CPU_ARM_FAMILY)
-#define DEBUG_BREAK() asm("bkpt 0")
+#define DEBUG_BREAK() asm("nop")
 #else
 #define DEBUG_BREAK() asm("int3")
 #endif
