Do not use armv4 incopatible code
--- a/src/third_party/skia/include/core/SkMath.h
+++ b/src/third_party/skia/include/core/SkMath.h
@@ -163,7 +163,7 @@ static inline bool SkIsPow2(int value) {
 #if defined(__arm__) \
   && !defined(__thumb__) \
   && !defined(__ARM_ARCH_4T__) \
-  && !defined(__ARM_ARCH_5T__)
+  && 0
     static inline int32_t SkMulS16(S16CPU x, S16CPU y) {
         SkASSERT((int16_t)x == x);
         SkASSERT((int16_t)y == y);
--- a/src/build/common.gypi
+++ b/src/build/common.gypi
@@ -1426,6 +1426,7 @@
                   # silences those warnings, as they are not helpful and
                   # clutter legitimate warnings.
                   '-Wno-abi',
+                  '-fno-tree-sink',
                 ],
                 'conditions': [
                   ['arm_thumb == 1', {
--- a/src/base/debug/debugger_posix.cc
+++ b/src/base/debug/debugger_posix.cc
@@ -187,7 +187,7 @@ bool BeingDebugged() {
 } while (0)
 #else
 // ARM && !ANDROID
-#define DEBUG_BREAK() asm("bkpt 0")
+#define DEBUG_BREAK() asm("nop")
 #endif
 #else
 #define DEBUG_BREAK() asm("int3")
--- a/src/third_party/skia/src/opts/SkBlitRow_opts_arm.cpp
+++ b/src/third_party/skia/src/opts/SkBlitRow_opts_arm.cpp
@@ -685,8 +685,8 @@ static void S32A_Blend_BlitRow32_arm(SkP
                   /* dst1_scale and dst2_scale*/
                   "lsr    r9, r5, #24                \n\t" /* src >> 24 */
                   "lsr    r10, r6, #24               \n\t" /* src >> 24 */
-                  "smulbb r9, r9, %[alpha]           \n\t" /* r9 = SkMulS16 r9 with src_scale */
-                  "smulbb r10, r10, %[alpha]         \n\t" /* r10 = SkMulS16 r10 with src_scale */
+                  "mul    r9, r9, %[alpha]           \n\t" /* r9 = SkMulS16 r9 with src_scale */
+                  "mul    r10, r10, %[alpha]         \n\t" /* r10 = SkMulS16 r10 with src_scale */
                   "lsr    r9, r9, #8                 \n\t" /* r9 >> 8 */
                   "lsr    r10, r10, #8               \n\t" /* r10 >> 8 */
                   "rsb    r9, r9, #256               \n\t" /* dst1_scale = r9 = 255 - r9 + 1 */
@@ -755,7 +755,7 @@ static void S32A_Blend_BlitRow32_arm(SkP
 
                   "lsr    r6, r5, #24                \n\t" /* src >> 24 */
                   "and    r8, r12, r5, lsr #8        \n\t" /* ag = r8 = r5 masked by r12 lsr by #8 */
-                  "smulbb r6, r6, %[alpha]           \n\t" /* r6 = SkMulS16 with src_scale */
+                  "mul    r6, r6, %[alpha]           \n\t" /* r6 = SkMulS16 with src_scale */
                   "and    r9, r12, r5                \n\t" /* rb = r9 = r5 masked by r12 */
                   "lsr    r6, r6, #8                 \n\t" /* r6 >> 8 */
                   "mul    r8, r8, %[alpha]           \n\t" /* ag = r8 times scale */
