Do not use armv4 incopatible code
Index: sid/src/build/common.gypi
===================================================================
--- sid.orig/src/build/common.gypi	2013-06-24 11:48:31.368039228 +0200
+++ sid/src/build/common.gypi	2013-06-24 11:48:31.364039228 +0200
@@ -2924,6 +2924,7 @@
                   # silences those warnings, as they are not helpful and
                   # clutter legitimate warnings.
                   '-Wno-abi',
+                  '-fno-tree-sink',
                 ],
                 'conditions': [
                   ['arm_arch!=""', {
