Do not use armv4 incopatible code
Index: chromium/src/build/common.gypi
===================================================================
--- chromium.orig/src/build/common.gypi	2013-08-24 20:10:50.661823606 +0000
+++ chromium/src/build/common.gypi	2013-08-24 20:10:50.661823606 +0000
@@ -2961,6 +2961,7 @@
                   # silences those warnings, as they are not helpful and
                   # clutter legitimate warnings.
                   '-Wno-abi',
+                  '-fno-tree-sink',
                 ],
                 'conditions': [
                   ['arm_arch!=""', {
