Do not use armv4 incopatible code
Index: chromium/src/build/common.gypi
===================================================================
--- chromium.orig/src/build/common.gypi	2013-05-22 03:14:34.821477965 +0000
+++ chromium/src/build/common.gypi	2013-05-22 03:14:34.817477965 +0000
@@ -2745,6 +2745,7 @@
                   # silences those warnings, as they are not helpful and
                   # clutter legitimate warnings.
                   '-Wno-abi',
+                  '-fno-tree-sink',
                 ],
                 'conditions': [
                   ['arm_thumb==1 and android_webview_build==0', {
