Do not use armv4 incopatible code
--- a/src/third_party/skia/include/core/SkMath.h
+++ b/src/third_party/skia/include/core/SkMath.h
@@ -177,7 +177,7 @@ static inline bool SkIsPow2(int value) {
 #if defined(__arm__) \
   && !defined(__thumb__) \
   && !defined(__ARM_ARCH_4T__) \
-  && !defined(__ARM_ARCH_5T__)
+  && 0
     static inline int32_t SkMulS16(S16CPU x, S16CPU y) {
         SkASSERT((int16_t)x == x);
         SkASSERT((int16_t)y == y);
--- a/src/build/common.gypi
+++ b/src/build/common.gypi
@@ -1100,6 +1100,7 @@
                   # silences those warnings, as they are not helpful and
                   # clutter legitimate warnings.
                   '-Wno-abi',
+                  '-fno-tree-sink',
                 ],
                 'conditions': [
                   ['arm_thumb == 1', {
--- a/src/base/debug/debugger_posix.cc
+++ b/src/base/debug/debugger_posix.cc
@@ -167,7 +167,7 @@ bool BeingDebugged() {
 // http://code.google.com/p/nativeclient/issues/detail?id=645
 #define DEBUG_BREAK() abort()
 #elif defined(ARCH_CPU_ARM_FAMILY)
-#define DEBUG_BREAK() asm("bkpt 0")
+#define DEBUG_BREAK() asm("nop")
 #else
 #define DEBUG_BREAK() asm("int3")
 #endif
