Do not use armv4 incopatible code
Index: sid/src/build/common.gypi
===================================================================
--- sid.orig/src/build/common.gypi	2013-03-16 09:36:04.000000000 +0100
+++ sid/src/build/common.gypi	2013-03-30 17:36:33.403742676 +0100
@@ -2662,6 +2662,7 @@
                   # silences those warnings, as they are not helpful and
                   # clutter legitimate warnings.
                   '-Wno-abi',
+                  '-fno-tree-sink',
                 ],
                 'conditions': [
                   ['arm_thumb==1', {
Index: sid/src/base/debug/debugger_posix.cc
===================================================================
--- sid.orig/src/base/debug/debugger_posix.cc	2013-03-16 09:36:05.000000000 +0100
+++ sid/src/base/debug/debugger_posix.cc	2013-03-30 17:36:33.407742677 +0100
@@ -230,7 +230,7 @@
 } while (0)
 #else
 // ARM && !ANDROID
-#define DEBUG_BREAK() asm("bkpt 0")
+#define DEBUG_BREAK() asm("nop")
 #endif
 #elif defined(ARCH_CPU_MIPS_FAMILY)
 #define DEBUG_BREAK() asm("break 2")
