description: set chromedriver version as undefined

Index: chromium/src/chrome/test/chromedriver/embed_version_in_cpp.py
===================================================================
--- chromium.orig/src/chrome/test/chromedriver/embed_version_in_cpp.py	2014-04-11 01:43:42.224278647 +0000
+++ chromium/src/chrome/test/chromedriver/embed_version_in_cpp.py	2014-04-11 01:43:42.220278647 +0000
@@ -25,7 +25,7 @@
   options, args = parser.parse_args()
 
   version = open(options.version_file, 'r').read().strip()
-  revision = lastchange.FetchVersionInfo(None).revision
+  revision = "undefined"
   if revision:
     version += '.' + revision.strip()
 
