---
 webkit/glue/webkit_glue.cc |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

Index: src/webkit/glue/webkit_glue.cc
===================================================================
--- src.orig/webkit/glue/webkit_glue.cc
+++ src/webkit/glue/webkit_glue.cc
@@ -366,18 +366,24 @@
   // Get the product name and version, and replace Safari's Version/X string
   // with it.  This is done to expose our product name in a manner that is
   // maximally compatible with Safari, we hope!!
   std::string product;
 
   if (!mimic_safari) {
     scoped_ptr<FileVersionInfo> version_info(
         FileVersionInfo::CreateFileVersionInfoForCurrentModule());
-    if (version_info.get())
+    if (version_info.get()) {
+      std::string product_name;
+
+      product_name = WideToASCII(version_info->product_name());
       product = "Chrome/" + WideToASCII(version_info->product_version());
+      if (LowerCaseEqualsASCII(product_name, "chromium"))
+        product += " " + product_name + "/" + WideToASCII(version_info->product_version());
+    }
   }
 
   if (product.empty())
     product = "Version/3.2.1";
 
   // Derived from Safari's UA string.
   StringAppendF(
       result,
