---
 webkit/glue/webkit_glue.cc |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/webkit/glue/webkit_glue.cc
===================================================================
--- src.orig/webkit/glue/webkit_glue.cc
+++ src/webkit/glue/webkit_glue.cc
@@ -367,17 +367,17 @@
   // with it.  This is done to expose our product name in a manner that is
   // maximally compatible with Safari, we hope!!
   std::string product;
 
   if (!mimic_safari) {
     scoped_ptr<FileVersionInfo> version_info(
         FileVersionInfo::CreateFileVersionInfoForCurrentModule());
     if (version_info.get())
-      product = "Chrome/" + WideToASCII(version_info->product_version());
+      product = WideToASCII(version_info->product_name()) + "/" + WideToASCII(version_info->product_version());
   }
 
   if (product.empty())
     product = "Version/3.2.1";
 
   // Derived from Safari's UA string.
   StringAppendF(
       result,
