description: set chromedriver version as undefined

Index: sid/src/chrome/test/chromedriver/embed_version_in_cpp.py
===================================================================
--- sid.orig/src/chrome/test/chromedriver/embed_version_in_cpp.py	2014-03-14 17:20:19.260905865 +0100
+++ sid/src/chrome/test/chromedriver/embed_version_in_cpp.py	2014-03-14 17:20:19.256905865 +0100
@@ -25,7 +25,7 @@
   options, args = parser.parse_args()
 
   version = open(options.version_file, 'r').read().strip()
-  revision = lastchange.FetchVersionInfo(None).revision
+  revision = "undefined"
   if revision:
     version += '.' + revision.strip()
 
