Description: Fix build failure due to cups 1.5 header changes
Author: Michael Gilbert <michael.s.gilbert@gmail.com>
Index: sid/src/printing/backend/cups_helper.h
===================================================================
--- sid.orig/src/printing/backend/cups_helper.h	2012-08-06 10:27:46.000000000 +0200
+++ sid/src/printing/backend/cups_helper.h	2012-08-06 12:15:11.021549299 +0200
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <cups/cups.h>
+#include <cups/ppd.h>
 
 #include "printing/printing_export.h"
 
Index: sid/src/chrome/service/cloud_print/print_system_cups.cc
===================================================================
--- sid.orig/src/chrome/service/cloud_print/print_system_cups.cc	2012-08-06 10:29:01.000000000 +0200
+++ sid/src/chrome/service/cloud_print/print_system_cups.cc	2012-08-06 12:15:11.021549299 +0200
@@ -5,6 +5,7 @@
 #include "chrome/service/cloud_print/print_system.h"
 
 #include <cups/cups.h>
+#include <cups/ppd.h>
 #include <dlfcn.h>
 #include <errno.h>
 #include <pthread.h>
