Description: Fix build failure due to cups 1.5 header changes
Author: Michael Gilbert <michael.s.gilbert@gmail.com>
--- a/src/printing/backend/cups_helper.h
+++ b/src/printing/backend/cups_helper.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <cups/cups.h>
+#include <cups/ppd.h>
 
 class GURL;
 
--- a/src/chrome/service/cloud_print/print_system_cups.cc
+++ b/src/chrome/service/cloud_print/print_system_cups.cc
@@ -5,6 +5,7 @@
 #include "chrome/service/cloud_print/print_system.h"
 
 #include <cups/cups.h>
+#include <cups/ppd.h>
 #include <dlfcn.h>
 #include <errno.h>
 #include <pthread.h>
--- a/src/chrome/browser/ui/webui/print_preview_handler.cc
+++ b/src/chrome/browser/ui/webui/print_preview_handler.cc
@@ -46,6 +46,7 @@
 
 #if defined(USE_CUPS)
 #include <cups/cups.h>
+#include <cups/ppd.h>
 
 #include "base/file_util.h"
 #endif
