---
 base/logging.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/base/logging.h
+++ b/src/base/logging.h
@@ -416,7 +416,7 @@ DECLARE_CHECK_STROP_IMPL(_stricmp, false
 // uses OFFICIAL_BUILD and other platforms use the branding flag when NDEBUG is
 // defined.
 #if ( defined(OS_WIN) && defined(OFFICIAL_BUILD)) || \
-    (!defined(OS_WIN) && defined(NDEBUG) && defined(GOOGLE_CHROME_BUILD))
+    (!defined(OS_WIN) && defined(NDEBUG))
 // In order to have optimized code for official builds, remove DLOGs and
 // DCHECKs.
 #define OMIT_DLOG_AND_DCHECK 1
