---
 base/logging.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/base/logging.h
===================================================================
--- src.orig/base/logging.h
+++ src/base/logging.h
@@ -382,17 +382,17 @@
 //   #ifndef NDEBUG
 //     foo.CheckThatFoo();
 //   #endif
 
 // http://crbug.com/16512 is open for a real fix for this.  For now, Windows
 // uses OFFICIAL_BUILD and other platforms use the branding flag when NDEBUG is
 // defined.
 #if ( defined(OS_WIN) && defined(OFFICIAL_BUILD)) || \
-    (!defined(OS_WIN) && defined(NDEBUG) && defined(GOOGLE_CHROME_BUILD))
+    (!defined(OS_WIN) && defined(NDEBUG))
 // In order to have optimized code for official builds, remove DLOGs and
 // DCHECKs.
 #define OMIT_DLOG_AND_DCHECK 1
 #endif
 
 #ifdef OMIT_DLOG_AND_DCHECK
 
 #define DLOG(severity) \
