--- a/src/build/common.gypi
+++ b/src/build/common.gypi
@@ -609,17 +609,6 @@
           'enable_settings_app%': 0,
         }],
 
-        ['OS=="linux" and target_arch=="arm" and chromeos==0', {
-          # Set some defaults for arm/linux chrome builds
-          'armv7%': 1,
-          'linux_breakpad%': 0,
-          'linux_use_tcmalloc%': 0,
-          'linux_use_gold_flags%': 0,
-          # sysroot needs to be an absolute path otherwise it generates
-          # incorrect results when passed to pkg-config
-          'sysroot%': '<!(cd <(DEPTH) && pwd -P)/arm-sysroot',
-        }], # OS=="linux" and target_arch=="arm" and chromeos==0
-
         ['target_arch=="mipsel"', {
           'sysroot': '<!(cd <(DEPTH) && pwd -P)/native_client/toolchain/linux_mips-trusted/sysroot',
         }],
