---
 v8/tools/gyp/v8.gyp |   12 ------------
 1 file changed, 12 deletions(-)

Index: src/v8/tools/gyp/v8.gyp
===================================================================
--- src.orig/v8/tools/gyp/v8.gyp
+++ src/v8/tools/gyp/v8.gyp
@@ -196,20 +196,16 @@
         '<(SHARED_INTERMEDIATE_DIR)/libraries.cc',
         '../../src/snapshot-empty.cc',
       ],
       'conditions': [
         # The ARM assembler assumes the host is 32 bits, so force building
         # 32-bit host tools.
         # TODO(piman): This assumes that the host is ia32 or amd64. Fixing the
         # code would be better
-        ['target_arch=="arm" and _toolset=="host"', {
-          'cflags': ['-m32'],
-          'ldflags': ['-m32'],
-        }]
       ]
     },
     {
       'target_name': 'v8_base',
       'type': '<(library)',
       'toolsets': ['host', 'target'],
       'include_dirs+': [
         '../../src',
@@ -425,20 +421,16 @@
             '../../src/arm/virtual-frame-arm.cc',
             '../../src/arm/virtual-frame-arm.h',
           ],
           'conditions': [
             # The ARM assembler assumes the host is 32 bits, so force building
             # 32-bit host tools.
             # TODO(piman): This assumes that the host is ia32 or amd64. Fixing
             # the code would be better
-            ['_toolset=="host"', {
-              'cflags': ['-m32'],
-              'ldflags': ['-m32'],
-            }]
           ]
         }],
         ['target_arch=="ia32"', {
           'include_dirs+': [
             '../../src/ia32',
           ],
           'sources': [
             '../../src/ia32/assembler-ia32-inl.h',
@@ -595,20 +587,16 @@
       'sources': [
         '../../src/mksnapshot.cc',
       ],
       'conditions': [
         # The ARM assembler assumes the host is 32 bits, so force building
         # 32-bit host tools.
         # TODO(piman): This assumes that the host is ia32 or amd64. Fixing
         # the code would be better
-        ['target_arch=="arm" and _toolset=="host"', {
-          'cflags': ['-m32'],
-          'ldflags': ['-m32'],
-        }]
       ]
     },
     {
       'target_name': 'v8_shell',
       'type': 'executable',
       'dependencies': [
         'v8'
       ],
