Include system copy of expat.h
Index: sid/src/third_party/libjingle/source/talk/xmllite/xmlparser.h
===================================================================
--- sid.orig/src/third_party/libjingle/source/talk/xmllite/xmlparser.h	2014-02-02 19:27:53.356401802 +0100
+++ sid/src/third_party/libjingle/source/talk/xmllite/xmlparser.h	2014-02-02 19:27:53.348401802 +0100
@@ -32,9 +32,9 @@
 
 #include "talk/xmllite/xmlnsstack.h"
 #ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
+#include <expat.h>
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 struct XML_ParserStruct;
Index: sid/src/third_party/libjingle/source/talk/xmpp/xmppstanzaparser.cc
===================================================================
--- sid.orig/src/third_party/libjingle/source/talk/xmpp/xmppstanzaparser.cc	2014-02-02 19:27:53.356401802 +0100
+++ sid/src/third_party/libjingle/source/talk/xmpp/xmppstanzaparser.cc	2014-02-02 19:27:53.348401802 +0100
@@ -31,9 +31,9 @@
 #include "talk/base/common.h"
 #include "talk/xmpp/constants.h"
 #ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
+#include <expat.h>
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif
 
 namespace buzz {
Index: sid/src/third_party/skia/src/ports/SkXMLParser_expat.cpp
===================================================================
--- sid.orig/src/third_party/skia/src/ports/SkXMLParser_expat.cpp	2014-02-02 19:27:53.356401802 +0100
+++ sid/src/third_party/skia/src/ports/SkXMLParser_expat.cpp	2014-02-02 19:27:53.348401802 +0100
@@ -11,7 +11,7 @@
 #include "SkString.h"
 #include "SkStream.h"
 
-#include "expat.h"
+#include <expat.h>
 
 #ifdef SK_BUILD_FOR_PPI
 #define CHAR_16_TO_9
Index: sid/src/third_party/skia/src/ports/SkXMLPullParser_expat.cpp
===================================================================
--- sid.orig/src/third_party/skia/src/ports/SkXMLPullParser_expat.cpp	2014-02-02 19:27:53.356401802 +0100
+++ sid/src/third_party/skia/src/ports/SkXMLPullParser_expat.cpp	2014-02-02 19:27:53.348401802 +0100
@@ -12,7 +12,7 @@
 #include "SkString.h"
 #include "SkStream.h"
 
-#include "expat.h"
+#include <expat.h>
 
 static inline char* dupstr(SkChunkAlloc& chunk, const char src[], size_t len)
 {
Index: sid/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h
===================================================================
--- sid.orig/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h	2014-02-02 19:27:53.356401802 +0100
+++ sid/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h	2014-02-02 19:27:53.352401802 +0100
@@ -33,11 +33,7 @@
 #include "talk/base/scoped_ptr.h"
 #include "talk/xmllite/xmlparser.h"
 
-#ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
-#else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
-#endif  // EXPAT_RELATIVE_PATH
+#include <expat.h>
 
 namespace buzz {
 
