Include system copy of expat.h

--- a/third_party/libjingle/source/talk/xmllite/xmlparser.h
+++ b/third_party/libjingle/source/talk/xmllite/xmlparser.h
@@ -32,9 +32,9 @@
 
 #include "talk/xmllite/xmlnsstack.h"
 #ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
+#include <expat.h>
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 struct XML_ParserStruct;
--- a/third_party/libjingle/source/talk/xmpp/xmppstanzaparser.cc
+++ b/third_party/libjingle/source/talk/xmpp/xmppstanzaparser.cc
@@ -31,9 +31,9 @@
 #include "webrtc/libjingle/xmpp/constants.h"
 #include "webrtc/base/common.h"
 #ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
+#include <expat.h>
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif
 
 namespace buzz {
--- a/third_party/libjingle/source/talk/xmllite/xmlbuilder.h
+++ b/third_party/libjingle/source/talk/xmllite/xmlbuilder.h
@@ -33,11 +33,7 @@
 #include "talk/xmllite/xmlparser.h"
 #include "webrtc/base/scoped_ptr.h"
 
-#ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
-#else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
-#endif  // EXPAT_RELATIVE_PATH
+#include <expat.h>
 
 namespace buzz {
 
