--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -344,6 +344,11 @@
           "//ui/base",
         ]
       }
+
+      if (current_cpu == "arm") {
+        lib_dirs = [ "/usr/lib/llvm-13/lib/clang/13.0.1/lib/linux" ]
+        libs = [ ":libclang_rt.builtins-armhf.a" ]
+      }
     }
 
     if (is_chromeos_ash || is_win) {
--- a/tools/v8_context_snapshot/BUILD.gn
+++ b/tools/v8_context_snapshot/BUILD.gn
@@ -93,6 +93,11 @@
         "//v8:external_startup_data",
         ":disable_icf",
       ]
+
+      if (current_cpu == "arm") {
+        lib_dirs = [ "/usr/lib/llvm-13/lib/clang/13.0.1/lib/linux" ]
+        libs = [ ":libclang_rt.builtins-armhf.a" ]
+      }
     }
   }
 
--- a/v8/BUILD.gn
+++ b/v8/BUILD.gn
@@ -6374,6 +6374,11 @@
       ":v8_turboshaft",
       "//build/win:default_exe_manifest",
     ]
+
+    if (current_cpu == "arm") {
+      lib_dirs = [ "/usr/lib/llvm-13/lib/clang/13.0.1/lib/linux" ]
+      libs = [ ":libclang_rt.builtins-armhf.a" ]
+    }
   }
 }
 
