Description: Removed warning in help option.
Forwarded: http://code.google.com/p/chromium/issues/detail?id=73591
Author: Daniel Echeverry <epsilon77@gmail.com>
Origin: vendor
Last-Update: 2011-02-20

--- a/src/chrome/app/resources/manpage.1.in
+++ b/src/chrome/app/resources/manpage.1.in
@@ -103,6 +103,7 @@
 See the GTK documentation for more:
 .IP
 <http://library.gnome.org/devel/gtk/stable/gtk-running.html>
+.IP
 <http://library.gnome.org/devel/gtk/stable/gtk-x11.html>
 
 .SH ENVIRONMENT
