Description: Removed warning in help option.
Forwarded: http://code.google.com/p/chromium/issues/detail?id=73591
Author: Daniel Echeverry <epsilon77@gmail.com>
Origin: vendor
Last-Update: 2011-02-20

Index: sid/src/chrome/app/resources/manpage.1.in
===================================================================
--- sid.orig/src/chrome/app/resources/manpage.1.in	2012-08-06 10:28:20.000000000 +0200
+++ sid/src/chrome/app/resources/manpage.1.in	2012-08-06 12:15:05.973549461 +0200
@@ -1,5 +1,5 @@
-." This file is processed by chrome.gyp to generate manpages in the
-." build diretory.
+.\" This file is processed by chrome.gyp to generate manpages in the
+.\" build diretory.
 .TH @@FILENAME@@ 1 "" "" "USER COMMANDS"
 
 .SH NAME
@@ -110,6 +110,7 @@
 See the GTK documentation for more:
 .IP
 <http://library.gnome.org/devel/gtk/stable/gtk-running.html>
+.IP
 <http://library.gnome.org/devel/gtk/stable/gtk-x11.html>
 
 .SH ENVIRONMENT
