--- a/third_party/node/node.py
+++ b/third_party/node/node.py
@@ -11,14 +11,13 @@
 
 
 def GetBinaryPath():
-  darwin_name = ('node-darwin-arm64' if platform.machine() == 'arm64' else
-                 'node-darwin-x64')
-  return os_path.join(os_path.dirname(__file__), *{
-    'Darwin': ('mac', darwin_name, 'bin', 'node'),
-    'Linux': ('linux', 'node-linux-x64', 'bin', 'node'),
-    'Windows': ('win', 'node.exe'),
-  }[platform.system()])
-
+  if platform.machine() == 'x86_64':
+    return os_path.join(os_path.dirname(__file__), 'linux', 'node-linux-x64', 'bin', 'node')
+  if platform.machine() == 'armv7l':
+    return os_path.join(os_path.dirname(__file__), 'linux', 'node-linux-armv7l', 'bin', 'node')
+  if platform.machine() == 'aarch64':
+    return os_path.join(os_path.dirname(__file__), 'linux', 'node-linux-arm64', 'bin', 'node')
+  return ''
 
 def RunNode(cmd_parts, stdout=None):
   cmd = [GetBinaryPath()] + cmd_parts
