---
 base/process_util_unittest.cc |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: chromium-browser/base/process_util_unittest.cc
===================================================================
--- chromium-browser.orig/base/process_util_unittest.cc
+++ chromium-browser/base/process_util_unittest.cc
@@ -176,26 +176,29 @@
   // InitLogging always opens a file at startup.
   int expected_num_open_fds = 1;
 #if defined(OS_LINUX)
   // On Linux, '/etc/localtime' is opened before the test's main() enters.
   expected_num_open_fds += 1;
 #endif  // defined(OS_LINUX)
   num_open_files -= expected_num_open_fds;
 
-  write(write_pipe, &num_open_files, sizeof(num_open_files));
+  ssize_t size = write(write_pipe, &num_open_files, sizeof(num_open_files));
+  if (size != sizeof(num_open_files))
+    return -1; // what are we supposed to return here??
   close(write_pipe);
 
   return 0;
 }
 
 TEST_F(ProcessUtilTest, FDRemapping) {
   // Open some files to check they don't get leaked to the child process.
   int fds[2];
-  pipe(fds);
+  int ret = pipe(fds);
+  ASSERT_EQ(ret, 0);
   int pipe_read_fd = fds[0];
   int pipe_write_fd = fds[1];
 
   // open some dummy fds to make sure they don't propogate over to the
   // child process.
   int dev_null = open("/dev/null", O_RDONLY);
   int sockets[2];
   socketpair(AF_UNIX, SOCK_STREAM, 0, sockets);
