Index: sid/src/chrome/browser/search_engines/template_url_prepopulate_data.h
===================================================================
--- sid.orig/src/chrome/browser/search_engines/template_url_prepopulate_data.h	2012-09-26 07:51:08.000000000 +0200
+++ sid/src/chrome/browser/search_engines/template_url_prepopulate_data.h	2012-09-26 11:12:44.010698853 +0200
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <string>
 #include <vector>
+#include <cstddef>
 
 #include "base/string16.h"
 #include "chrome/browser/search_engines/search_engine_type.h"
Index: sid/src/gpu/command_buffer/common/types.h
===================================================================
--- sid.orig/src/gpu/command_buffer/common/types.h	2012-09-26 07:50:10.000000000 +0200
+++ sid/src/gpu/command_buffer/common/types.h	2012-09-26 11:12:44.014698853 +0200
@@ -12,6 +12,7 @@
 #endif
 #include <cstddef>
 #include <string>
+#include <cstddef>
 
 typedef signed char         schar;
 typedef signed char         int8;
Index: sid/src/ipc/ipc_channel.h
===================================================================
--- sid.orig/src/ipc/ipc_channel.h	2012-09-26 07:50:43.000000000 +0200
+++ sid/src/ipc/ipc_channel.h	2012-09-26 11:12:44.014698853 +0200
@@ -1,3 +1,4 @@
+#include <sys/types.h>
 // Copyright (c) 2012 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
Index: sid/src/third_party/ots/src/os2.cc
===================================================================
--- sid.orig/src/third_party/ots/src/os2.cc	2012-09-26 07:52:42.000000000 +0200
+++ sid/src/third_party/ots/src/os2.cc	2012-09-26 11:12:44.014698853 +0200
@@ -5,6 +5,7 @@
 #include "os2.h"
 
 #include "head.h"
+#include <cstddef>
 
 // OS/2 - OS/2 and Windows Metrics
 // http://www.microsoft.com/opentype/otspec/os2.htm
Index: sid/src/third_party/tcmalloc/chromium/src/base/stl_allocator.h
===================================================================
--- sid.orig/src/third_party/tcmalloc/chromium/src/base/stl_allocator.h	2012-09-26 07:50:24.000000000 +0200
+++ sid/src/third_party/tcmalloc/chromium/src/base/stl_allocator.h	2012-09-26 11:12:44.014698853 +0200
@@ -39,6 +39,7 @@
 
 #include <stddef.h>   // for std::ptrdiff_t
 #include <limits>
+#include <cstddef>
 
 #include "base/logging.h"
 
Index: sid/src/third_party/tcmalloc/chromium/src/base/vdso_support.h
===================================================================
--- sid.orig/src/third_party/tcmalloc/chromium/src/base/vdso_support.h	2012-09-26 07:50:24.000000000 +0200
+++ sid/src/third_party/tcmalloc/chromium/src/base/vdso_support.h	2012-09-26 11:12:44.014698853 +0200
@@ -63,6 +63,7 @@
 
 #define HAVE_VDSO_SUPPORT 1
 
+#include <cstddef>
 #include <stdlib.h>     // for NULL
 
 namespace base {
Index: sid/src/third_party/tcmalloc/chromium/src/common.cc
===================================================================
--- sid.orig/src/third_party/tcmalloc/chromium/src/common.cc	2012-09-26 07:50:24.000000000 +0200
+++ sid/src/third_party/tcmalloc/chromium/src/common.cc	2012-09-26 11:12:44.014698853 +0200
@@ -33,6 +33,7 @@
 #include "config.h"
 #include "common.h"
 #include "system-alloc.h"
+#include <cstddef>
 
 #if defined(HAVE_UNISTD_H) && defined(HAVE_GETPAGESIZE)
 #include <unistd.h>                     // for getpagesize
Index: sid/src/third_party/tcmalloc/chromium/src/symbolize.h
===================================================================
--- sid.orig/src/third_party/tcmalloc/chromium/src/symbolize.h	2012-09-26 07:50:24.000000000 +0200
+++ sid/src/third_party/tcmalloc/chromium/src/symbolize.h	2012-09-26 11:12:44.014698853 +0200
@@ -40,6 +40,7 @@
 #endif
 #include <stddef.h>  // for NULL
 #include <map>
+#include <cstddef>
 
 using std::map;
 
Index: sid/src/ui/gfx/codec/jpeg_codec.h
===================================================================
--- sid.orig/src/ui/gfx/codec/jpeg_codec.h	2012-09-26 07:51:52.000000000 +0200
+++ sid/src/ui/gfx/codec/jpeg_codec.h	2012-09-26 11:12:44.014698853 +0200
@@ -7,6 +7,7 @@
 
 #include <stddef.h>
 #include <vector>
+#include <cstddef>
 
 #include "ui/base/ui_export.h"
 
