Index: chromium/src/build/common.gypi
===================================================================
--- chromium.orig/src/build/common.gypi	2013-07-12 15:18:31.970094698 +0000
+++ chromium/src/build/common.gypi	2013-07-12 15:18:31.966094698 +0000
@@ -2832,6 +2832,14 @@
           },
         },
         'conditions': [
+          [ 'gcc_version>=48', {
+            'cflags': [
+              '-Wno-unused-local-typedefs',
+            ],
+            'cflags_cc': [
+              '-fpermissive',
+            ],
+          }],
           ['target_arch=="ia32"', {
             'target_conditions': [
               ['_toolset=="target"', {
