Remove gecko-mediaplayer from blacklist
--- a/src/webkit/plugins/npapi/plugin_list_posix.cc
+++ b/src/webkit/plugins/npapi/plugin_list_posix.cc
@@ -92,7 +92,7 @@ bool IsUndesirablePlugin(const WebPlugin
 bool IsBlacklistedPlugin(const FilePath& path) {
   const char* kBlackListedPlugins[] = {
     "nppdf.so",           // Adobe PDF
-    "gecko-mediaplayer",  // Gecko Media Player
+    //"gecko-mediaplayer",  // Gecko Media Player
   };
   std::string filename = path.BaseName().value();
   for (size_t i = 0; i < arraysize(kBlackListedPlugins); i++) {
