--- a/src/build/linux/system.gyp
+++ b/src/build/linux/system.gyp
@@ -341,6 +341,23 @@
       ],
     },
     {
+      'target_name': 'glew',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other glewmx)',
+            ],
+            'libraries': [
+              '<!@(<(pkg-config) --libs-only-l glewmx)',
+              '-lGL',
+              '-lX11',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'dbus-glib',
       'type': 'settings',
       'direct_dependent_settings': {
--- a/src/chrome/chrome.gyp
+++ b/src/chrome/chrome.gyp
@@ -664,6 +664,7 @@
         'common',
         '../media/media.gyp:media',
         '../skia/skia.gyp:skia',
+        '../build/linux/system.gyp:glew',
       ],
       'sources': [
         'gpu/gpu_channel.cc',
--- a/src/build/common.gypi
+++ b/src/build/common.gypi
@@ -576,6 +576,7 @@
       ['enable_gpu==1', {
         'defines': [
           'ENABLE_GPU=1',
+          'GLEW_MX',
         ],
       }],
       ['use_openssl==1', {
