--- a/src/build/linux/system.gyp
+++ b/src/build/linux/system.gyp
@@ -399,6 +399,23 @@
       ],
     },
     {
+      'target_name': 'glew',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other glewmx)',
+            ],
+            'libraries': [
+              '<!@(<(pkg-config) --libs-only-l glewmx)',
+              '-lGL',
+              '-lX11',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'dbus-glib',
       'type': 'settings',
       'direct_dependent_settings': {
--- a/src/chrome/chrome.gyp
+++ b/src/chrome/chrome.gyp
@@ -482,6 +482,7 @@
       'dependencies': [
         '../base/base.gyp:base',
         '../skia/skia.gyp:skia',
+        '../build/linux/system.gyp:glew',
       ],
       'sources': [
         'utility/utility_main.cc',
--- a/src/build/common.gypi
+++ b/src/build/common.gypi
@@ -720,6 +720,7 @@
       ['enable_gpu==1', {
         'defines': [
           'ENABLE_GPU=1',
+          'GLEW_MX',
         ],
       }],
       ['use_openssl==1', {
