description: use system glew library
Index: chromium/src/build/linux/system.gyp
===================================================================
--- chromium.orig/src/build/linux/system.gyp	2012-02-10 23:37:18.000000000 -0500
+++ chromium/src/build/linux/system.gyp	2012-02-10 23:37:19.000000000 -0500
@@ -493,6 +493,23 @@
       },
     },
     {
+      'target_name': 'glew',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other glewmx)',
+            ],
+            'libraries': [
+              '<!@(<(pkg-config) --libs-only-l glewmx)',
+              '-lGL',
+              '-lX11',
+            ],
+          },
+      }]]
+    },
+    {
       # TODO(satorux): Remove this once dbus-glib clients are gone.
       'target_name': 'dbus-glib',
       'type': 'none',
Index: chromium/src/chrome/chrome.gyp
===================================================================
--- chromium.orig/src/chrome/chrome.gyp	2012-02-10 23:36:45.000000000 -0500
+++ chromium/src/chrome/chrome.gyp	2012-02-10 23:37:19.000000000 -0500
@@ -225,6 +225,7 @@
         '../base/base.gyp:base',
         '../content/content.gyp:content_utility',
         '../skia/skia.gyp:skia',
+        '../build/linux/system.gyp:glew',
       ],
       'sources': [
         'utility/chrome_content_utility_client.cc',
Index: chromium/src/build/common.gypi
===================================================================
--- chromium.orig/src/build/common.gypi	2012-02-10 23:37:08.000000000 -0500
+++ chromium/src/build/common.gypi	2012-02-10 23:37:19.000000000 -0500
@@ -1093,6 +1093,7 @@
       ['enable_gpu==1', {
         'defines': [
           'ENABLE_GPU=1',
+          'GLEW_MX',
         ],
       }],
       ['use_openssl==1', {
