description: use system glew library
Index: exp/src/build/linux/system.gyp
===================================================================
--- exp.orig/src/build/linux/system.gyp	2012-05-28 18:34:27.218750792 +0200
+++ exp/src/build/linux/system.gyp	2012-05-28 18:35:41.310751006 +0200
@@ -497,6 +497,23 @@
       },
     },
     {
+      'target_name': 'glew',
+      'type': 'none',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other glewmx)',
+            ],
+            'libraries': [
+              '<!@(<(pkg-config) --libs-only-l glewmx)',
+              '-lGL',
+              '-lX11',
+            ],
+          },
+      }]]
+    },
+    {
       # TODO(satorux): Remove this once dbus-glib clients are gone.
       'target_name': 'dbus-glib',
       'type': 'none',
Index: exp/src/chrome/chrome.gyp
===================================================================
--- exp.orig/src/chrome/chrome.gyp	2012-05-28 11:16:49.078675730 +0200
+++ exp/src/chrome/chrome.gyp	2012-05-28 18:34:27.238750792 +0200
@@ -247,6 +247,7 @@
         '../base/base.gyp:base',
         '../content/content.gyp:content_utility',
         '../skia/skia.gyp:skia',
+        '../build/linux/system.gyp:glew',
       ],
       'sources': [
         'utility/chrome_content_utility_client.cc',
Index: exp/src/build/common.gypi
===================================================================
--- exp.orig/src/build/common.gypi	2012-05-28 18:34:27.190750790 +0200
+++ exp/src/build/common.gypi	2012-05-28 18:34:27.238750792 +0200
@@ -1373,6 +1373,7 @@
       ['enable_gpu==1', {
         'defines': [
           'ENABLE_GPU=1',
+          'GLEW_MX',
         ],
       }],
       ['use_openssl==1', {
