description: use system glew library
Index: chromium-browser/src/build/linux/system.gyp
===================================================================
--- chromium-browser.orig/src/build/linux/system.gyp	2012-03-29 13:57:24.643692557 -0400
+++ chromium-browser/src/build/linux/system.gyp	2012-03-29 13:57:30.367620612 -0400
@@ -493,6 +493,23 @@
       },
     },
     {
+      'target_name': 'glew',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other glewmx)',
+            ],
+            'libraries': [
+              '<!@(<(pkg-config) --libs-only-l glewmx)',
+              '-lGL',
+              '-lX11',
+            ],
+          },
+      }]]
+    },
+    {
       # TODO(satorux): Remove this once dbus-glib clients are gone.
       'target_name': 'dbus-glib',
       'type': 'none',
Index: chromium-browser/src/chrome/chrome.gyp
===================================================================
--- chromium-browser.orig/src/chrome/chrome.gyp	2012-03-28 21:07:26.745959676 -0400
+++ chromium-browser/src/chrome/chrome.gyp	2012-03-29 13:57:30.399620199 -0400
@@ -237,6 +237,7 @@
         '../base/base.gyp:base',
         '../content/content.gyp:content_utility',
         '../skia/skia.gyp:skia',
+        '../build/linux/system.gyp:glew',
       ],
       'sources': [
         'utility/chrome_content_utility_client.cc',
Index: chromium-browser/src/build/common.gypi
===================================================================
--- chromium-browser.orig/src/build/common.gypi	2012-03-29 13:56:28.044404104 -0400
+++ chromium-browser/src/build/common.gypi	2012-03-29 13:57:30.399620199 -0400
@@ -1141,6 +1141,7 @@
       ['enable_gpu==1', {
         'defines': [
           'ENABLE_GPU=1',
+          'GLEW_MX',
         ],
       }],
       ['use_openssl==1', {
