description: include glib.h directly (closes: #666640)
author: Michael Gilbert <michael.s.gilbert@gmail.com>
--- chromium-browser-18.0.1025.142~r129054.orig/src/ui/base/l10n/l10n_util.cc
+++ chromium-browser-18.0.1025.142~r129054/src/ui/base/l10n/l10n_util.cc
@@ -5,7 +5,7 @@
 #include "ui/base/l10n/l10n_util.h"
 
 #if defined(USE_X11)
-#include <glib/gutils.h>
+#include <glib.h>
 #endif
 
 #include <algorithm>
