---
 net/base/mime_util.cc |    4 ----
 1 file changed, 4 deletions(-)

Index: src/net/base/mime_util.cc
===================================================================
--- src.orig/net/base/mime_util.cc
+++ src/net/base/mime_util.cc
@@ -183,39 +183,35 @@
 // A list of media types: http://en.wikipedia.org/wiki/Internet_media_type
 // A comprehensive mime type list: http://plugindoc.mozdev.org/winmime.php
 static const char* const supported_media_types[] = {
   // Ogg.
   "video/ogg",
   "audio/ogg",
   "application/ogg",
 
-#if defined(GOOGLE_CHROME_BUILD)
   // MPEG-4.
   "video/mp4",
   "video/x-m4v",
   "audio/mp4",
   "audio/x-m4a",
 
   // MP3.
   "audio/mp3",
   "audio/x-mp3",
   "audio/mpeg",
-#endif
 };
 
 // List of supported codecs when passed in with <source type="...">.
 //
 // Refer to http://wiki.whatwg.org/wiki/Video_type_parameters#Browser_Support
 // for more information.
 static const char* const supported_media_codecs[] = {
-#if defined(GOOGLE_CHROME_BUILD)
   "avc1",
   "mp4a",
-#endif
   "theora",
   "vorbis",
 };
 
 // Note: does not include javascript types list (see supported_javascript_types)
 static const char* const supported_non_image_types[] = {
   "text/html",
   "text/xml",
