---
 net/base/mime_util.cc |    4 ----
 1 file changed, 4 deletions(-)

--- a/src/net/base/mime_util.cc
+++ b/src/net/base/mime_util.cc
@@ -189,7 +189,6 @@ static const char* const supported_media
   "audio/ogg",
   "application/ogg",
 
-#if defined(GOOGLE_CHROME_BUILD)
   // MPEG-4.
   "video/mp4",
   "video/x-m4v",
@@ -200,7 +199,6 @@ static const char* const supported_media
   "audio/mp3",
   "audio/x-mp3",
   "audio/mpeg",
-#endif
 };
 
 // List of supported codecs when passed in with <source type="...">.
@@ -208,10 +206,8 @@ static const char* const supported_media
 // Refer to http://wiki.whatwg.org/wiki/Video_type_parameters#Browser_Support
 // for more information.
 static const char* const supported_media_codecs[] = {
-#if defined(GOOGLE_CHROME_BUILD)
   "avc1",
   "mp4a",
-#endif
   "theora",
   "vorbis",
 };
