Index: chromium/src/chrome/browser/ui/gtk/gtk_util.cc
===================================================================
--- chromium.orig/src/chrome/browser/ui/gtk/gtk_util.cc	2013-05-22 03:15:59.425474873 +0000
+++ chromium/src/chrome/browser/ui/gtk/gtk_util.cc	2013-05-22 03:15:59.421474874 +0000
@@ -54,7 +54,7 @@
 #if defined(GOOGLE_CHROME_BUILD)
 static const char* kIconName = "google-chrome";
 #else
-static const char* kIconName = "chromium-browser";
+static const char* kIconName = "chromium";
 #endif
 
 const char kBoldLabelMarkup[] = "<span weight='bold'>%s</span>";
