--- a/src/chrome/browser/chromeos/webui/system_settings_provider.h
+++ b/src/chrome/browser/chromeos/webui/system_settings_provider.h
@@ -10,7 +10,7 @@
 #include "base/string16.h"
 #include "chrome/browser/chromeos/cros/system_library.h"
 #include "chrome/browser/chromeos/cros_settings_provider.h"
-#include "third_party/icu/public/i18n/unicode/timezone.h"
+#include <unicode/timezone.h>
 
 class Value;
 class ListValue;
