Index: sid/ash/ash.gyp
===================================================================
--- sid.orig/ash/ash.gyp
+++ sid/ash/ash.gyp
@@ -40,8 +40,6 @@
         '../ui/web_dialogs/web_dialogs.gyp:web_dialogs',
         '../ui/wm/wm.gyp:wm',
         '../url/url.gyp:url_lib',
-        'ash_resources.gyp:ash_resources',
-        'ash_strings.gyp:ash_strings',
       ],
       'defines': [
         'ASH_IMPLEMENTATION',
@@ -766,9 +764,6 @@
         '../ui/views/views.gyp:views',
         '../ui/web_dialogs/web_dialogs.gyp:web_dialogs',
         '../url/url.gyp:url_lib',
-        'ash',
-        'ash_resources.gyp:ash_resources',
-        'ash_strings.gyp:ash_strings',
       ],
       'defines': [
         'ASH_WITH_CONTENT_IMPLEMENTATION',
@@ -794,8 +789,6 @@
         '../ui/accessibility/accessibility.gyp:ax_gen',
         '../ui/app_list/app_list.gyp:app_list_test_support',
         '../ui/views/views.gyp:views_test_support',
-        'ash',
-        'ash_resources.gyp:ash_resources',
       ],
       'sources': [
         'desktop_background/desktop_background_controller_test_api.cc',
@@ -911,11 +904,6 @@
         '../ui/web_dialogs/web_dialogs.gyp:web_dialogs_test_support',
         '../ui/wm/wm.gyp:wm',
         '../url/url.gyp:url_lib',
-        'ash',
-        'ash_resources.gyp:ash_resources',
-        'ash_strings.gyp:ash_strings',
-        'ash_test_support',
-        'ash_with_content',
       ],
       'sources': [
         'accelerators/accelerator_commands_unittest.cc',
@@ -1133,11 +1121,6 @@
         '../ui/views/examples/examples.gyp:views_examples_with_content_lib',
         '../ui/views/views.gyp:views',
         '../ui/views/views.gyp:views_test_support',
-        'ash',
-        'ash_resources.gyp:ash_resources',
-        'ash_strings.gyp:ash_strings',
-        'ash_test_support',
-        'ash_with_content',
       ],
       'sources': [
         '../content/app/startup_helper_win.cc',
Index: sid/chrome/chrome_browser_ui.gypi
===================================================================
--- sid.orig/chrome/chrome_browser_ui.gypi
+++ sid/chrome/chrome_browser_ui.gypi
@@ -2654,9 +2654,6 @@
             'browser/ui/views/touch_uma/touch_uma_ash.cc',
           ],
           'dependencies': [
-            '../ash/ash.gyp:ash',
-            '../ash/ash.gyp:ash_with_content',
-            '../ash/ash_strings.gyp:ash_strings',
           ],
         }, { # use_ash==0
           'sources': [
@@ -2704,7 +2701,6 @@
           ],
           'dependencies': [
             # aura uses some of ash resources.
-            '../ash/ash_resources.gyp:ash_resources',
             '../ui/aura/aura.gyp:aura',
             '../ui/keyboard/keyboard.gyp:keyboard',
             '../ui/keyboard/keyboard.gyp:keyboard_resources',
Index: sid/chrome/browser/ui/host_desktop.cc
===================================================================
--- sid.orig/chrome/browser/ui/host_desktop.cc
+++ sid/chrome/browser/ui/host_desktop.cc
@@ -8,8 +8,11 @@
 #include <windows.h>
 #endif
 
+#if defined(USE_ASH)
 #include "ash/shell.h"
 #include "chrome/browser/ui/ash/ash_util.h"
+#endif
+
 #include "chrome/browser/ui/aura/active_desktop_monitor.h"
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_list.h"
Index: sid/chrome/browser/ui/views/frame/browser_frame.cc
===================================================================
--- sid.orig/chrome/browser/ui/views/frame/browser_frame.cc
+++ sid/chrome/browser/ui/views/frame/browser_frame.cc
@@ -4,7 +4,6 @@
 
 #include "chrome/browser/ui/views/frame/browser_frame.h"
 
-#include "ash/shell.h"
 #include "base/command_line.h"
 #include "base/debug/leak_annotations.h"
 #include "base/i18n/rtl.h"
Index: sid/chrome/browser/ui/views/screen_capture_notification_ui_views.cc
===================================================================
--- sid.orig/chrome/browser/ui/views/screen_capture_notification_ui_views.cc
+++ sid/chrome/browser/ui/views/screen_capture_notification_ui_views.cc
@@ -4,7 +4,10 @@
 
 #include "chrome/browser/ui/screen_capture_notification_ui.h"
 
+#if defined(USE_ASH)
 #include "ash/shell.h"
+#endif
+
 #include "chrome/app/chrome_dll_resource.h"
 #include "chrome/browser/ui/views/chrome_views_export.h"
 #include "grit/generated_resources.h"
Index: sid/chrome/browser/lifetime/application_lifetime.cc
===================================================================
--- sid.orig/chrome/browser/lifetime/application_lifetime.cc
+++ sid/chrome/browser/lifetime/application_lifetime.cc
@@ -4,7 +4,10 @@
 
 #include "chrome/browser/lifetime/application_lifetime.h"
 
+#if defined(USE_ASH)
 #include "ash/shell.h"
+#endif
+
 #include "base/bind.h"
 #include "base/command_line.h"
 #include "base/debug/trace_event.h"
Index: sid/chrome/browser/profiles/avatar_menu.cc
===================================================================
--- sid.orig/chrome/browser/profiles/avatar_menu.cc
+++ sid/chrome/browser/profiles/avatar_menu.cc
@@ -4,7 +4,11 @@
 
 #include "chrome/browser/profiles/avatar_menu.h"
 
+#if defined(USE_ASH)
 #include "ash/ash_switches.h"
+#include "chrome/browser/ui/ash/chrome_shell_delegate.h"
+#endif
+
 #include "base/bind.h"
 #include "base/i18n/case_conversion.h"
 #include "base/metrics/field_trial.h"
@@ -18,7 +22,6 @@
 #include "chrome/browser/profiles/profile_metrics.h"
 #include "chrome/browser/profiles/profile_window.h"
 #include "chrome/browser/profiles/profiles_state.h"
-#include "chrome/browser/ui/ash/chrome_shell_delegate.h"
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/browser_dialogs.h"
 #include "chrome/browser/ui/host_desktop.h"
Index: sid/chrome/browser/ui/views/chrome_views_delegate_aura.cc
===================================================================
--- sid.orig/chrome/browser/ui/views/chrome_views_delegate_aura.cc
+++ sid/chrome/browser/ui/views/chrome_views_delegate_aura.cc
@@ -4,7 +4,9 @@
 
 #include "chrome/browser/ui/views/chrome_views_delegate.h"
 
+#if defined(USE_ASH)
 #include "chrome/browser/ui/ash/ash_util.h"
+#endif
 
 views::Widget::InitParams::WindowOpacity
 ChromeViewsDelegate::GetOpacityForInitParams(
Index: sid/chrome/browser/sync/sync_global_error_factory.cc
===================================================================
--- sid.orig/chrome/browser/sync/sync_global_error_factory.cc
+++ sid/chrome/browser/sync/sync_global_error_factory.cc
@@ -4,7 +4,6 @@
 
 #include "chrome/browser/sync/sync_global_error_factory.h"
 
-#include "ash/shell.h"
 #include "chrome/browser/browser_process.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/sync/profile_sync_service.h"
