Index: chromium/breakpad/breakpad.gyp
===================================================================
--- chromium.orig/breakpad/breakpad.gyp
+++ chromium/breakpad/breakpad.gyp
@@ -789,7 +789,6 @@
               'breakpad/breakpad.gyp',
             ],
           },
-          'includes': ['../build/ios/mac_build.gypi'],
           'actions': [
             {
               'action_name': 'compile breakpad utilities',
Index: chromium/third_party/protobuf/protobuf.gyp
===================================================================
--- chromium.orig/third_party/protobuf/protobuf.gyp
+++ chromium/third_party/protobuf/protobuf.gyp
@@ -53,7 +53,6 @@
               # normal protoc target under the condition that "OS==iOS".
               'target_name': 'compile_protoc',
               'type': 'none',
-              'includes': ['../../build/ios/mac_build.gypi'],
               'actions': [
                 {
                   'action_name': 'compile protoc',
Index: chromium/chrome/common/chrome_paths.cc
===================================================================
--- chromium.orig/chrome/common/chrome_paths.cc
+++ chromium/chrome/common/chrome_paths.cc
@@ -7,7 +7,6 @@
 #include "base/file_util.h"
 #include "base/lazy_instance.h"
 #include "base/logging.h"
-#include "base/mac/bundle_locations.h"
 #include "base/path_service.h"
 #include "base/strings/string_util.h"
 #include "base/sys_info.h"
@@ -23,6 +22,7 @@
 #endif
 
 #if defined(OS_MACOSX)
+#include "base/mac/bundle_locations.h"
 #include "base/mac/foundation_util.h"
 #endif
 
Index: chromium/content/common/content_paths.cc
===================================================================
--- chromium.orig/content/common/content_paths.cc
+++ chromium/content/common/content_paths.cc
@@ -5,9 +5,12 @@
 #include "content/public/common/content_paths.h"
 
 #include "base/file_util.h"
-#include "base/mac/bundle_locations.h"
 #include "base/path_service.h"
 
+#if defined(OS_MACOSX)
+#include "base/mac/bundle_locations.h"
+#endif
+
 namespace content {
 
 bool PathProvider(int key, base::FilePath* result) {
Index: chromium/sandbox/sandbox.gyp
===================================================================
--- chromium.orig/sandbox/sandbox.gyp
+++ chromium/sandbox/sandbox.gyp
@@ -17,11 +17,6 @@
         'linux/sandbox_linux.gypi',
       ],
     }],
-    [ 'OS=="mac" and OS!="ios"', {
-      'includes': [
-        'mac/sandbox_mac.gypi',
-      ],
-    }],
     [ 'OS!="win" and OS!="mac" and OS!="linux" and OS!="android"', {
       # A 'default' to accomodate the "sandbox" target.
       'targets': [
Index: chromium/chrome/browser/resources/bookmark_manager/css/bmm.css
===================================================================
--- chromium.orig/chrome/browser/resources/bookmark_manager/css/bmm.css
+++ chromium/chrome/browser/resources/bookmark_manager/css/bmm.css
@@ -187,17 +187,6 @@ html[dir=rtl] list > .folder > .label {
       url('../../../../../ui/resources/default_200_percent/common/folder_closed_rtl.png') 2x);
 }
 
-<if expr="is_macosx">
-list > .folder > .label,
-.tree-label,
-.tree-row[may-have-children] > .tree-label,
-.tree-item[expanded] > .tree-row > .tree-label {
-  background-image: -webkit-image-set(
-      url('../../../../app/theme/default_100_percent/mac/bookmark_bar_folder.png') 1x,
-      url('../../../../app/theme/default_200_percent/mac/bookmark_bar_folder.png') 2x);
-}
-</if>
-
 .main {
   -webkit-flex: 1;
   display: -webkit-flex;
Index: chromium/content/browser/renderer_host/render_process_host_impl.cc
===================================================================
--- chromium.orig/content/browser/renderer_host/render_process_host_impl.cc
+++ chromium/content/browser/renderer_host/render_process_host_impl.cc
@@ -91,7 +91,6 @@
 #include "content/browser/renderer_host/render_widget_helper.h"
 #include "content/browser/renderer_host/render_widget_host_impl.h"
 #include "content/browser/renderer_host/socket_stream_dispatcher_host.h"
-#include "content/browser/renderer_host/text_input_client_message_filter.h"
 #include "content/browser/renderer_host/websocket_dispatcher_host.h"
 #include "content/browser/resolve_proxy_msg_helper.h"
 #include "content/browser/screen_orientation/screen_orientation_dispatcher_host.h"
@@ -153,6 +152,10 @@
 #include "content/browser/media/android/browser_demuxer_android.h"
 #endif
 
+#if defined(OS_MACOSX)
+#include "content/browser/renderer_host/text_input_client_message_filter.h"
+#endif
+
 #if defined(OS_WIN)
 #include "base/win/scoped_com_initializer.h"
 #include "content/common/font_cache_dispatcher_win.h"
Index: chromium/content/content_browser.gypi
===================================================================
--- chromium.orig/content/content_browser.gypi
+++ chromium/content/content_browser.gypi
@@ -1122,8 +1122,6 @@
     'browser/renderer_host/socket_stream_host.h',
     'browser/renderer_host/text_input_client_mac.h',
     'browser/renderer_host/text_input_client_mac.mm',
-    'browser/renderer_host/text_input_client_message_filter.h',
-    'browser/renderer_host/text_input_client_message_filter.mm',
     'browser/renderer_host/ui_events_helper.cc',
     'browser/renderer_host/ui_events_helper.h',
     'browser/renderer_host/web_input_event_aura.cc',
Index: chromium/content/renderer/text_input_client_observer.cc
===================================================================
--- chromium.orig/content/renderer/text_input_client_observer.cc
+++ chromium/content/renderer/text_input_client_observer.cc
@@ -13,9 +13,12 @@
 #include "third_party/WebKit/public/platform/WebString.h"
 #include "third_party/WebKit/public/web/WebLocalFrame.h"
 #include "third_party/WebKit/public/web/WebView.h"
-#include "third_party/WebKit/public/web/mac/WebSubstringUtil.h"
 #include "ui/gfx/rect.h"
 
+#if defined(OS_MACOSX)
+#include "third_party/WebKit/public/web/mac/WebSubstringUtil.h"
+#endif
+
 namespace content {
 
 TextInputClientObserver::TextInputClientObserver(RenderViewImpl* render_view)
