Index: sid/ui/app_list/views/app_list_view.cc
===================================================================
--- sid.orig/ui/app_list/views/app_list_view.cc
+++ sid/ui/app_list/views/app_list_view.cc
@@ -8,7 +8,6 @@
 
 #include "base/command_line.h"
 #include "base/strings/string_util.h"
-#include "base/win/windows_version.h"
 #include "ui/app_list/app_list_constants.h"
 #include "ui/app_list/app_list_model.h"
 #include "ui/app_list/app_list_view_delegate.h"
@@ -40,6 +39,7 @@
 #include "ui/views/bubble/bubble_window_targeter.h"
 #if defined(OS_WIN)
 #include "ui/base/win/shell.h"
+#include "base/win/windows_version.h"
 #endif
 #if !defined(OS_CHROMEOS)
 #include "ui/views/widget/desktop_aura/desktop_native_widget_aura.h"
Index: sid/content/app/content_main_runner.cc
===================================================================
--- sid.orig/content/app/content_main_runner.cc
+++ sid/content/app/content_main_runner.cc
@@ -46,7 +46,6 @@
 #include "ipc/ipc_descriptors.h"
 #include "ipc/ipc_switches.h"
 #include "media/base/media.h"
-#include "sandbox/win/src/sandbox_types.h"
 #include "ui/base/ui_base_paths.h"
 #include "ui/base/ui_base_switches.h"
 
Index: sid/chrome/browser/ui/views/create_application_shortcut_view.cc
===================================================================
--- sid.orig/chrome/browser/ui/views/create_application_shortcut_view.cc
+++ sid/chrome/browser/ui/views/create_application_shortcut_view.cc
@@ -7,11 +7,14 @@
 #include <algorithm>
 #include <cmath>
 
+#if defined(OS_WIN)
+#include "base/win/windows_version.h"
+#endif
+
 #include "base/bind.h"
 #include "base/bind_helpers.h"
 #include "base/prefs/pref_service.h"
 #include "base/strings/utf_string_conversions.h"
-#include "base/win/windows_version.h"
 #include "chrome/browser/extensions/tab_helper.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/ui/browser.h"
Index: sid/chrome/browser/component_updater/component_updater_configurator.cc
===================================================================
--- sid.orig/chrome/browser/component_updater/component_updater_configurator.cc
+++ sid/chrome/browser/component_updater/component_updater_configurator.cc
@@ -8,10 +8,13 @@
 #include <string>
 #include <vector>
 
+#if defined(OS_WIN)
+#include "base/win/windows_version.h"
+#endif
+
 #include "base/command_line.h"
 #include "base/compiler_specific.h"
 #include "base/strings/string_util.h"
-#include "base/win/windows_version.h"
 #include "build/build_config.h"
 #include "chrome/browser/component_updater/component_patcher.h"
 #include "chrome/common/chrome_switches.h"
Index: sid/chrome/browser/component_updater/component_updater_utils.cc
===================================================================
--- sid.orig/chrome/browser/component_updater/component_updater_utils.cc
+++ sid/chrome/browser/component_updater/component_updater_utils.cc
@@ -6,6 +6,10 @@
 
 #include <cmath>
 
+#if defined(OS_WIN)
+#include "base/win/windows_version.h"
+#endif
+
 #include "base/file_util.h"
 #include "base/files/file_path.h"
 #include "base/guid.h"
@@ -14,7 +18,6 @@
 #include "base/strings/string_util.h"
 #include "base/strings/stringprintf.h"
 #include "base/sys_info.h"
-#include "base/win/windows_version.h"
 #include "chrome/browser/component_updater/component_updater_service.h"
 #include "chrome/browser/component_updater/crx_update_item.h"
 #include "chrome/browser/omaha_query_params/omaha_query_params.h"
Index: sid/chrome/browser/component_updater/pnacl/pnacl_component_installer.cc
===================================================================
--- sid.orig/chrome/browser/component_updater/pnacl/pnacl_component_installer.cc
+++ sid/chrome/browser/component_updater/pnacl/pnacl_component_installer.cc
@@ -7,6 +7,10 @@
 #include <string>
 #include <vector>
 
+#if defined(OS_WIN)
+#include "base/win/windows_version.h"
+#endif
+
 #include "base/atomicops.h"
 #include "base/base_paths.h"
 #include "base/bind.h"
@@ -22,7 +26,6 @@
 #include "base/strings/string_util.h"
 #include "base/values.h"
 #include "base/version.h"
-#include "base/win/windows_version.h"
 #include "build/build_config.h"
 #include "chrome/browser/browser_process.h"
 #include "chrome/browser/component_updater/component_updater_service.h"
Index: sid/chrome/browser/ui/views/profiles/new_avatar_button.cc
===================================================================
--- sid.orig/chrome/browser/ui/views/profiles/new_avatar_button.cc
+++ sid/chrome/browser/ui/views/profiles/new_avatar_button.cc
@@ -5,7 +5,6 @@
 #include "chrome/browser/ui/views/profiles/new_avatar_button.h"
 
 #include "base/strings/utf_string_conversions.h"
-#include "base/win/windows_version.h"
 #include "chrome/browser/browser_process.h"
 #include "chrome/browser/profiles/profile_manager.h"
 #include "chrome/browser/profiles/profiles_state.h"
Index: sid/chrome/browser/ui/views/screen_capture_notification_ui_views.cc
===================================================================
--- sid.orig/chrome/browser/ui/views/screen_capture_notification_ui_views.cc
+++ sid/chrome/browser/ui/views/screen_capture_notification_ui_views.cc
@@ -193,6 +193,7 @@ gfx::NativeViewId ScreenCaptureNotificat
   set_background(views::Background::CreateSolidBackground(GetNativeTheme()->
       GetSystemColor(ui::NativeTheme::kColorId_DialogBackground)));
 
+#if defined(OS_WIN)
   gfx::Screen* screen = gfx::Screen::GetNativeScreen();
   // TODO(sergeyu): Move the notification to the display being captured when
   // per-display screen capture is supported.
@@ -209,7 +210,6 @@ gfx::NativeViewId ScreenCaptureNotificat
   // This has to be called after Show() to have effect.
   widget->SetOpacity(0xFF * kWindowAlphaValue);
 
-#if defined(OS_WIN)
   return gfx::NativeViewId(views::HWNDForWidget(widget));
 #else
   return 0;
Index: sid/chrome/browser/omaha_query_params/omaha_query_params.cc
===================================================================
--- sid.orig/chrome/browser/omaha_query_params/omaha_query_params.cc
+++ sid/chrome/browser/omaha_query_params/omaha_query_params.cc
@@ -4,9 +4,12 @@
 
 #include "chrome/browser/omaha_query_params/omaha_query_params.h"
 
+#if defined(OS_WIN)
+#include "base/win/windows_version.h"
+#endif
+
 #include "base/compiler_specific.h"
 #include "base/strings/stringprintf.h"
-#include "base/win/windows_version.h"
 #include "chrome/browser/browser_process.h"
 #include "chrome/common/chrome_version_info.h"
 
