Include system copy of prtime.h
Index: sid/src/base/pr_time_unittest.cc
===================================================================
--- sid.orig/src/base/pr_time_unittest.cc	2013-01-11 11:03:38.000000000 +0100
+++ sid/src/base/pr_time_unittest.cc	2013-01-11 12:57:06.590733818 +0100
@@ -5,7 +5,7 @@
 #include <time.h>
 
 #include "base/compiler_specific.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
Index: sid/src/base/time.cc
===================================================================
--- sid.orig/src/base/time.cc	2013-01-11 11:03:38.000000000 +0100
+++ sid/src/base/time.cc	2013-01-11 12:57:06.590733818 +0100
@@ -12,7 +12,7 @@
 #include <limits>
 
 #include "base/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 
 #include "base/logging.h"
 
Index: sid/src/chrome/browser/metrics/metrics_log.cc
===================================================================
--- sid.orig/src/chrome/browser/metrics/metrics_log.cc	2013-01-11 11:03:57.000000000 +0100
+++ sid/src/chrome/browser/metrics/metrics_log.cc	2013-01-11 12:57:06.590733818 +0100
@@ -16,7 +16,7 @@
 #include "base/string_number_conversions.h"
 #include "base/string_util.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "base/tracked_objects.h"
 #include "base/utf_string_conversions.h"
Index: sid/src/base/base.gypi
===================================================================
--- sid.orig/src/base/base.gypi	2013-01-11 11:03:38.000000000 +0100
+++ sid/src/base/base.gypi	2013-01-11 12:57:06.594733817 +0100
@@ -17,7 +17,6 @@
           'third_party/dmg_fp/dtoa_wrapper.cc',
           'third_party/icu/icu_utf.cc',
           'third_party/icu/icu_utf.h',
-          'third_party/nspr/prtime.cc',
           'third_party/nspr/prtime.h',
           'third_party/nspr/prcpucfg_linux.h',
           'third_party/xdg_mime/xdgmime.h',
Index: sid/src/crypto/nss_util.cc
===================================================================
--- sid.orig/src/crypto/nss_util.cc	2013-01-11 11:03:17.000000000 +0100
+++ sid/src/crypto/nss_util.cc	2013-01-11 12:57:06.594733817 +0100
@@ -9,7 +9,7 @@
 #include <plarena.h>
 #include <prerror.h>
 #include <prinit.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <pk11pub.h>
 #include <secmod.h>
 
Index: sid/src/net/base/x509_certificate_nss.cc
===================================================================
--- sid.orig/src/net/base/x509_certificate_nss.cc	2013-01-11 11:03:21.000000000 +0100
+++ sid/src/net/base/x509_certificate_nss.cc	2013-01-11 12:57:06.594733817 +0100
@@ -9,7 +9,7 @@
 #include <keyhi.h>
 #include <nss.h>
 #include <pk11pub.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <secder.h>
 #include <sechash.h>
 
