Include system copy of prtime.h
Index: sid/src/base/pr_time_unittest.cc
===================================================================
--- sid.orig/src/base/pr_time_unittest.cc	2013-03-16 09:36:05.000000000 +0100
+++ sid/src/base/pr_time_unittest.cc	2013-03-30 17:31:00.279750643 +0100
@@ -5,7 +5,7 @@
 #include <time.h>
 
 #include "base/compiler_specific.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
Index: sid/src/base/time.cc
===================================================================
--- sid.orig/src/base/time.cc	2013-03-16 09:36:05.000000000 +0100
+++ sid/src/base/time.cc	2013-03-30 17:34:24.303745766 +0100
@@ -12,7 +12,7 @@
 #include <limits>
 
 #include "base/strings/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 
 #include "base/logging.h"
 
Index: sid/src/chrome/browser/metrics/metrics_log.cc
===================================================================
--- sid.orig/src/chrome/browser/metrics/metrics_log.cc	2013-03-16 09:36:22.000000000 +0100
+++ sid/src/chrome/browser/metrics/metrics_log.cc	2013-03-30 17:31:00.279750643 +0100
@@ -19,7 +19,7 @@
 #include "base/string_util.h"
 #include "base/strings/string_number_conversions.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "base/tracked_objects.h"
 #include "base/utf_string_conversions.h"
Index: sid/src/chrome/common/metrics/metrics_log_base.cc
===================================================================
--- sid.orig/src/chrome/common/metrics/metrics_log_base.cc	2013-03-16 09:37:14.000000000 +0100
+++ sid/src/chrome/common/metrics/metrics_log_base.cc	2013-03-30 17:33:36.375746910 +0100
@@ -13,7 +13,7 @@
 #include "base/strings/string_number_conversions.h"
 #include "base/sys_byteorder.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/utf_string_conversions.h"
 #include "chrome/common/chrome_version_info.h"
 #include "chrome/common/logging_chrome.h"
