Include system copy of prtime.h
--- a/src/base/pr_time_unittest.cc
+++ b/src/base/pr_time_unittest.cc
@@ -4,7 +4,7 @@
 
 #include <time.h>
 
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
--- a/src/base/time.cc
+++ b/src/base/time.cc
@@ -5,7 +5,7 @@
 #include "base/time.h"
 #include "base/string_util.h"
 #include "base/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 
 #include "base/logging.h"
 
