Include system copy of prtime.h
Index: exp/src/base/pr_time_unittest.cc
===================================================================
--- exp.orig/src/base/pr_time_unittest.cc	2012-05-19 16:54:40.000000000 +0200
+++ exp/src/base/pr_time_unittest.cc	2012-05-19 19:18:03.918668070 +0200
@@ -5,7 +5,7 @@
 #include <time.h>
 
 #include "base/compiler_specific.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
Index: exp/src/base/time.cc
===================================================================
--- exp.orig/src/base/time.cc	2012-05-19 16:54:40.000000000 +0200
+++ exp/src/base/time.cc	2012-05-19 19:17:29.718667704 +0200
@@ -4,7 +4,7 @@
 
 #include "base/time.h"
 #include "base/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 
 #include "base/logging.h"
 
Index: exp/src/chrome/browser/metrics/metrics_log.cc
===================================================================
--- exp.orig/src/chrome/browser/metrics/metrics_log.cc	2012-05-19 16:54:55.000000000 +0200
+++ exp/src/chrome/browser/metrics/metrics_log.cc	2012-05-19 19:17:29.718667704 +0200
@@ -16,7 +16,7 @@
 #include "base/string_number_conversions.h"
 #include "base/string_util.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "base/tracked_objects.h"
 #include "base/utf_string_conversions.h"
Index: exp/src/base/base.gypi
===================================================================
--- exp.orig/src/base/base.gypi	2012-05-19 16:54:40.000000000 +0200
+++ exp/src/base/base.gypi	2012-05-19 19:17:29.718667704 +0200
@@ -17,7 +17,6 @@
           'third_party/dmg_fp/dtoa_wrapper.cc',
           'third_party/icu/icu_utf.cc',
           'third_party/icu/icu_utf.h',
-          'third_party/nspr/prtime.cc',
           'third_party/nspr/prtime.h',
           'third_party/nspr/prcpucfg_linux.h',
           'third_party/xdg_mime/xdgmime.h',
Index: exp/src/crypto/nss_util.cc
===================================================================
--- exp.orig/src/crypto/nss_util.cc	2012-05-19 16:54:16.000000000 +0200
+++ exp/src/crypto/nss_util.cc	2012-05-19 19:17:29.718667704 +0200
@@ -9,7 +9,7 @@
 #include <plarena.h>
 #include <prerror.h>
 #include <prinit.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <pk11pub.h>
 #include <secmod.h>
 
Index: exp/src/net/base/x509_certificate_nss.cc
===================================================================
--- exp.orig/src/net/base/x509_certificate_nss.cc	2012-05-19 16:54:19.000000000 +0200
+++ exp/src/net/base/x509_certificate_nss.cc	2012-05-19 19:17:29.718667704 +0200
@@ -9,7 +9,7 @@
 #include <keyhi.h>
 #include <nss.h>
 #include <pk11pub.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <secder.h>
 #include <sechash.h>
