Include system copy of prtime.h
Index: chromium-browser/src/base/pr_time_unittest.cc
===================================================================
--- chromium-browser.orig/src/base/pr_time_unittest.cc	2012-03-28 21:11:12.000000000 -0400
+++ chromium-browser/src/base/pr_time_unittest.cc	2012-03-29 12:40:18.697434130 -0400
@@ -4,7 +4,7 @@
 
 #include <time.h>
 
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
Index: chromium-browser/src/base/time.cc
===================================================================
--- chromium-browser.orig/src/base/time.cc	2012-03-28 21:11:10.000000000 -0400
+++ chromium-browser/src/base/time.cc	2012-03-29 12:40:18.697434130 -0400
@@ -4,7 +4,7 @@
 
 #include "base/time.h"
 #include "base/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 
 #include "base/logging.h"
 
Index: chromium-browser/src/chrome/browser/metrics/metrics_log.cc
===================================================================
--- chromium-browser.orig/src/chrome/browser/metrics/metrics_log.cc	2012-03-28 21:08:38.000000000 -0400
+++ chromium-browser/src/chrome/browser/metrics/metrics_log.cc	2012-03-29 12:40:18.697434130 -0400
@@ -14,7 +14,7 @@
 #include "base/perftimer.h"
 #include "base/string_util.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "base/utf_string_conversions.h"
 #include "chrome/browser/autocomplete/autocomplete.h"
Index: chromium-browser/src/base/base.gypi
===================================================================
--- chromium-browser.orig/src/base/base.gypi	2012-03-28 21:11:11.000000000 -0400
+++ chromium-browser/src/base/base.gypi	2012-03-29 12:40:18.697434130 -0400
@@ -17,7 +17,6 @@
           'third_party/dmg_fp/dtoa_wrapper.cc',
           'third_party/icu/icu_utf.cc',
           'third_party/icu/icu_utf.h',
-          'third_party/nspr/prtime.cc',
           'third_party/nspr/prtime.h',
           'third_party/nspr/prcpucfg_linux.h',
           'third_party/xdg_mime/xdgmime.h',
Index: chromium-browser/src/crypto/nss_util.cc
===================================================================
--- chromium-browser.orig/src/crypto/nss_util.cc	2012-03-28 21:07:02.000000000 -0400
+++ chromium-browser/src/crypto/nss_util.cc	2012-03-29 12:40:18.697434130 -0400
@@ -9,7 +9,7 @@
 #include <plarena.h>
 #include <prerror.h>
 #include <prinit.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <pk11pub.h>
 #include <secmod.h>
 
Index: chromium-browser/src/net/base/x509_certificate_nss.cc
===================================================================
--- chromium-browser.orig/src/net/base/x509_certificate_nss.cc	2012-03-28 21:07:16.000000000 -0400
+++ chromium-browser/src/net/base/x509_certificate_nss.cc	2012-03-29 12:40:18.697434130 -0400
@@ -10,7 +10,7 @@
 #include <nss.h>
 #include <pk11pub.h>
 #include <prerror.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <secder.h>
 #include <secerr.h>
 #include <sechash.h>
