Include system copy of prtime.h
Index: chromium/src/base/pr_time_unittest.cc
===================================================================
--- chromium.orig/src/base/pr_time_unittest.cc	2012-02-10 23:31:43.000000000 -0500
+++ chromium/src/base/pr_time_unittest.cc	2012-02-10 23:31:49.000000000 -0500
@@ -4,7 +4,7 @@
 
 #include <time.h>
 
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
Index: chromium/src/base/time.cc
===================================================================
--- chromium.orig/src/base/time.cc	2012-02-10 23:31:43.000000000 -0500
+++ chromium/src/base/time.cc	2012-02-10 23:31:49.000000000 -0500
@@ -4,7 +4,7 @@
 
 #include "base/time.h"
 #include "base/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 
 #include "base/logging.h"
 
Index: chromium/src/chrome/browser/metrics/metrics_log.cc
===================================================================
--- chromium.orig/src/chrome/browser/metrics/metrics_log.cc	2012-02-10 23:31:44.000000000 -0500
+++ chromium/src/chrome/browser/metrics/metrics_log.cc	2012-02-10 23:31:49.000000000 -0500
@@ -14,7 +14,7 @@
 #include "base/perftimer.h"
 #include "base/string_util.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "base/utf_string_conversions.h"
 #include "chrome/browser/autocomplete/autocomplete.h"
Index: chromium/src/chrome/common/metrics_helpers.cc
===================================================================
--- chromium.orig/src/chrome/common/metrics_helpers.cc	2012-02-10 23:31:44.000000000 -0500
+++ chromium/src/chrome/common/metrics_helpers.cc	2012-02-10 23:31:49.000000000 -0500
@@ -11,7 +11,7 @@
 #include "base/perftimer.h"
 #include "base/string_number_conversions.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "base/utf_string_conversions.h"
 #include "chrome/common/logging_chrome.h"
Index: chromium/src/base/base.gypi
===================================================================
--- chromium.orig/src/base/base.gypi	2012-02-10 23:31:43.000000000 -0500
+++ chromium/src/base/base.gypi	2012-02-10 23:31:49.000000000 -0500
@@ -17,7 +17,6 @@
           'third_party/dmg_fp/dtoa_wrapper.cc',
           'third_party/icu/icu_utf.cc',
           'third_party/icu/icu_utf.h',
-          'third_party/nspr/prtime.cc',
           'third_party/nspr/prtime.h',
           'third_party/nspr/prcpucfg_linux.h',
           'third_party/xdg_mime/xdgmime.h',
Index: chromium/src/crypto/nss_util.cc
===================================================================
--- chromium.orig/src/crypto/nss_util.cc	2012-02-10 23:31:44.000000000 -0500
+++ chromium/src/crypto/nss_util.cc	2012-02-10 23:31:49.000000000 -0500
@@ -9,7 +9,7 @@
 #include <plarena.h>
 #include <prerror.h>
 #include <prinit.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <pk11pub.h>
 #include <secmod.h>
 
Index: chromium/src/net/base/x509_certificate_nss.cc
===================================================================
--- chromium.orig/src/net/base/x509_certificate_nss.cc	2012-02-10 23:31:44.000000000 -0500
+++ chromium/src/net/base/x509_certificate_nss.cc	2012-02-10 23:31:49.000000000 -0500
@@ -10,7 +10,7 @@
 #include <nss.h>
 #include <pk11pub.h>
 #include <prerror.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <secder.h>
 #include <secerr.h>
 #include <sechash.h>
