Include system copy of prtime.h
Index: chromium-browser/src/base/pr_time_unittest.cc
===================================================================
--- chromium-browser.orig/src/base/pr_time_unittest.cc	2013-03-30 14:55:16.652387438 +0000
+++ chromium-browser/src/base/pr_time_unittest.cc	2013-03-30 14:55:16.640387437 +0000
@@ -5,7 +5,7 @@
 #include <time.h>
 
 #include "base/compiler_specific.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
Index: chromium-browser/src/base/time.cc
===================================================================
--- chromium-browser.orig/src/base/time.cc	2013-03-30 14:55:16.652387438 +0000
+++ chromium-browser/src/base/time.cc	2013-03-30 14:55:44.116387204 +0000
@@ -12,7 +12,7 @@
 #include <limits>
 
 #include "base/strings/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 
 #include "base/logging.h"
 
Index: chromium-browser/src/chrome/browser/metrics/metrics_log.cc
===================================================================
--- chromium-browser.orig/src/chrome/browser/metrics/metrics_log.cc	2013-03-30 14:55:16.652387438 +0000
+++ chromium-browser/src/chrome/browser/metrics/metrics_log.cc	2013-03-30 14:55:16.644387438 +0000
@@ -19,7 +19,7 @@
 #include "base/string_util.h"
 #include "base/strings/string_number_conversions.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time.h"
 #include "base/tracked_objects.h"
 #include "base/utf_string_conversions.h"
Index: chromium-browser/src/base/base.gypi
===================================================================
--- chromium-browser.orig/src/base/base.gypi	2013-03-30 14:55:16.652387438 +0000
+++ chromium-browser/src/base/base.gypi	2013-03-30 14:56:24.860387023 +0000
@@ -25,7 +25,6 @@
           'third_party/nspr/prcpucfg_openbsd.h',
           'third_party/nspr/prcpucfg_solaris.h',
           'third_party/nspr/prcpucfg_win.h',
-          'third_party/nspr/prtime.cc',
           'third_party/nspr/prtime.h',
           'third_party/nspr/prtypes.h',
           'third_party/xdg_mime/xdgmime.h',
Index: chromium-browser/src/net/base/x509_certificate_nss.cc
===================================================================
--- chromium-browser.orig/src/net/base/x509_certificate_nss.cc	2013-03-30 14:55:16.652387438 +0000
+++ chromium-browser/src/net/base/x509_certificate_nss.cc	2013-03-30 14:56:54.568387012 +0000
@@ -9,7 +9,7 @@
 #include <keyhi.h>
 #include <nss.h>
 #include <pk11pub.h>
-#include <prtime.h>
+#include <nspr/prtime.h>
 #include <seccomon.h>
 #include <secder.h>
 #include <sechash.h>
