Author:  Raymond Wooninck (tittiatcoke)
Source: OpenSUSE (chromium-master-prefs-path.patch)
Index: sid/src/chrome/browser/first_run/first_run_linux.cc
===================================================================
--- sid.orig/src/chrome/browser/first_run/first_run_linux.cc	2012-09-26 07:51:08.000000000 +0200
+++ sid/src/chrome/browser/first_run/first_run_linux.cc	2012-09-26 11:00:24.998714580 +0200
@@ -101,8 +101,7 @@
 FilePath MasterPrefsPath() {
   // The standard location of the master prefs is next to the chrome binary.
   FilePath master_prefs;
-  if (!PathService::Get(base::DIR_EXE, &master_prefs))
-    return FilePath();
+  master_prefs = FilePath("/etc/chromium");
   return master_prefs.AppendASCII(installer::kDefaultMasterPrefs);
 }
 
