Author:  Raymond Wooninck (tittiatcoke)
Source: OpenSUSE (chromium-master-prefs-path.patch)
Index: chromium/src/chrome/browser/first_run/first_run_linux.cc
===================================================================
--- chromium.orig/src/chrome/browser/first_run/first_run_linux.cc
+++ chromium/src/chrome/browser/first_run/first_run_linux.cc
@@ -100,8 +100,7 @@ void AutoImport(
 FilePath MasterPrefsPath() {
   // The standard location of the master prefs is next to the chrome binary.
   FilePath master_prefs;
-  if (!PathService::Get(base::DIR_EXE, &master_prefs))
-    return FilePath();
+  master_prefs = FilePath("/etc/chromium");
   return master_prefs.AppendASCII(installer::kDefaultMasterPrefs);
 }
 

