Author:  Raymond Wooninck (tittiatcoke)
Source: OpenSUSE (chromium-master-prefs-path.patch)
Index: sid/src/chrome/browser/first_run/first_run_linux.cc
===================================================================
--- sid.orig/src/chrome/browser/first_run/first_run_linux.cc	2013-01-11 11:04:05.000000000 +0100
+++ sid/src/chrome/browser/first_run/first_run_linux.cc	2013-01-11 12:57:25.266734018 +0100
@@ -101,8 +101,7 @@
 FilePath MasterPrefsPath() {
   // The standard location of the master prefs is next to the chrome binary.
   FilePath master_prefs;
-  if (!PathService::Get(base::DIR_EXE, &master_prefs))
-    return FilePath();
+  master_prefs = FilePath("/etc/chromium");
   return master_prefs.AppendASCII(installer::kDefaultMasterPrefs);
 }
 
