--- a/src/chrome/app/policy/cloud_policy_codegen.gyp
+++ b/src/chrome/app/policy/cloud_policy_codegen.gyp
@@ -67,7 +67,7 @@
         {
           'action_name': 'compile_generated_proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '<(policy_out_dir)/policy/cloud_policy.proto',
           ],
           'outputs': [
@@ -76,7 +76,7 @@
             '<(protoc_out_dir)/<(proto_path_substr)/cloud_policy.pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=<(policy_out_dir)/policy',
             '<(policy_out_dir)/policy/cloud_policy.proto',
             '--cpp_out=<(protoc_out_dir)/<(proto_path_substr)',
@@ -86,7 +86,7 @@
         },
       ],
       'dependencies': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protoc#host',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
         'cloud_policy_code_generate',
       ],
       'direct_dependent_settings': {
@@ -109,7 +109,7 @@
           'rule_name': 'gen_proto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'outputs': [
             '<(PRODUCT_DIR)/pyproto/device_management_pb/<(RULE_INPUT_ROOT)_pb2.py',
@@ -117,7 +117,7 @@
             '<(protoc_out_dir)/<(proto_path_substr)/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=<(proto_rel_path)',
             '<(proto_rel_path)/<(RULE_INPUT_NAME)',
             '--cpp_out=<(protoc_out_dir)/<(proto_path_substr)',
@@ -127,7 +127,7 @@
         },
       ],
       'dependencies': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protoc#host',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
@@ -164,7 +164,7 @@
         'cloud_policy_code_generate',
         'cloud_policy_proto_compile',
         'cloud_policy_backend_header_compile',
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
       ],
     },
   ],
--- a/src/chrome/browser/sync/protocol/sync_proto.gyp
+++ b/src/chrome/browser/sync/protocol/sync_proto.gyp
@@ -33,7 +33,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'outputs': [
             '<(PRODUCT_DIR)/pyproto/sync_pb/<(RULE_INPUT_ROOT)_pb2.py',
@@ -41,7 +41,7 @@
             '<(protoc_out_dir)/chrome/browser/sync/protocol/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=.',
             './<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(protoc_out_dir)/chrome/browser/sync/protocol',
@@ -51,7 +51,7 @@
         },
       ],
       'dependencies': [
-        '../../../../third_party/protobuf/protobuf.gyp:protoc#host',
+        '../../../../build/linux/system.gyp:protobuf',
       ],
     },
     {
@@ -84,11 +84,11 @@
         '<(protoc_out_dir)/chrome/browser/sync/protocol/typed_url_specifics.pb.h',
       ],
       'export_dependent_settings': [
-        '../../../../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../../../../build/linux/system.gyp:protobuf',
         'sync_proto',
       ],
       'dependencies': [
-        '../../../../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../../../../build/linux/system.gyp:protobuf',
         'sync_proto',
       ],
       'direct_dependent_settings': {
--- a/src/chrome/chrome_browser.gypi
+++ b/src/chrome/chrome_browser.gypi
@@ -4213,8 +4213,7 @@
                 ['exclude', '^browser/ui/login/login_prompt_gtk.cc'],
               ],
               'dependencies': [
-                '../third_party/protobuf/protobuf.gyp:protobuf_lite',
-                '../third_party/protobuf/protobuf.gyp:protoc#host',
+                '../build/linux/system.gyp:protobuf',
               ],
               'conditions': [
                 ['system_libcros==0', {
@@ -4237,7 +4236,7 @@
                 {
                   'action_name': 'my_proto',
                   'inputs': [
-                    '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+                    '/usr/bin/protoc',
                     'browser/metrics/system_metrics.proto',
                   ],
                   'outputs': [
@@ -4245,10 +4244,10 @@
                     '<(INTERMEDIATE_DIR)/chrome/browser/metrics/system_metrics.pb.h',
                   ],
                   'dependencies': [
-                    '../third_party/protobuf/protobuf.gyp:protoc',
+                    '../build/linux/system.gyp:protobuf',
                   ],
                   'action': [
-                    '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+                    '/usr/bin/protoc',
                     'browser/metrics/system_metrics.proto',
                     '--cpp_out=<(INTERMEDIATE_DIR)/chrome',
                   ],
@@ -4322,7 +4321,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'variables': {
             # The protoc compiler requires a proto_path argument with the
@@ -4335,7 +4334,7 @@
             '<(protoc_out_dir)/chrome/<(rule_input_relpath)/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=./<(rule_input_relpath)',
             './<(rule_input_relpath)/<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(protoc_out_dir)/chrome/<(rule_input_relpath)',
@@ -4344,8 +4343,7 @@
         },
       ],
       'dependencies': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
-        '../third_party/protobuf/protobuf.gyp:protoc#host',
+        '../build/linux/system.gyp:protobuf',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
@@ -4353,7 +4351,7 @@
         ]
       },
       'export_dependent_settings': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
       ],
     },
     {
@@ -4367,7 +4365,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'variables': {
             # The protoc compiler requires a proto_path argument with the
@@ -4381,7 +4379,7 @@
             '<(protoc_out_dir)/chrome/<(rule_input_relpath)/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=./<(rule_input_relpath)',
             './<(rule_input_relpath)/<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(protoc_out_dir)/chrome/<(rule_input_relpath)',
@@ -4390,8 +4388,7 @@
         },
       ],
       'dependencies': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
-        '../third_party/protobuf/protobuf.gyp:protoc#host',
+        '../build/linux/system.gyp:protobuf',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
@@ -4399,7 +4396,7 @@
         ]
       },
       'export_dependent_settings': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
       ],
     },
     {
@@ -4413,7 +4410,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'variables': {
             # The protoc compiler requires a proto_path argument with the
@@ -4427,7 +4424,7 @@
             '<(protoc_out_dir)/chrome/<(rule_input_relpath)/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=./<(rule_input_relpath)',
             './<(rule_input_relpath)/<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(protoc_out_dir)/chrome/<(rule_input_relpath)',
@@ -4436,8 +4433,7 @@
         },
       ],
       'dependencies': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
-        '../third_party/protobuf/protobuf.gyp:protoc#host',
+        '../build/linux/system.gyp:protobuf',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
@@ -4445,7 +4441,7 @@
         ]
       },
       'export_dependent_settings': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
       ],
     },
     {
--- a/src/chrome/chrome_renderer.gypi
+++ b/src/chrome/chrome_renderer.gypi
@@ -229,7 +229,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'variables': {
             # The protoc compiler requires a proto_path argument with the
@@ -242,7 +242,7 @@
             '<(protoc_out_dir)/chrome/<(rule_input_relpath)/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=./<(rule_input_relpath)',
             './<(rule_input_relpath)/<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(protoc_out_dir)/chrome/<(rule_input_relpath)',
@@ -251,8 +251,7 @@
         },
       ],
       'dependencies': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
-        '../third_party/protobuf/protobuf.gyp:protoc#host',
+        '../build/linux/system.gyp:protobuf',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
@@ -260,7 +259,7 @@
         ]
       },
       'export_dependent_settings': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
       ],
     },
   ],
--- a/src/net/net.gyp
+++ b/src/net/net.gyp
@@ -1216,7 +1216,7 @@
           'dependencies': [
             '../chrome/app/policy/cloud_policy_codegen.gyp:cloud_policy_proto_compile',
             '../chrome/browser/sync/protocol/sync_proto.gyp:sync_proto',
-            '../third_party/protobuf/protobuf.gyp:py_proto',
+            '../build/linux/system.gyp:protobuf',
           ],
         }],
         ['OS == "linux" or OS == "freebsd" or OS == "openbsd"', {
--- a/src/remoting/proto/chromotocol.gyp
+++ b/src/remoting/proto/chromotocol.gyp
@@ -24,14 +24,14 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'outputs': [
             '<(out_dir)/<(RULE_INPUT_ROOT).pb.cc',
             '<(out_dir)/<(RULE_INPUT_ROOT).pb.h',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=.',
             './<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(out_dir)',
@@ -40,7 +40,7 @@
         },
       ],
       'dependencies': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protoc#host',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
@@ -56,11 +56,11 @@
       'target_name': 'chromotocol_proto_lib',
       'type': '<(library)',
       'export_dependent_settings': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
         'chromotocol_proto',
       ],
       'dependencies': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
         'chromotocol_proto',
       ],
       # This target exports a hard dependency because depedents require
--- a/src/remoting/proto/trace.gyp
+++ b/src/remoting/proto/trace.gyp
@@ -20,7 +20,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'outputs': [
             '<(PRODUCT_DIR)/pyproto/<(RULE_INPUT_ROOT)_pb/'
@@ -29,7 +29,7 @@
             '<(out_dir)/<(RULE_INPUT_ROOT).pb.h',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=.',
             './<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(out_dir)',
@@ -39,7 +39,7 @@
         },
       ],
       'dependencies': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protoc#host',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
       ],
       # This target exports a hard dependency because it generates header
       # files.
@@ -50,11 +50,11 @@
       'target_name': 'trace_proto_lib',
       'type': '<(library)',
       'export_dependent_settings': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
         'trace_proto',
       ],
       'dependencies': [
-        '<(DEPTH)/third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '<(DEPTH)/build/linux/system.gyp:protobuf',
         'trace_proto',
       ],
       # This target exports a hard dependency because depedents require
--- a/src/remoting/remoting.gyp
+++ b/src/remoting/remoting.gyp
@@ -116,7 +116,7 @@
       'dependencies': [
         '../ui/ui.gyp:ui_gfx',
         '../media/media.gyp:media',
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
         '../third_party/libvpx/libvpx.gyp:libvpx_include',
         '../third_party/zlib/zlib.gyp:zlib',
         'chromoting_jingle_glue',
@@ -126,7 +126,7 @@
         #'third_party/on2/on2.gyp:vp8',
       ],
       'export_dependent_settings': [
-        '../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
         'proto/chromotocol.gyp:chromotocol_proto_lib',
       ],
       # This target needs a hard dependency because dependent targets
--- a/src/third_party/cacheinvalidation/cacheinvalidation.gyp
+++ b/src/third_party/cacheinvalidation/cacheinvalidation.gyp
@@ -17,7 +17,7 @@
     # Where files generated from proto files are put.
     'protoc_out_dir': '<(SHARED_INTERMEDIATE_DIR)/protoc_out',
     # The path to the protoc executable.
-    'protoc': '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+    'protoc': '/usr/bin/protoc',
   },
   'targets': [
     # The rule/action to generate files from the cacheinvalidation proto
@@ -55,7 +55,7 @@
         },
       ],
       'dependencies': [
-        '../../third_party/protobuf/protobuf.gyp:protoc#host',
+        '../../build/linux/system.gyp:protobuf',
       ],
     },
     # The C++ files generated from the cache invalidation protocol buffers.
@@ -71,7 +71,7 @@
         '<(protoc_out_dir)/<(proto_dir_relpath)/types.pb.cc',
       ],
       'dependencies': [
-        '../../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../../build/linux/system.gyp:protobuf',
         'cacheinvalidation_proto',
       ],
       'include_dirs': [
@@ -83,7 +83,7 @@
         ],
       },
       'export_dependent_settings': [
-        '../../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../../build/linux/system.gyp:protobuf',
       ],
       # This target exports a hard dependency because it contains generated
       # header files.
--- a/src/third_party/safe_browsing/safe_browsing.gyp
+++ b/src/third_party/safe_browsing/safe_browsing.gyp
@@ -13,10 +13,10 @@
         'testing/testing_input.dat',
       ],
       'export_dependent_settings': [
-        '../protobuf/protobuf.gyp:py_proto',
+        '../../build/linux/system.gyp:protobuf',
       ],
       'dependencies': [
-        '../protobuf/protobuf.gyp:py_proto',
+        '../../build/linux/system.gyp:protobuf',
       ],
      },
   ],
--- a/src/chrome/browser/sync/protocol/encryption.proto
+++ b/src/chrome/browser/sync/protocol/encryption.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/sync.proto
+++ b/src/chrome/browser/sync/protocol/sync.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/app_specifics.proto
+++ b/src/chrome/browser/sync/protocol/app_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/autofill_specifics.proto
+++ b/src/chrome/browser/sync/protocol/autofill_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/bookmark_specifics.proto
+++ b/src/chrome/browser/sync/protocol/bookmark_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/extension_specifics.proto
+++ b/src/chrome/browser/sync/protocol/extension_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/nigori_specifics.proto
+++ b/src/chrome/browser/sync/protocol/nigori_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/password_specifics.proto
+++ b/src/chrome/browser/sync/protocol/password_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/preference_specifics.proto
+++ b/src/chrome/browser/sync/protocol/preference_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/session_specifics.proto
+++ b/src/chrome/browser/sync/protocol/session_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/test.proto
+++ b/src/chrome/browser/sync/protocol/test.proto
@@ -7,7 +7,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/theme_specifics.proto
+++ b/src/chrome/browser/sync/protocol/theme_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
--- a/src/chrome/browser/sync/protocol/typed_url_specifics.proto
+++ b/src/chrome/browser/sync/protocol/typed_url_specifics.proto
@@ -10,7 +10,7 @@
 syntax = "proto2";
 
 option optimize_for = LITE_RUNTIME;
-option retain_unknown_fields = true;
+//option retain_unknown_fields = true;
 
 package sync_pb;
 
