Do not embedd protobuf
Index: chromium-browser-6.0.472.36~r55963/src/chrome/chrome_browser.gypi
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/chrome/chrome_browser.gypi	2010-08-18 16:18:55.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/chrome/chrome_browser.gypi	2010-08-18 16:18:55.000000000 +0000
@@ -46,6 +46,7 @@
       ],
       'defines': [
         '<@(nacl_defines)',
+        'GOOGLE_PROTOBUF_NO_RTTI',
       ],
       'direct_dependent_settings': {
         'defines': [
@@ -3527,16 +3528,14 @@
                 ['exclude', '^browser/notifications/balloon_collection_linux.cc'],
               ],
               'dependencies': [
-                '../third_party/protobuf2/protobuf.gyp:protobuf_lite',
-                #'../third_party/protobuf2/protobuf.gyp:protobuf',
-                '../third_party/protobuf2/protobuf.gyp:protoc#host',
+                '../build/linux/system.gyp:protobuf',
                 '../third_party/cros/cros_api.gyp:cros_api',
               ],
               'actions': [
                 {
                   'action_name': 'my_proto',
                   'inputs': [
-                    '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+                    '/usr/bin/protoc',
                     'browser/metrics/system_metrics.proto',
                   ],
                   'outputs': [
@@ -3544,10 +3543,10 @@
                     '<(INTERMEDIATE_DIR)/chrome/browser/metrics/system_metrics.pb.h',
                   ],
                   'dependencies': [
-                    '../third_party/protobuf2/protobuf.gyp:protoc',
+                    '../build/linux/system.gyp:protobuf',
                   ],
                   'action': [
-                    '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+                    '/usr/bin/protoc',
                     'browser/metrics/system_metrics.proto',
                     '--cpp_out=<(INTERMEDIATE_DIR)/chrome',
                   ],
@@ -3631,7 +3630,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'variables': {
             # The protoc compiler requires a proto_path argument with the
@@ -3644,7 +3643,7 @@
             '<(protoc_out_dir)/chrome/<(rule_input_relpath)/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=./<(rule_input_relpath)',
             './<(rule_input_relpath)/<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(protoc_out_dir)/chrome/<(rule_input_relpath)',
@@ -3653,8 +3652,7 @@
         },
       ],
       'dependencies': [
-        '../third_party/protobuf2/protobuf.gyp:protobuf',
-        '../third_party/protobuf2/protobuf.gyp:protoc#host',
+        '../build/linux/system.gyp:protobuf',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
@@ -3662,7 +3660,7 @@
         ]
       },
       'export_dependent_settings': [
-        '../third_party/protobuf2/protobuf.gyp:protobuf',
+        '../build/linux/system.gyp:protobuf',
       ],
     },
   ],
Index: chromium-browser-6.0.472.36~r55963/src/build/linux/system.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/build/linux/system.gyp	2010-08-18 16:18:55.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/build/linux/system.gyp	2010-08-18 16:19:51.553746647 +0000
@@ -276,6 +276,26 @@
       }]]
     },
     {
+      'target_name': 'protobuf',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(pkg-config --cflags protobuf)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other protobuf)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l protobuf)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'v8',
       'type': 'settings',
       'conditions': [
Index: chromium-browser-6.0.472.36~r55963/src/chrome/browser/sync/protocol/sync_proto.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/chrome/browser/sync/protocol/sync_proto.gyp	2010-08-04 15:01:18.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/chrome/browser/sync/protocol/sync_proto.gyp	2010-08-18 16:18:55.000000000 +0000
@@ -29,7 +29,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'outputs': [
             '<(PRODUCT_DIR)/pyproto/sync_pb/<(RULE_INPUT_ROOT)_pb2.py',
@@ -37,7 +37,7 @@
             '<(SHARED_INTERMEDIATE_DIR)/protoc_out/chrome/browser/sync/protocol/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=.',
             './<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(SHARED_INTERMEDIATE_DIR)/protoc_out/chrome/browser/sync/protocol',
@@ -47,7 +47,7 @@
         },
       ],
       'dependencies': [
-        '../../../../third_party/protobuf2/protobuf.gyp:protoc#host',
+        '../../../../build/linux/system.gyp:protobuf',
       ],
     },
     {
@@ -56,13 +56,13 @@
       'export_dependent_settings': [
         # TODO(akalin): Change back to protobuf_lite once it supports
         # preserving unknown fields.
-        '../../../../third_party/protobuf2/protobuf.gyp:protobuf',
+        '../../../../build/linux/system.gyp:protobuf',
         'sync_proto',
       ],
       'dependencies': [
         # TODO(akalin): Change back to protobuf_lite once it supports
         # preserving unknown fields.
-        '../../../../third_party/protobuf2/protobuf.gyp:protobuf',
+        '../../../../build/linux/system.gyp:protobuf',
         'sync_proto',
       ],
       'direct_dependent_settings': {
Index: chromium-browser-6.0.472.36~r55963/src/chrome/chrome.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/chrome/chrome.gyp	2010-08-18 16:18:55.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/chrome/chrome.gyp	2010-08-18 16:19:51.557747075 +0000
@@ -953,7 +953,7 @@
         'browser/sync/protocol/sync_proto.gyp:sync_proto_cpp',
         # TODO(akalin): Change back to protobuf_lite once it supports
         # preserving unknown fields.
-        '../third_party/protobuf2/protobuf.gyp:protobuf#target',
+        '../build/linux/system.gyp:protobuf',
       ],
       'conditions': [
         ['OS=="win"', {
Index: chromium-browser-6.0.472.36~r55963/src/native_client/build/all.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/native_client/build/all.gyp	2010-08-04 15:02:34.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/native_client/build/all.gyp	2010-08-18 16:18:55.000000000 +0000
@@ -16,7 +16,7 @@
 #        '../src/shared/platform/platform_tests.gyp:*',
         '../src/shared/srpc/srpc.gyp:*',
         '../src/shared/utils/utils.gyp:*',
-        '../src/third_party/protobuf/protobuf.gyp:*',
+        '../build/linux/system.gyp:protobuf',
         '../src/trusted/debug_stub/debug_stub.gyp:*',
         '../src/trusted/desc/desc.gyp:*',
         '../src/trusted/nacl_breakpad/nacl_breakpad.gyp:*',
Index: chromium-browser-6.0.472.36~r55963/src/net/net.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/net/net.gyp	2010-08-18 16:18:55.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/net/net.gyp	2010-08-18 16:18:55.000000000 +0000
@@ -924,7 +924,7 @@
         ['inside_chromium_build==1', {
           'dependencies': [
             '../chrome/browser/sync/protocol/sync_proto.gyp:sync_proto',
-            '../third_party/protobuf2/protobuf.gyp:py_proto',
+            '../build/linux/system.gyp:protobuf',
           ],
         }],
         ['OS == "linux" or OS == "freebsd" or OS == "openbsd"', {
Index: chromium-browser-6.0.472.36~r55963/src/remoting/base/protocol/chromotocol.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/remoting/base/protocol/chromotocol.gyp	2010-08-04 14:59:33.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/remoting/base/protocol/chromotocol.gyp	2010-08-18 16:18:55.000000000 +0000
@@ -20,7 +20,7 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'outputs': [
             '<(PRODUCT_DIR)/pyproto/chromotocol_pb/<(RULE_INPUT_ROOT)_pb2.py',
@@ -28,7 +28,7 @@
             '<(out_dir)/<(RULE_INPUT_ROOT).pb.h',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=.',
             './<(RULE_INPUT_ROOT)<(RULE_INPUT_EXT)',
             '--cpp_out=<(out_dir)',
@@ -38,7 +38,7 @@
         },
       ],
       'dependencies': [
-        '../../../third_party/protobuf2/protobuf.gyp:protoc#host',
+        '../../../build/linux/system.gyp:protobuf',
       ],
       # This target exports a hard dependency because it generates header
       # files.
@@ -49,11 +49,11 @@
       'target_name': 'chromotocol_proto_lib',
       'type': '<(library)',
       'export_dependent_settings': [
-        '../../../third_party/protobuf2/protobuf.gyp:protobuf_lite',
+        '../../../build/linux/system.gyp:protobuf',
         'chromotocol_proto',
       ],
       'dependencies': [
-        '../../../third_party/protobuf2/protobuf.gyp:protobuf_lite',
+        '../../../build/linux/system.gyp:protobuf',
         'chromotocol_proto',
       ],
       # This target exports a hard dependency because depedents require
Index: chromium-browser-6.0.472.36~r55963/src/remoting/remoting.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/remoting/remoting.gyp	2010-08-04 14:59:33.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/remoting/remoting.gyp	2010-08-18 16:18:55.000000000 +0000
@@ -112,7 +112,7 @@
       'dependencies': [
         '../gfx/gfx.gyp:gfx',
         '../media/media.gyp:media',
-        '../third_party/protobuf2/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
         '../third_party/zlib/zlib.gyp:zlib',
         'base/protocol/chromotocol.gyp:chromotocol_proto_lib',
         'chromoting_jingle_glue',
@@ -120,7 +120,7 @@
         #'third_party/on2/on2.gyp:vp8',
       ],
       'export_dependent_settings': [
-        '../third_party/protobuf2/protobuf.gyp:protobuf_lite',
+        '../build/linux/system.gyp:protobuf',
         'base/protocol/chromotocol.gyp:chromotocol_proto_lib',
         # TODO(hclam): Enable VP8 in the build.
         #'third_party/on2/on2.gyp:vp8',
Index: chromium-browser-6.0.472.36~r55963/src/third_party/cacheinvalidation/cacheinvalidation.gyp
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/third_party/cacheinvalidation/cacheinvalidation.gyp	2010-08-04 15:00:05.000000000 +0000
+++ chromium-browser-6.0.472.36~r55963/src/third_party/cacheinvalidation/cacheinvalidation.gyp	2010-08-18 16:18:55.000000000 +0000
@@ -33,14 +33,14 @@
           'rule_name': 'genproto',
           'extension': 'proto',
           'inputs': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
           ],
           'outputs': [
             '<(protoc_out_dir)/<(proto_dir_relpath)/<(RULE_INPUT_ROOT).pb.h',
             '<(protoc_out_dir)/<(proto_dir_relpath)/<(RULE_INPUT_ROOT).pb.cc',
           ],
           'action': [
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)protoc<(EXECUTABLE_SUFFIX)',
+            '/usr/bin/protoc',
             '--proto_path=<(proto_dir_root)',
             # This path needs to be prefixed by proto_path, so we can't
             # use RULE_INPUT_PATH (which is an absolute path).
@@ -51,7 +51,7 @@
         },
       ],
       'dependencies': [
-        '../../third_party/protobuf2/protobuf.gyp:protoc#host',
+        '../../build/linux/system.gyp:protobuf',
       ],
       # This target exports a hard dependency because it generates header
       # files.
@@ -97,7 +97,7 @@
       ],
       'dependencies': [
         '../../base/base.gyp:base',
-        '../../third_party/protobuf2/protobuf.gyp:protobuf_lite',
+        '../../build/linux/system.gyp:protobuf',
         'cacheinvalidation_proto',
       ],
       # This target exports a hard dependency because depedents require
@@ -111,7 +111,7 @@
         ],
       },
       'export_dependent_settings': [
-        '../../third_party/protobuf2/protobuf.gyp:protobuf_lite',
+        '../../build/linux/system.gyp:protobuf',
         'cacheinvalidation_proto',
       ],
     },
Index: chromium-browser-6.0.472.36~r55963/src/build/common.gypi
===================================================================
--- chromium-browser-6.0.472.36~r55963.orig/src/build/common.gypi	2010-08-18 16:20:10.114496753 +0000
+++ chromium-browser-6.0.472.36~r55963/src/build/common.gypi	2010-08-18 16:28:58.069501686 +0000
@@ -738,6 +738,9 @@
             'cflags': [
              '<@(release_extra_cflags)',
             ],
+            'defines': [
+              'GOOGLE_PROTOBUF_NO_RTTI',
+            ],
           }],
         ],
       },
