description: add ps printing capability
author: Salvatore Bonaccorso
Index: chromium/src/chrome/browser/printing/print_dialog_gtk.cc
===================================================================
--- chromium.orig/src/chrome/browser/printing/print_dialog_gtk.cc	2013-11-13 07:52:35.136670940 +0000
+++ chromium/src/chrome/browser/printing/print_dialog_gtk.cc	2013-11-13 07:52:35.136670940 +0000
@@ -269,6 +269,7 @@
   // Since we only generate PDF, only show printers that support PDF.
   // TODO(thestig) Add more capabilities to support?
   GtkPrintCapabilities cap = static_cast<GtkPrintCapabilities>(
+      GTK_PRINT_CAPABILITY_GENERATE_PS |
       GTK_PRINT_CAPABILITY_GENERATE_PDF |
       GTK_PRINT_CAPABILITY_PAGE_SET |
       GTK_PRINT_CAPABILITY_COPIES |
