description: add ps printing capability
author: Salvatore Bonaccorso
Index: sid/src/chrome/browser/printing/print_dialog_gtk.cc
===================================================================
--- sid.orig/src/chrome/browser/printing/print_dialog_gtk.cc	2014-02-02 19:28:13.932401861 +0100
+++ sid/src/chrome/browser/printing/print_dialog_gtk.cc	2014-02-02 19:28:13.928401861 +0100
@@ -245,6 +245,7 @@
   // Since we only generate PDF, only show printers that support PDF.
   // TODO(thestig) Add more capabilities to support?
   GtkPrintCapabilities cap = static_cast<GtkPrintCapabilities>(
+      GTK_PRINT_CAPABILITY_GENERATE_PS |
       GTK_PRINT_CAPABILITY_GENERATE_PDF |
       GTK_PRINT_CAPABILITY_PAGE_SET |
       GTK_PRINT_CAPABILITY_COPIES |
