Index: sid/src/media/audio/pulse/pulse_output.cc
===================================================================
--- sid.orig/src/media/audio/pulse/pulse_output.cc	2013-01-29 17:08:01.000000000 +0100
+++ sid/src/media/audio/pulse/pulse_output.cc	2013-02-06 13:38:14.129887904 +0100
@@ -86,7 +86,7 @@
 
   // All channel maps have the same size array of channel positions.
   for (unsigned int channel = 0; channel != CHANNELS_MAX; ++channel) {
-    int channel_position = kChannelOrderings[channel_layout][channel];
+    int channel_position = ChannelOrder(channel_layout, static_cast<Channels>(channel));
     if (channel_position > -1) {
       channel_map.map[channel_position] = ChromiumToPAChannelPosition(
           static_cast<Channels>(channel));
