--- a/third_party/skia/src/ports/SkFontHost_FreeType_common.cpp
+++ b/third_party/skia/src/ports/SkFontHost_FreeType_common.cpp
@@ -31,6 +31,10 @@
 #include FT_SYNTHESIS_H
 
 #ifdef TT_SUPPORT_COLRV1
+#undef TT_SUPPORT_COLRV1
+#endif
+
+#ifdef TT_SUPPORT_COLRV1
 #include "src/core/SkScopeExit.h"
 #endif
 
--- a/third_party/skia/src/ports/SkFontHost_FreeType.cpp
+++ b/third_party/skia/src/ports/SkFontHost_FreeType.cpp
@@ -53,6 +53,10 @@
 #include <freetype/t1tables.h>
 #include <freetype/ftfntfmt.h>
 
+#ifdef TT_SUPPORT_COLRV1
+#undef TT_SUPPORT_COLRV1
+#endif
+
 // SK_FREETYPE_MINIMUM_RUNTIME_VERSION 0x<major><minor><patch><flags>
 // Flag SK_FREETYPE_DLOPEN: also try dlopen to get newer features.
 #define SK_FREETYPE_DLOPEN (0x1)
