Use system copy of libv8
Index: chromium/src/build/linux/system.gyp
===================================================================
--- chromium.orig/src/build/linux/system.gyp	2012-02-10 23:37:19.000000000 -0500
+++ chromium/src/build/linux/system.gyp	2012-02-10 23:37:29.000000000 -0500
@@ -400,6 +400,18 @@
       }]]
     },
     {
+      'target_name': 'v8',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'libraries': [
+              '-lv8',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
Index: chromium/src/net/net.gyp
===================================================================
--- chromium.orig/src/net/net.gyp	2012-02-10 23:36:44.000000000 -0500
+++ chromium/src/net/net.gyp	2012-02-10 23:37:31.000000000 -0500
@@ -35,7 +35,7 @@
         '../third_party/icu/icu.gyp:icui18n',
         '../third_party/icu/icu.gyp:icuuc',
         '../third_party/zlib/zlib.gyp:zlib',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         'net_resources',
         'ssl_false_start_blacklist_process#host',
       ],
Index: chromium/src/build/all.gyp
===================================================================
--- chromium.orig/src/build/all.gyp	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/build/all.gyp	2012-02-10 23:37:31.000000000 -0500
@@ -50,7 +50,7 @@
         '../third_party/sqlite/sqlite.gyp:*',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:*',
         '../third_party/zlib/zlib.gyp:*',
-        '../v8/tools/gyp/v8.gyp:*',
+        '../build/linux/system.gyp:v8',
         '../webkit/support/webkit_support.gyp:*',
         '../webkit/webkit.gyp:*',
         'util/build_util.gyp:*',
Index: chromium/src/chrome/renderer/loadtimes_extension_bindings.cc
===================================================================
--- chromium.orig/src/chrome/renderer/loadtimes_extension_bindings.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/loadtimes_extension_bindings.cc	2012-02-10 23:37:31.000000000 -0500
@@ -9,7 +9,7 @@
 #include "base/time.h"
 #include "content/public/renderer/document_state.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDataSource;
 using WebKit::WebFrame;
Index: chromium/src/net/proxy/proxy_resolver_v8.cc
===================================================================
--- chromium.orig/src/net/proxy/proxy_resolver_v8.cc	2012-02-10 23:36:44.000000000 -0500
+++ chromium/src/net/proxy/proxy_resolver_v8.cc	2012-02-10 23:37:31.000000000 -0500
@@ -24,7 +24,7 @@
 #include "net/proxy/proxy_resolver_js_bindings.h"
 #include "net/proxy/proxy_resolver_request_context.h"
 #include "net/proxy/proxy_resolver_script.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // Notes on the javascript environment:
 //
Index: chromium/src/webkit/glue/webkit_glue.cc
===================================================================
--- chromium.orig/src/webkit/glue/webkit_glue.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/glue/webkit_glue.cc	2012-02-11 00:21:29.000000000 -0500
@@ -45,7 +45,7 @@
 #if defined(OS_WIN)
 #include "third_party/WebKit/Source/WebKit/chromium/public/win/WebInputEventFactory.h"
 #endif
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/glue_serialize.h"
 #include "webkit/glue/user_agent.h"
 
Index: chromium/src/webkit/tools/test_shell/listener_leak_test.cc
===================================================================
--- chromium.orig/src/webkit/tools/test_shell/listener_leak_test.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/tools/test_shell/listener_leak_test.cc	2012-02-10 23:37:31.000000000 -0500
@@ -8,8 +8,8 @@
 #include "base/string_number_conversions.h"
 #include "base/string_util.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
-#include "v8/include/v8-profiler.h"
+#include <v8.h>
+#include <v8-profiler.h>
 #include "webkit/glue/webkit_glue.h"
 #include "webkit/tools/test_shell/test_shell.h"
 #include "webkit/tools/test_shell/test_shell_test.h"
Index: chromium/src/chrome/renderer/search_extension.cc
===================================================================
--- chromium.orig/src/chrome/renderer/search_extension.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/search_extension.cc	2012-02-10 23:37:31.000000000 -0500
@@ -11,7 +11,7 @@
 #include "chrome/renderer/searchbox.h"
 #include "content/public/renderer/render_view.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebView;
Index: chromium/src/webkit/glue/webkit_glue.gypi
===================================================================
--- chromium.orig/src/webkit/glue/webkit_glue.gypi	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/glue/webkit_glue.gypi	2012-02-10 23:37:31.000000000 -0500
@@ -130,7 +130,7 @@
         '<(DEPTH)/third_party/icu/icu.gyp:icuuc',
         '<(DEPTH)/third_party/npapi/npapi.gyp:npapi',
         '<(DEPTH)/ui/ui.gyp:ui',
-        '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+        '<(DEPTH)/build/linux/system.gyp:v8',
         'fileapi',
         'webkit_media',
         'webkit_resources',
@@ -521,7 +521,7 @@
                ],
                'export_dependent_settings': [
                  '<(DEPTH)/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
-                 '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+                 '<(DEPTH)/build/linux/system.gyp:v8',
                ],
             }],
           ],
Index: chromium/src/chrome/renderer/searchbox_extension.cc
===================================================================
--- chromium.orig/src/chrome/renderer/searchbox_extension.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/searchbox_extension.cc	2012-02-11 00:20:37.000000000 -0500
@@ -15,7 +15,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebScriptSource;
Index: chromium/src/v8/tools/gyp/v8.gyp
===================================================================
--- chromium.orig/src/v8/tools/gyp/v8.gyp	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/v8/tools/gyp/v8.gyp	2012-02-10 23:37:31.000000000 -0500
@@ -932,7 +932,7 @@
             }],
           ],
           'dependencies': [
-            'v8'
+            '../../../build/linux/system.gyp:v8',
           ],
         },
       ],
Index: chromium/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
===================================================================
--- chromium.orig/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp	2012-02-10 23:37:31.000000000 -0500
@@ -1003,7 +1003,7 @@
         '<(chromium_src_dir)/third_party/libwebp/libwebp.gyp:libwebp',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'include_dirs': [
@@ -1127,7 +1127,7 @@
         '<(chromium_src_dir)/third_party/ots/ots.gyp:ots',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'export_dependent_settings': [
@@ -1145,7 +1145,7 @@
         '<(chromium_src_dir)/third_party/ots/ots.gyp:ots',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       # This is needed for mac because of webkit_system_interface. It'd be nice
@@ -1900,7 +1900,7 @@
         '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'export_dependent_settings': [
         'webcore_bindings',
@@ -1909,7 +1909,7 @@
         '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
Index: chromium/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp
===================================================================
--- chromium.orig/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp	2012-02-10 23:37:31.000000000 -0500
@@ -32,8 +32,8 @@
 
 #include "TestShell.h"
 #include "webkit/support/webkit_support.h"
-#include <v8/include/v8-testing.h>
-#include <v8/include/v8.h>
+#include <v8-testing.h>
+#include <v8.h>
 #include <wtf/Vector.h>
 
 using namespace std;
Index: chromium/src/chrome/browser/task_manager/task_manager_resource_providers.cc
===================================================================
--- chromium.orig/src/chrome/browser/task_manager/task_manager_resource_providers.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/browser/task_manager/task_manager_resource_providers.cc	2012-02-10 23:37:31.000000000 -0500
@@ -55,7 +55,7 @@
 #include "third_party/sqlite/sqlite3.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(OS_MACOSX)
 #include "skia/ext/skia_utils_mac.h"
Index: chromium/src/chrome/chrome.gyp
===================================================================
--- chromium.orig/src/chrome/chrome.gyp	2012-02-10 23:37:19.000000000 -0500
+++ chromium/src/chrome/chrome.gyp	2012-02-10 23:37:31.000000000 -0500
@@ -1094,7 +1094,7 @@
             '../testing/gtest.gyp:*',
             '../tools/memory_watcher/memory_watcher.gyp:*',
             '../ui/ui.gyp:*',
-            '../v8/tools/gyp/v8.gyp:v8_shell',
+            '../build/linux/system.gyp:v8',
             '../webkit/support/webkit_support.gyp:*',
             '../webkit/webkit.gyp:*',
             '<(libjpeg_gyp_path):*',
Index: chromium/src/chrome/chrome_tests.gypi
===================================================================
--- chromium.orig/src/chrome/chrome_tests.gypi	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/chrome_tests.gypi	2012-02-10 23:37:31.000000000 -0500
@@ -1179,7 +1179,7 @@
         '../ui/ui.gyp:ui_resources',
         '../ui/ui.gyp:ui_resources_standard',
         '../ui/ui.gyp:ui_test_support',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         'chrome_resources.gyp:chrome_resources',
         'chrome_resources.gyp:chrome_strings',
       ],
@@ -2393,7 +2393,7 @@
         '../third_party/icu/icu.gyp:icui18n',
         '../third_party/icu/icu.gyp:icuuc',
         '../third_party/leveldatabase/leveldatabase.gyp:leveldatabase',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         '../webkit/webkit.gyp:test_shell_test_support',
         # Runtime dependencies
         '../third_party/mesa/mesa.gyp:osmesa',
@@ -2992,7 +2992,7 @@
         ['target_arch!="arm"', {
           'dependencies': [
             # build time dependency.
-            '../v8/tools/gyp/v8.gyp:v8_shell#host',
+            '../build/linux/system.gyp:v8',
             # run time dependency
             '../webkit/webkit.gyp:copy_npapi_test_plugin',
           ],
Index: chromium/src/chrome/renderer/chrome_render_process_observer.cc
===================================================================
--- chromium.orig/src/chrome/renderer/chrome_render_process_observer.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/chrome_render_process_observer.cc	2012-02-10 23:37:31.000000000 -0500
@@ -42,7 +42,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebRuntimeFeatures.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(OS_WIN)
 #include "base/win/iat_patch_function.h"
Index: chromium/src/chrome/renderer/extensions/extension_dispatcher.cc
===================================================================
--- chromium.orig/src/chrome/renderer/extensions/extension_dispatcher.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/extensions/extension_dispatcher.cc	2012-02-11 00:21:56.000000000 -0500
@@ -32,7 +32,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURLRequest.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 static const int64 kInitialExtensionIdleHandlerDelayMs = 5*1000;
Index: chromium/src/chrome/renderer/external_extension.cc
===================================================================
--- chromium.orig/src/chrome/renderer/external_extension.cc	2012-02-10 23:36:44.000000000 -0500
+++ chromium/src/chrome/renderer/external_extension.cc	2012-02-10 23:37:31.000000000 -0500
@@ -12,7 +12,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebView;
Index: chromium/src/chrome/renderer/translate_helper.cc
===================================================================
--- chromium.orig/src/chrome/renderer/translate_helper.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/translate_helper.cc	2012-02-10 23:37:31.000000000 -0500
@@ -18,7 +18,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "third_party/cld/encodings/compact_lang_det/win/cld_unicodetext.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/dom_operations.h"
 
 using WebKit::WebDocument;
Index: chromium/src/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp
===================================================================
--- chromium.orig/src/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp	2012-02-10 23:37:31.000000000 -0500
@@ -74,13 +74,13 @@
                 '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
                 '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
                 '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
             ],
             'export_dependent_settings': [
                 '<(chromium_src_dir)/skia/skia.gyp:skia',
                 '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
                 '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
             ],
             'include_dirs': [
                 'public',
@@ -694,7 +694,7 @@
                                 '<(chromium_src_dir)/third_party/modp_b64/modp_b64.gyp:modp_b64',
                                 '<(chromium_src_dir)/third_party/ots/ots.gyp:ots',
                                 '<(chromium_src_dir)/third_party/zlib/zlib.gyp:zlib',
-                                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                                 # We must not add webkit_support here because of cyclic dependency.
                             ],
                             'direct_dependent_settings': {
@@ -704,7 +704,7 @@
                             },
                             'export_dependent_settings': [
                                 '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
-                                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                             ],
                             'include_dirs': [
                                 # WARNING: Do not view this particular case as a precedent for
@@ -1091,7 +1091,7 @@
                 '../../JavaScriptCore/JavaScriptCore.gyp/JavaScriptCore.gyp:wtf_config',
                 '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
                 '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                 '<(chromium_src_dir)/webkit/support/webkit_support.gyp:blob',
                 '<(chromium_src_dir)/webkit/support/webkit_support.gyp:webkit_support',
                 '<(chromium_src_dir)/webkit/support/webkit_support.gyp:webkit_user_agent',
Index: chromium/src/webkit/tools/test_shell/test_shell_webkit_init.cc
===================================================================
--- chromium.orig/src/webkit/tools/test_shell/test_shell_webkit_init.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/tools/test_shell/test_shell_webkit_init.cc	2012-02-11 00:22:28.000000000 -0500
@@ -14,7 +14,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptController.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebSecurityPolicy.h"
 #include "ui/gfx/gl/gl_bindings_skia_in_process.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/plugins/npapi/plugin_list.h"
 #include "webkit/plugins/webplugininfo.h"
 #include "webkit/tools/test_shell/simple_socket_stream_bridge.h"
Index: chromium/src/content/content_renderer.gypi
===================================================================
--- chromium.orig/src/content/content_renderer.gypi	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/content/content_renderer.gypi	2012-02-10 23:37:31.000000000 -0500
@@ -17,7 +17,7 @@
     '../third_party/npapi/npapi.gyp:npapi',
     '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
     '../ui/gfx/surface/surface.gyp:surface',
-    '../v8/tools/gyp/v8.gyp:v8',
+    '../build/linux/system.gyp:v8',
     '../webkit/support/webkit_support.gyp:webkit_media',
     '../webkit/support/webkit_support.gyp:webkit_gpu',
   ],
Index: chromium/src/third_party/WebKit/Source/ThirdParty/gyp/samples/samples
===================================================================
--- chromium.orig/src/third_party/WebKit/Source/ThirdParty/gyp/samples/samples	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/WebKit/Source/ThirdParty/gyp/samples/samples	2012-02-10 23:37:31.000000000 -0500
@@ -38,7 +38,7 @@
     'third_party/npapi/npapi.gyp',
     'third_party/sqlite/sqlite.gyp',
     'third_party/zlib/zlib.gyp',
-    'v8/tools/gyp/v8.gyp',
+    'build/linux/system.gyp:v8',
     'webkit/activex_shim/activex_shim.gyp',
     'webkit/activex_shim_dll/activex_shim_dll.gyp',
     'webkit/build/action_csspropertynames.py',
Index: chromium/src/webkit/tools/test_shell/test_shell.gypi
===================================================================
--- chromium.orig/src/webkit/tools/test_shell/test_shell.gypi	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/tools/test_shell/test_shell.gypi	2012-02-10 23:37:31.000000000 -0500
@@ -38,7 +38,7 @@
         '<(DEPTH)/testing/gtest.gyp:gtest',
         '<(DEPTH)/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:inspector_resources',
         '<(DEPTH)/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
-        '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+        '<(DEPTH)/build/linux/system.gyp:v8',
         '<(DEPTH)/webkit/support/webkit_support.gyp:appcache',
         '<(DEPTH)/webkit/support/webkit_support.gyp:blob',
         '<(DEPTH)/webkit/support/webkit_support.gyp:database',
@@ -355,7 +355,7 @@
         '<(DEPTH)/testing/gmock.gyp:gmock',
         '<(DEPTH)/testing/gtest.gyp:gtest',
         '<(DEPTH)/third_party/leveldatabase/leveldatabase.gyp:leveldatabase',
-        '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+        '<(DEPTH)/build/linux/system.gyp:v8',
       ],
       'sources': [
         '../../../skia/ext/convolver_unittest.cc',
Index: chromium/src/chrome/browser/ui/webui/options/chromeos/about_page_handler.cc
===================================================================
--- chromium.orig/src/chrome/browser/ui/webui/options/chromeos/about_page_handler.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/browser/ui/webui/options/chromeos/about_page_handler.cc	2012-02-10 23:37:31.000000000 -0500
@@ -32,7 +32,7 @@
 #include "grit/theme_resources.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/user_agent.h"
 #include "webkit/glue/webkit_glue.h"
 #include "webkit/glue/user_agent.h"
Index: chromium/src/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp
===================================================================
--- chromium.orig/src/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp	2012-02-10 23:37:31.000000000 -0500
@@ -28,7 +28,7 @@
 
 #include "WebCoreTestSupport.h"
 #include "WebFrame.h"
-#include <v8/include/v8.h>
+#include <v8.h>
 
 namespace WebKit {
 
Index: chromium/src/chrome/renderer/chrome_render_view_observer.cc
===================================================================
--- chromium.orig/src/chrome/renderer/chrome_render_view_observer.cc	2012-02-10 23:36:44.000000000 -0500
+++ chromium/src/chrome/renderer/chrome_render_view_observer.cc	2012-02-10 23:37:31.000000000 -0500
@@ -53,7 +53,7 @@
 #include "webkit/glue/image_decoder.h"
 #include "webkit/glue/image_resource_fetcher.h"
 #include "webkit/glue/webkit_glue.h"
-#include "v8/include/v8-testing.h"
+#include <v8-testing.h>
 
 using WebKit::WebAccessibilityObject;
 using WebKit::WebCString;
Index: chromium/src/chrome/renderer/extensions/chrome_webstore_bindings.cc
===================================================================
--- chromium.orig/src/chrome/renderer/extensions/chrome_webstore_bindings.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/extensions/chrome_webstore_bindings.cc	2012-02-10 23:37:31.000000000 -0500
@@ -16,7 +16,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNode.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNodeList.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDocument;
 using WebKit::WebElement;
Index: chromium/src/chrome/renderer/extensions/event_bindings.cc
===================================================================
--- chromium.orig/src/chrome/renderer/extensions/event_bindings.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/renderer/extensions/event_bindings.cc	2012-02-10 23:37:31.000000000 -0500
@@ -28,7 +28,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURLRequest.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebSecurityOrigin;
Index: chromium/src/chrome/test/base/v8_unit_test.h
===================================================================
--- chromium.orig/src/chrome/test/base/v8_unit_test.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/chrome/test/base/v8_unit_test.h	2012-02-10 23:37:31.000000000 -0500
@@ -11,7 +11,7 @@
 
 #include "base/file_path.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace base {
 class StringPiece;
Index: chromium/src/content/renderer/mock_content_renderer_client.cc
===================================================================
--- chromium.orig/src/content/renderer/mock_content_renderer_client.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/content/renderer/mock_content_renderer_client.cc	2012-02-10 23:37:31.000000000 -0500
@@ -5,7 +5,7 @@
 #include "content/renderer/mock_content_renderer_client.h"
 
 #include <string>
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace content {
 
Index: chromium/src/third_party/v8-i18n/build/all.gyp
===================================================================
--- chromium.orig/src/third_party/v8-i18n/build/all.gyp	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/build/all.gyp	2012-02-10 23:37:31.000000000 -0500
@@ -51,7 +51,7 @@
         '<(icu_path)/public/common',
       ],
       'dependencies': [
-        '<(v8_path)/v8/tools/gyp/v8.gyp:v8',
+        '<(v8_path)/build/linux/system.gyp:v8',
         '<(icu_path)/icu.gyp:*',
         'api2c#host',
       ],
Index: chromium/src/third_party/v8-i18n/include/extension.h
===================================================================
--- chromium.orig/src/third_party/v8-i18n/include/extension.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/include/extension.h	2012-02-10 23:37:31.000000000 -0500
@@ -15,7 +15,7 @@
 #ifndef V8_I18N_INCLUDE_EXTENSION_H_
 #define V8_I18N_INCLUDE_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace v8_i18n {
 
Index: chromium/src/third_party/v8-i18n/src/break-iterator.h
===================================================================
--- chromium.orig/src/third_party/v8-i18n/src/break-iterator.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/src/break-iterator.h	2012-02-10 23:37:31.000000000 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_BREAK_ITERATOR_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class BreakIterator;
Index: chromium/src/third_party/v8-i18n/src/collator.h
===================================================================
--- chromium.orig/src/third_party/v8-i18n/src/collator.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/src/collator.h	2012-02-10 23:37:31.000000000 -0500
@@ -29,7 +29,7 @@
 #define V8_I18N_SRC_COLLATOR_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class Collator;
Index: chromium/src/third_party/v8-i18n/src/datetime-format.h
===================================================================
--- chromium.orig/src/third_party/v8-i18n/src/datetime-format.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/src/datetime-format.h	2012-02-10 23:37:31.000000000 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_DATETIME_FORMAT_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class SimpleDateFormat;
Index: chromium/src/third_party/v8-i18n/src/language-matcher.h
===================================================================
--- chromium.orig/src/third_party/v8-i18n/src/language-matcher.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/src/language-matcher.h	2012-02-10 23:37:31.000000000 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_LANGUAGE_MATCHER_H_
 
 #include "unicode/uloc.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace v8_i18n {
 
Index: chromium/src/third_party/v8-i18n/src/number-format.h
===================================================================
--- chromium.orig/src/third_party/v8-i18n/src/number-format.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/src/number-format.h	2012-02-10 23:37:31.000000000 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_NUMBER_FORMAT_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class DecimalFormat;
Index: chromium/src/third_party/v8-i18n/src/utils.h
===================================================================
--- chromium.orig/src/third_party/v8-i18n/src/utils.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/third_party/v8-i18n/src/utils.h	2012-02-10 23:37:31.000000000 -0500
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_UTILS_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class UnicodeString;
Index: chromium/src/v8/src/extensions/experimental/break-iterator.h
===================================================================
--- chromium.orig/src/v8/src/extensions/experimental/break-iterator.h	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/v8/src/extensions/experimental/break-iterator.h	2012-02-10 23:37:31.000000000 -0500
@@ -28,7 +28,7 @@
 #ifndef V8_EXTENSIONS_EXPERIMENTAL_BREAK_ITERATOR_H_
 #define V8_EXTENSIONS_EXPERIMENTAL_BREAK_ITERATOR_H_
 
-#include "include/v8.h"
+#include <v8.h>
 
 #include "unicode/uversion.h"
 
Index: chromium/src/webkit/extensions/v8/gc_extension.cc
===================================================================
--- chromium.orig/src/webkit/extensions/v8/gc_extension.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/extensions/v8/gc_extension.cc	2012-02-10 23:37:31.000000000 -0500
@@ -4,7 +4,7 @@
 
 #include "webkit/extensions/v8/gc_extension.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 const char kGCExtensionName[] = "v8/GCController";
 
Index: chromium/src/webkit/extensions/v8/heap_profiler_extension.cc
===================================================================
--- chromium.orig/src/webkit/extensions/v8/heap_profiler_extension.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/extensions/v8/heap_profiler_extension.cc	2012-02-10 23:37:31.000000000 -0500
@@ -5,7 +5,7 @@
 #include "webkit/extensions/v8/heap_profiler_extension.h"
 
 #include "base/basictypes.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(USE_TCMALLOC) && !defined(OS_WIN)
 #include "third_party/tcmalloc/chromium/src/google/heap-profiler.h"
Index: chromium/src/webkit/glue/webkitplatformsupport_impl.cc
===================================================================
--- chromium.orig/src/webkit/glue/webkitplatformsupport_impl.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/glue/webkitplatformsupport_impl.cc	2012-02-10 23:37:31.000000000 -0500
@@ -46,7 +46,7 @@
 #include "webkit/plugins/webplugininfo.h"
 
 #if defined(OS_LINUX)
-#include "v8/include/v8.h"
+#include <v8.h>
 #endif
 
 using WebKit::WebAudioBus;
Index: chromium/src/webkit/plugins/ppapi/ppb_var_impl.cc
===================================================================
--- chromium.orig/src/webkit/plugins/ppapi/ppb_var_impl.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/plugins/ppapi/ppb_var_impl.cc	2012-02-11 00:20:47.000000000 -0500
@@ -17,7 +17,7 @@
 #include "webkit/plugins/ppapi/plugin_module.h"
 #include "webkit/plugins/ppapi/plugin_object.h"
 #include "webkit/plugins/ppapi/ppapi_plugin_instance.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using ppapi::NPObjectVar;
 using ppapi::PpapiGlobals;
Index: chromium/src/webkit/support/test_webkit_platform_support.cc
===================================================================
--- chromium.orig/src/webkit/support/test_webkit_platform_support.cc	2012-02-10 23:36:43.000000000 -0500
+++ chromium/src/webkit/support/test_webkit_platform_support.cc	2012-02-11 00:23:05.000000000 -0500
@@ -33,7 +33,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebStorageNamespace.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/appcache/web_application_cache_host_impl.h"
 #include "webkit/database/vfs_backend.h"
 #include "webkit/extensions/v8/gc_extension.h"
