Use system speex
Index: sid/src/build/linux/system.gyp
===================================================================
--- sid.orig/src/build/linux/system.gyp	2014-02-13 15:44:53.227430884 +0100
+++ sid/src/build/linux/system.gyp	2014-02-13 15:44:53.223430884 +0100
@@ -337,6 +337,26 @@
       ],
     },
     {
+      'target_name': 'speex',
+      'type': 'none',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(pkg-config --cflags speex)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other speex)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l speex)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
Index: sid/src/content/browser/speech/audio_encoder.cc
===================================================================
--- sid.orig/src/content/browser/speech/audio_encoder.cc	2014-02-13 15:44:53.227430884 +0100
+++ sid/src/content/browser/speech/audio_encoder.cc	2014-02-13 15:44:53.223430884 +0100
@@ -11,7 +11,7 @@
 #include "base/strings/string_number_conversions.h"
 #include "content/browser/speech/audio_buffer.h"
 #include "third_party/flac/include/FLAC/stream_encoder.h"
-#include "third_party/speex/include/speex/speex.h"
+#include <speex/speex.h>
 
 namespace content {
 namespace {
Index: sid/src/content/content_browser.gypi
===================================================================
--- sid.orig/src/content/content_browser.gypi	2014-02-13 15:44:53.227430884 +0100
+++ sid/src/content/content_browser.gypi	2014-02-13 15:44:53.223430884 +0100
@@ -1555,7 +1555,7 @@
     }, {  # input_speech==1
       'dependencies': [
         '../third_party/flac/flac.gyp:libflac',
-        '../third_party/speex/speex.gyp:libspeex',
+        '../build/linux/system.gyp:speex',
       ],
     }],
   ],
