--- a/build/linux/unbundle/replace_gn_files.py
+++ b/build/linux/unbundle/replace_gn_files.py
@@ -21,6 +21,7 @@
   'freetype': 'build/config/freetype/freetype.gni',
   'harfbuzz-ng': 'third_party/harfbuzz-ng/harfbuzz.gni',
   'icu': 'third_party/icu/BUILD.gn',
+  'jsoncpp' : 'third_party/jsoncpp/BUILD.gn',
   'libdrm': 'third_party/libdrm/BUILD.gn',
   'libevent': 'base/third_party/libevent/BUILD.gn',
   'libjpeg': 'third_party/libjpeg.gni',
--- a/third_party/webrtc/rtc_base/strings/json.h
+++ b/third_party/webrtc/rtc_base/strings/json.h
@@ -16,11 +16,7 @@
 
 #include "absl/strings/string_view.h"
 
-#if !defined(WEBRTC_EXTERNAL_JSON)
-#include "json/json.h"
-#else
 #include "third_party/jsoncpp/json.h"
-#endif
 
 namespace rtc {
 
