Use system speex
Index: chromium-browser/src/build/linux/system.gyp
===================================================================
--- chromium-browser.orig/src/build/linux/system.gyp	2013-03-30 15:03:35.028383135 +0000
+++ chromium-browser/src/build/linux/system.gyp	2013-03-30 15:03:35.024383136 +0000
@@ -619,6 +619,26 @@
       ],
     },
     {
+      'target_name': 'speex',
+      'type': 'none',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(pkg-config --cflags speex)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other speex)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l speex)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
Index: chromium-browser/src/content/browser/speech/audio_encoder.cc
===================================================================
--- chromium-browser.orig/src/content/browser/speech/audio_encoder.cc	2013-03-30 15:03:35.028383135 +0000
+++ chromium-browser/src/content/browser/speech/audio_encoder.cc	2013-03-30 15:06:23.812381682 +0000
@@ -11,7 +11,7 @@
 #include "base/string_number_conversions.h"
 #include "content/browser/speech/audio_buffer.h"
 #include "third_party/flac/include/FLAC/stream_encoder.h"
-#include "third_party/speex/include/speex/speex.h"
+#include <speex/speex.h>
 
 namespace content {
 namespace {
Index: chromium-browser/src/content/content_browser.gypi
===================================================================
--- chromium-browser.orig/src/content/content_browser.gypi	2013-03-30 15:03:35.028383135 +0000
+++ chromium-browser/src/content/content_browser.gypi	2013-03-30 15:03:35.028383135 +0000
@@ -1097,7 +1097,7 @@
     }, { # OS!="android"
       'dependencies': [
         '../third_party/flac/flac.gyp:libflac',
-        '../third_party/speex/speex.gyp:libspeex',
+        '../build/linux/system.gyp:speex',
       ],
     }],
     ['OS=="mac"', {
