--- a/src/third_party/WebKit/Source/WebCore/bindings/v8/V8DOMWindowShell.cpp
+++ b/src/third_party/WebKit/Source/WebCore/bindings/v8/V8DOMWindowShell.cpp
@@ -368,8 +368,8 @@ v8::Persistent<v8::Context> V8DOMWindowS
 
 #if ENABLE(JAVASCRIPT_I18N_API)
     // Enables experimental i18n API in V8.
-    if (RuntimeEnabledFeatures::javaScriptI18NAPIEnabled() && !V8Proxy::registeredExtensionWithV8(v8::internal::I18NExtension::get()))
-        V8Proxy::registerExtension(v8::internal::I18NExtension::get());
+    if (RuntimeEnabledFeatures::javaScriptI18NAPIEnabled() && !V8Proxy::registeredExtensionWithV8(v8_i18n::I18NExtension::get()))
+        V8Proxy::registerExtension(v8_i18n::I18NExtension::get());
 #endif
 
     // Dynamically tell v8 about our extensions now.
