Use system speex
--- a/src/build/linux/system.gyp
+++ b/src/build/linux/system.gyp
@@ -340,6 +340,26 @@
       }]]
     },
     {
+      'target_name': 'speex',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(pkg-config --cflags speex)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other speex)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l speex)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome-keyring',
       'type': 'settings',
       'conditions': [
--- a/src/content/browser/speech/audio_encoder.cc
+++ b/src/content/browser/speech/audio_encoder.cc
@@ -10,7 +10,7 @@
 #include "base/stl_util-inl.h"
 #include "base/string_number_conversions.h"
 #include "third_party/flac/flac.h"
-#include "third_party/speex/speex.h"
+#include <speex/speex.h>
 
 using std::string;
 
--- a/src/content/content_browser.gypi
+++ b/src/content/content_browser.gypi
@@ -15,7 +15,7 @@
         '../third_party/flac/flac.gyp:libflac',
         # TODO(ericu): remove leveldb ref after crbug.com/6955013 is fixed.
         '../third_party/leveldb/leveldb.gyp:leveldb',
-        '../third_party/speex/speex.gyp:libspeex',
+        '../build/linux/system.gyp:speex',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
         '../third_party/zlib/zlib.gyp:zlib',
         '../ui/ui.gyp:ui_base',
