Use system speex
Index: chromium/src/build/linux/system.gyp
===================================================================
--- chromium.orig/src/build/linux/system.gyp	2012-02-10 22:00:38.000000000 -0500
+++ chromium/src/build/linux/system.gyp	2012-02-10 22:25:22.000000000 -0500
@@ -380,6 +380,26 @@
       ],
     },
     {
+      'target_name': 'speex',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(pkg-config --cflags speex)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other speex)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l speex)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
Index: chromium/src/content/browser/speech/audio_encoder.cc
===================================================================
--- chromium.orig/src/content/browser/speech/audio_encoder.cc	2012-02-10 22:00:39.000000000 -0500
+++ chromium/src/content/browser/speech/audio_encoder.cc	2012-02-10 22:25:22.000000000 -0500
@@ -10,7 +10,7 @@
 #include "base/stl_util.h"
 #include "base/string_number_conversions.h"
 #include "third_party/flac/flac.h"
-#include "third_party/speex/speex.h"
+#include <speex/speex.h>
 
 using std::string;
 
Index: chromium/src/content/content_browser.gypi
===================================================================
--- chromium.orig/src/content/content_browser.gypi	2012-02-10 22:00:39.000000000 -0500
+++ chromium/src/content/content_browser.gypi	2012-02-10 22:25:41.000000000 -0500
@@ -10,7 +10,7 @@
     '../ppapi/ppapi_internal.gyp:ppapi_proxy',
     '../skia/skia.gyp:skia',
     '../third_party/flac/flac.gyp:libflac',
-    '../third_party/speex/speex.gyp:libspeex',
+    '../build/linux/system.gyp:speex',
     '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
     '../third_party/zlib/zlib.gyp:zlib',
     '../ui/gfx/surface/surface.gyp:surface',
