Use system copy of libv8
--- a/src/build/linux/system.gyp
+++ b/src/build/linux/system.gyp
@@ -322,6 +322,18 @@
       }]]
     },
     {
+      'target_name': 'v8',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'libraries': [
+              '-lv8',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome-keyring',
       'type': 'settings',
       'conditions': [
--- a/src/net/net.gyp
+++ b/src/net/net.gyp
@@ -744,7 +744,7 @@
       'conditions': [
         ['javascript_engine=="v8"', {
           'dependencies': [
-            '../v8/tools/gyp/v8.gyp:v8',
+            '../build/linux/system.gyp:v8',
           ],
         }],
         ['chromeos==1', {
--- a/src/build/all.gyp
+++ b/src/build/all.gyp
@@ -54,7 +54,7 @@
       'conditions': [
         ['javascript_engine=="v8"', {
           'dependencies': [
-            '../v8/tools/gyp/v8.gyp:*',
+            'linux/system.gyp:v8',
           ],
         }],
         ['OS=="mac" or OS=="linux" or OS=="freebsd" or OS=="openbsd"', {
--- a/src/chrome/browser/browser_about_handler.cc
+++ b/src/chrome/browser/browser_about_handler.cc
@@ -59,7 +59,7 @@
 #include "ui/base/resource/resource_bundle.h"
 #include "webkit/glue/webkit_glue.h"
 #ifdef CHROME_V8
-#include "v8/include/v8.h"
+#include <v8.h>
 #endif
 
 #if defined(OS_WIN)
--- a/src/chrome/browser/memory_purger.cc
+++ b/src/chrome/browser/memory_purger.cc
@@ -23,7 +23,7 @@
 #include "net/proxy/proxy_resolver.h"
 #include "net/url_request/url_request_context.h"
 #include "third_party/tcmalloc/chromium/src/google/malloc_extension.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // PurgeMemoryHelper -----------------------------------------------------------
 
--- a/src/chrome/renderer/automation/dom_automation_v8_extension.h
+++ b/src/chrome/renderer/automation/dom_automation_v8_extension.h
@@ -6,7 +6,7 @@
 #define CHROME_RENDERER_AUTOMATION_DOM_AUTOMATION_V8_EXTENSION_H_
 #pragma once
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class DomAutomationV8Extension {
  public:
--- a/src/chrome/renderer/extensions/bindings_utils.h
+++ b/src/chrome/renderer/extensions/bindings_utils.h
@@ -10,7 +10,7 @@
 #include "base/singleton.h"
 #include "base/string_piece.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <list>
 #include <string>
--- a/src/chrome/renderer/extensions/event_bindings.h
+++ b/src/chrome/renderer/extensions/event_bindings.h
@@ -8,7 +8,7 @@
 
 #include <string>
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class GURL;
 class RenderThreadBase;
--- a/src/chrome/renderer/extensions/extension_process_bindings.h
+++ b/src/chrome/renderer/extensions/extension_process_bindings.h
@@ -13,7 +13,7 @@
 #include <vector>
 
 #include "chrome/common/view_types.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class GURL;
 class URLPattern;
--- a/src/chrome/renderer/extensions/js_only_v8_extensions.h
+++ b/src/chrome/renderer/extensions/js_only_v8_extensions.h
@@ -6,7 +6,7 @@
 #define CHROME_RENDERER_EXTENSIONS_JS_ONLY_V8_EXTENSIONS_H_
 #pragma once
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // This file contains various V8 Extensions that are JavaScript only, and
 // don't have any C++ native functions.
--- a/src/chrome/renderer/extensions/renderer_extension_bindings.h
+++ b/src/chrome/renderer/extensions/renderer_extension_bindings.h
@@ -6,7 +6,7 @@
 #define CHROME_RENDERER_EXTENSIONS_RENDERER_EXTENSION_BINDINGS_H_
 #pragma once
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <string>
 
--- a/src/chrome/renderer/loadtimes_extension_bindings.cc
+++ b/src/chrome/renderer/loadtimes_extension_bindings.cc
@@ -9,7 +9,7 @@
 #include "base/time.h"
 #include "chrome/renderer/navigation_state.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDataSource;
 using WebKit::WebFrame;
--- a/src/chrome/renderer/render_thread.cc
+++ b/src/chrome/renderer/render_thread.cc
@@ -93,7 +93,7 @@
 #include "webkit/extensions/v8/gears_extension.h"
 #include "webkit/extensions/v8/playback_extension.h"
 #include "webkit/glue/webkit_glue.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // TODO(port)
 #if defined(OS_WIN)
--- a/src/chrome/test/v8_unit_test.h
+++ b/src/chrome/test/v8_unit_test.h
@@ -9,7 +9,7 @@
 #include <string>
 
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace base {
 class StringPiece;
--- a/src/net/proxy/proxy_resolver_v8.cc
+++ b/src/net/proxy/proxy_resolver_v8.cc
@@ -22,7 +22,7 @@
 #include "net/proxy/proxy_resolver_js_bindings.h"
 #include "net/proxy/proxy_resolver_request_context.h"
 #include "net/proxy/proxy_resolver_script.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // Notes on the javascript environment:
 //
--- a/src/webkit/extensions/v8/benchmarking_extension.h
+++ b/src/webkit/extensions/v8/benchmarking_extension.h
@@ -9,7 +9,7 @@
 #ifndef WEBKIT_EXTENSIONS_V8_BENCHMARKING_EXTENSION_H_
 #define WEBKIT_EXTENSIONS_V8_BENCHMARKING_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions_v8 {
 
--- a/src/webkit/extensions/v8/gc_extension.h
+++ b/src/webkit/extensions/v8/gc_extension.h
@@ -8,7 +8,7 @@
 #ifndef WEBKIT_EXTENSIONS_V8_GC_EXTENSION_H_
 #define WEBKIT_EXTENSIONS_V8_GC_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions_v8 {
 
--- a/src/webkit/extensions/v8/gears_extension.h
+++ b/src/webkit/extensions/v8/gears_extension.h
@@ -9,7 +9,7 @@
 #ifndef WEBKIT_EXTENSIONS_V8_GEARS_EXTENSION_H_
 #define WEBKIT_EXTENSIONS_V8_GEARS_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions_v8 {
 
--- a/src/webkit/extensions/v8/heap_profiler_extension.h
+++ b/src/webkit/extensions/v8/heap_profiler_extension.h
@@ -9,7 +9,7 @@
 #ifndef WEBKIT_EXTENSIONS_V8_HEAP_PROFILER_EXTENSION_H_
 #define WEBKIT_EXTENSIONS_V8_HEAP_PROFILER_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions_v8 {
 
--- a/src/webkit/extensions/v8/playback_extension.h
+++ b/src/webkit/extensions/v8/playback_extension.h
@@ -5,7 +5,7 @@
 #ifndef WEBKIT_EXTENSIONS_V8_PLAYBACK_EXTENSION_H_
 #define WEBKIT_EXTENSIONS_V8_PLAYBACK_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions_v8 {
 
--- a/src/webkit/extensions/v8/profiler_extension.h
+++ b/src/webkit/extensions/v8/profiler_extension.h
@@ -9,7 +9,7 @@
 #ifndef WEBKIT_EXTENSIONS_V8_PROFILER_EXTENSION_H_
 #define WEBKIT_EXTENSIONS_V8_PROFILER_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions_v8 {
 
--- a/src/webkit/glue/webkit_glue.cc
+++ b/src/webkit/glue/webkit_glue.cc
@@ -44,7 +44,7 @@
 #endif
 #include "webkit/glue/glue_serialize.h"
 #include "webkit/glue/user_agent.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebCanvas;
 using WebKit::WebData;
--- a/src/webkit/glue/webkitclient_impl.cc
+++ b/src/webkit/glue/webkitclient_impl.cc
@@ -44,7 +44,7 @@
 #include "webkit/glue/weburlloader_impl.h"
 
 #if defined(OS_LINUX)
-#include "v8/include/v8.h"
+#include <v8.h>
 #endif
 
 using WebKit::WebAudioBus;
--- a/src/webkit/support/test_webkit_client.cc
+++ b/src/webkit/support/test_webkit_client.cc
@@ -49,7 +49,7 @@
 #include "webkit/tools/test_shell/simple_webcookiejar_impl.h"
 #include "webkit/tools/test_shell/test_shell_request_context.h"
 #include "webkit/tools/test_shell/test_shell_webblobregistry_impl.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(OS_WIN)
 #include "third_party/WebKit/Source/WebKit/chromium/public/win/WebThemeEngine.h"
--- a/src/webkit/tools/test_shell/listener_leak_test.cc
+++ b/src/webkit/tools/test_shell/listener_leak_test.cc
@@ -8,8 +8,8 @@
 #include "base/string_number_conversions.h"
 #include "base/string_util.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
-#include "v8/include/v8-profiler.h"
+#include <v8.h>
+#include <v8-profiler.h>
 #include "webkit/glue/webkit_glue.h"
 #include "webkit/tools/test_shell/test_shell.h"
 #include "webkit/tools/test_shell/test_shell_test.h"
--- a/src/chrome/renderer/extensions/chrome_app_bindings.cc
+++ b/src/chrome/renderer/extensions/chrome_app_bindings.cc
@@ -11,7 +11,7 @@
 #include "chrome/renderer/render_thread.h"
 #include "chrome/renderer/render_view.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 
--- a/src/chrome/renderer/external_extension.cc
+++ b/src/chrome/renderer/external_extension.cc
@@ -9,7 +9,7 @@
 #include "chrome/common/render_messages_params.h"
 #include "chrome/renderer/render_view.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebView;
--- a/src/chrome/renderer/search_extension.cc
+++ b/src/chrome/renderer/search_extension.cc
@@ -12,7 +12,7 @@
 #include "chrome/renderer/render_view.h"
 #include "chrome/renderer/searchbox.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebView;
--- a/src/chrome/renderer/translate_helper.cc
+++ b/src/chrome/renderer/translate_helper.cc
@@ -13,7 +13,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/dom_operations.h"
 
 using WebKit::WebDocument;
--- a/src/webkit/glue/idb_bindings.cc
+++ b/src/webkit/glue/idb_bindings.cc
@@ -12,7 +12,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebIDBKeyPath.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebSerializedScriptValue.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebString.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace webkit_glue {
 
--- a/src/webkit/glue/webkit_glue.gypi
+++ b/src/webkit/glue/webkit_glue.gypi
@@ -532,11 +532,11 @@
             ['inside_chromium_build==1 and component=="shared_library"', {
               'dependencies': [
                 '<(DEPTH)/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
-                '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+                '<(DEPTH)/build/linux/system.gyp:v8',
                ],
                'export_dependent_settings': [
                  '<(DEPTH)/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
-                 '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+                 '<(DEPTH)/build/linux/system.gyp:v8',
                ],
             }],
           ],
--- a/src/chrome/renderer/searchbox_extension.cc
+++ b/src/chrome/renderer/searchbox_extension.cc
@@ -16,7 +16,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebString.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebScriptSource;
--- a/src/v8/tools/gyp/v8.gyp
+++ b/src/v8/tools/gyp/v8.gyp
@@ -832,7 +832,7 @@
           'target_name': 'v8_shell',
           'type': 'executable',
           'dependencies': [
-            'v8'
+            '../../../build/linux/system.gyp:v8',
           ],
           'sources': [
             '../../samples/shell.cc',
--- a/src/chrome/renderer/render_view.cc
+++ b/src/chrome/renderer/render_view.cc
@@ -174,8 +174,8 @@
 #include "ui/gfx/point.h"
 #include "ui/gfx/rect.h"
 #include "ui/gfx/skbitmap_operations.h"
-#include "v8/include/v8-testing.h"
-#include "v8/include/v8.h"
+#include <v8-testing.h>
+#include <v8.h>
 #include "webkit/appcache/web_application_cache_host_impl.h"
 #include "webkit/glue/alt_error_page_resource_fetcher.h"
 #include "webkit/glue/context_menu.h"
--- a/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
+++ b/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
@@ -849,7 +849,7 @@
       'conditions': [
         ['javascript_engine=="v8"', {
           'dependencies': [
-            '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+            '<(chromium_src_dir)/build/linux/system.gyp:v8',
           ],
           'conditions': [
             ['inside_chromium_build==1 and OS=="win" and component=="shared_library"', {
@@ -969,10 +969,10 @@
       'conditions': [
         ['javascript_engine=="v8"', {
           'dependencies': [
-            '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+            '<(chromium_src_dir)/build/linux/system.gyp:v8',
           ],
           'export_dependent_settings': [
-            '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+            '<(chromium_src_dir)/build/linux/system.gyp:v8',
           ],
           'conditions': [
             ['inside_chromium_build==1 and OS=="win" and component=="shared_library"', {
@@ -1595,10 +1595,10 @@
       'conditions': [
         ['javascript_engine=="v8"', {
           'dependencies': [
-            '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+            '<(chromium_src_dir)/build/linux/system.gyp:v8',
           ],
           'export_dependent_settings': [
-            '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+            '<(chromium_src_dir)/build/linux/system.gyp:v8',
           ],
         }],
         ['OS=="mac"', {
--- a/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp
+++ b/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp
@@ -32,8 +32,8 @@
 
 #include "TestShell.h"
 #include "webkit/support/webkit_support.h"
-#include <v8/include/v8-testing.h>
-#include <v8/include/v8.h>
+#include <v8-testing.h>
+#include <v8.h>
 #include <wtf/Vector.h>
 
 using namespace std;
--- a/src/webkit/mocks/mock_webframe.h
+++ b/src/webkit/mocks/mock_webframe.h
@@ -16,7 +16,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebSecurityOrigin.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebSize.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebString.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebAnimationController;
 using WebKit::WebCanvas;
--- a/src/webkit/plugins/ppapi/var.cc
+++ b/src/webkit/plugins/ppapi/var.cc
@@ -18,7 +18,7 @@
 #include "webkit/plugins/ppapi/plugin_object.h"
 #include "webkit/plugins/ppapi/ppapi_plugin_instance.h"
 #include "webkit/plugins/ppapi/resource_tracker.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebBindings;
 
--- a/src/chrome/browser/ui/webui/options/about_page_handler.cc
+++ b/src/chrome/browser/ui/webui/options/about_page_handler.cc
@@ -29,7 +29,7 @@
 #include "webkit/glue/webkit_glue.h"
 
 #if defined(CHROME_V8)
-#include "v8/include/v8.h"
+#include <v8.h>
 #endif
 
 #if defined(OS_CHROMEOS)
