Use system copy of libv8
Index: exp/src/build/linux/system.gyp
===================================================================
--- exp.orig/src/build/linux/system.gyp	2012-05-19 19:23:03.000000000 +0200
+++ exp/src/build/linux/system.gyp	2012-05-19 19:23:29.330671562 +0200
@@ -404,6 +404,18 @@
       }]]
     },
     {
+      'target_name': 'v8',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'libraries': [
+              '-lv8',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
Index: exp/src/net/net.gyp
===================================================================
--- exp.orig/src/net/net.gyp	2012-05-19 16:54:19.000000000 +0200
+++ exp/src/net/net.gyp	2012-05-19 19:23:29.330671562 +0200
@@ -35,7 +35,7 @@
         '../third_party/icu/icu.gyp:icui18n',
         '../third_party/icu/icu.gyp:icuuc',
         '../third_party/zlib/zlib.gyp:zlib',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         'net_resources',
       ],
       'sources': [
Index: exp/src/build/all.gyp
===================================================================
--- exp.orig/src/build/all.gyp	2012-05-19 16:54:39.000000000 +0200
+++ exp/src/build/all.gyp	2012-05-19 19:23:29.330671562 +0200
@@ -52,7 +52,7 @@
         '../third_party/WebKit/Source/WebKit/chromium/All.gyp:*',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:generate_devtools_zip',
         '../third_party/zlib/zlib.gyp:*',
-        '../v8/tools/gyp/v8.gyp:*',
+        '../build/linux/system.gyp:v8',
         '../webkit/support/webkit_support.gyp:*',
         '../webkit/webkit.gyp:*',
         'util/build_util.gyp:*',
Index: exp/src/chrome/renderer/loadtimes_extension_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/loadtimes_extension_bindings.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/loadtimes_extension_bindings.cc	2012-05-19 19:23:29.330671562 +0200
@@ -9,7 +9,7 @@
 #include "base/time.h"
 #include "content/public/renderer/document_state.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDataSource;
 using WebKit::WebFrame;
Index: exp/src/net/proxy/proxy_resolver_v8.cc
===================================================================
--- exp.orig/src/net/proxy/proxy_resolver_v8.cc	2012-05-19 16:54:17.000000000 +0200
+++ exp/src/net/proxy/proxy_resolver_v8.cc	2012-05-19 19:23:29.330671562 +0200
@@ -24,7 +24,7 @@
 #include "net/proxy/proxy_resolver_js_bindings.h"
 #include "net/proxy/proxy_resolver_request_context.h"
 #include "net/proxy/proxy_resolver_script.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // Notes on the javascript environment:
 //
Index: exp/src/webkit/glue/webkit_glue.cc
===================================================================
--- exp.orig/src/webkit/glue/webkit_glue.cc	2012-05-19 16:54:05.000000000 +0200
+++ exp/src/webkit/glue/webkit_glue.cc	2012-05-19 19:23:29.330671562 +0200
@@ -45,7 +45,7 @@
 #if defined(OS_WIN)
 #include "third_party/WebKit/Source/WebKit/chromium/public/win/WebInputEventFactory.h"
 #endif
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/glue_serialize.h"
 #include "webkit/glue/user_agent.h"
 
Index: exp/src/webkit/tools/test_shell/listener_leak_test.cc
===================================================================
--- exp.orig/src/webkit/tools/test_shell/listener_leak_test.cc	2012-05-19 16:54:06.000000000 +0200
+++ exp/src/webkit/tools/test_shell/listener_leak_test.cc	2012-05-19 19:23:29.330671562 +0200
@@ -8,8 +8,8 @@
 #include "base/string_number_conversions.h"
 #include "base/string_util.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
-#include "v8/include/v8-profiler.h"
+#include <v8.h>
+#include <v8-profiler.h>
 #include "webkit/glue/webkit_glue.h"
 #include "webkit/tools/test_shell/test_shell.h"
 #include "webkit/tools/test_shell/test_shell_test.h"
Index: exp/src/chrome/renderer/search_extension.cc
===================================================================
--- exp.orig/src/chrome/renderer/search_extension.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/search_extension.cc	2012-05-19 19:23:29.330671562 +0200
@@ -11,7 +11,7 @@
 #include "chrome/renderer/searchbox.h"
 #include "content/public/renderer/render_view.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebView;
Index: exp/src/webkit/glue/webkit_glue.gypi
===================================================================
--- exp.orig/src/webkit/glue/webkit_glue.gypi	2012-05-19 16:54:05.000000000 +0200
+++ exp/src/webkit/glue/webkit_glue.gypi	2012-05-19 19:23:29.330671562 +0200
@@ -132,7 +132,7 @@
         '<(DEPTH)/third_party/npapi/npapi.gyp:npapi',
         '<(DEPTH)/ui/gfx/gl/gl.gyp:gl',
         '<(DEPTH)/ui/ui.gyp:ui',
-        '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+        '<(DEPTH)/build/linux/system.gyp:v8',
         '<(webkit_src_dir)/Source/WebKit/chromium/WebKit.gyp:webkit',
         'fileapi',
         'webkit_media',
@@ -527,7 +527,7 @@
                ],
                'export_dependent_settings': [
                  '<(webkit_src_dir)/Source/WebKit/chromium/WebKit.gyp:webkit',
-                 '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+                 '<(DEPTH)/build/linux/system.gyp:v8',
                ],
             }],
           ],
Index: exp/src/chrome/renderer/searchbox_extension.cc
===================================================================
--- exp.orig/src/chrome/renderer/searchbox_extension.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/searchbox_extension.cc	2012-05-19 19:23:29.330671562 +0200
@@ -15,7 +15,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebScriptSource;
Index: exp/src/v8/tools/gyp/v8.gyp
===================================================================
--- exp.orig/src/v8/tools/gyp/v8.gyp	2012-05-19 16:57:28.000000000 +0200
+++ exp/src/v8/tools/gyp/v8.gyp	2012-05-19 19:23:29.330671562 +0200
@@ -1006,7 +1006,7 @@
             }],
           ],
           'dependencies': [
-            'v8'
+            '../../../build/linux/system.gyp:v8',
           ],
         },
       ],
Index: exp/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
===================================================================
--- exp.orig/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp	2012-05-19 16:58:47.000000000 +0200
+++ exp/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp	2012-05-19 19:23:29.330671562 +0200
@@ -1085,7 +1085,7 @@
         '<(chromium_src_dir)/third_party/libwebp/libwebp.gyp:libwebp',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'include_dirs': [
@@ -1211,7 +1211,7 @@
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
         '<(chromium_src_dir)/third_party/zlib/zlib.gyp:zlib',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'export_dependent_settings': [
@@ -1229,7 +1229,7 @@
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
         '<(chromium_src_dir)/third_party/zlib/zlib.gyp:zlib',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       # This is needed for mac because of webkit_system_interface. It'd be nice
@@ -1980,7 +1980,7 @@
         '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'export_dependent_settings': [
         'webcore_bindings',
@@ -1988,7 +1988,7 @@
         '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
Index: exp/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp
===================================================================
--- exp.orig/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp	2012-05-19 16:56:15.000000000 +0200
+++ exp/src/third_party/WebKit/Tools/DumpRenderTree/chromium/DumpRenderTree.cpp	2012-05-19 19:23:29.330671562 +0200
@@ -33,8 +33,8 @@
 #include "TestShell.h"
 #include "WebCompositor.h"
 #include "webkit/support/webkit_support.h"
-#include <v8/include/v8-testing.h>
-#include <v8/include/v8.h>
+#include <v8-testing.h>
+#include <v8.h>
 #include <wtf/Vector.h>
 
 using namespace std;
Index: exp/src/chrome/browser/task_manager/task_manager_resource_providers.cc
===================================================================
--- exp.orig/src/chrome/browser/task_manager/task_manager_resource_providers.cc	2012-05-19 16:54:55.000000000 +0200
+++ exp/src/chrome/browser/task_manager/task_manager_resource_providers.cc	2012-05-19 19:23:29.334671559 +0200
@@ -58,7 +58,7 @@
 #include "third_party/sqlite/sqlite3.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(OS_MACOSX)
 #include "skia/ext/skia_utils_mac.h"
Index: exp/src/chrome/chrome_tests.gypi
===================================================================
--- exp.orig/src/chrome/chrome_tests.gypi	2012-05-19 16:55:35.000000000 +0200
+++ exp/src/chrome/chrome_tests.gypi	2012-05-19 19:23:29.334671559 +0200
@@ -999,7 +999,7 @@
         '../ui/ui.gyp:ui_resources',
         '../ui/ui.gyp:ui_resources_standard',
         '../ui/ui.gyp:ui_test_support',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         'common/extensions/api/api.gyp:api',
         'chrome_resources.gyp:chrome_resources',
         'chrome_resources.gyp:chrome_strings',
@@ -2046,7 +2046,7 @@
         ['target_arch!="arm"', {
           'dependencies': [
             # build time dependency.
-            '../v8/tools/gyp/v8.gyp:v8_shell#host',
+            '../build/linux/system.gyp:v8',
           ],
         }],
         ['enable_extensions==0', {
@@ -2604,7 +2604,7 @@
         '../third_party/icu/icu.gyp:icui18n',
         '../third_party/icu/icu.gyp:icuuc',
         '../third_party/leveldatabase/leveldatabase.gyp:leveldatabase',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         '../webkit/webkit.gyp:test_shell_test_support',
         # Runtime dependencies
         '../ppapi/ppapi_internal.gyp:ppapi_tests',
@@ -3324,7 +3324,7 @@
         ['target_arch!="arm"', {
           'dependencies': [
             # build time dependency.
-            '../v8/tools/gyp/v8.gyp:v8_shell#host',
+            '../build/linux/system.gyp:v8',
             # run time dependency
             '../webkit/webkit.gyp:copy_npapi_test_plugin',
           ],
@@ -3409,7 +3409,7 @@
         '../third_party/icu/icu.gyp:icui18n',
         '../third_party/icu/icu.gyp:icuuc',
         '../third_party/leveldatabase/leveldatabase.gyp:leveldatabase',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         '../webkit/webkit.gyp:test_shell_test_support',
         # Runtime dependencies
         '../third_party/mesa/mesa.gyp:osmesa',
Index: exp/src/chrome/renderer/chrome_render_process_observer.cc
===================================================================
--- exp.orig/src/chrome/renderer/chrome_render_process_observer.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/chrome_render_process_observer.cc	2012-05-19 19:23:29.334671559 +0200
@@ -43,7 +43,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebRuntimeFeatures.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #if defined(OS_WIN)
 #include "base/win/iat_patch_function.h"
Index: exp/src/chrome/renderer/extensions/extension_dispatcher.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/extension_dispatcher.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/extension_dispatcher.cc	2012-05-19 19:23:29.334671559 +0200
@@ -57,7 +57,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURLRequest.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDataSource;
 using WebKit::WebDocument;
Index: exp/src/chrome/renderer/external_extension.cc
===================================================================
--- exp.orig/src/chrome/renderer/external_extension.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/external_extension.cc	2012-05-19 19:23:29.334671559 +0200
@@ -12,7 +12,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebView;
Index: exp/src/chrome/renderer/translate_helper.cc
===================================================================
--- exp.orig/src/chrome/renderer/translate_helper.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/translate_helper.cc	2012-05-19 19:23:29.334671559 +0200
@@ -19,7 +19,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptSource.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "third_party/cld/encodings/compact_lang_det/win/cld_unicodetext.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/dom_operations.h"
 
 using WebKit::WebDocument;
Index: exp/src/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp
===================================================================
--- exp.orig/src/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp	2012-05-19 16:56:54.000000000 +0200
+++ exp/src/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp	2012-05-19 19:23:29.334671559 +0200
@@ -73,13 +73,13 @@
                 '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
                 '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
                 '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
             ],
             'export_dependent_settings': [
                 '<(chromium_src_dir)/skia/skia.gyp:skia',
                 '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
                 '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
             ],
             'include_dirs': [
                 'public',
@@ -708,7 +708,7 @@
                                 '<(chromium_src_dir)/third_party/modp_b64/modp_b64.gyp:modp_b64',
                                 '<(chromium_src_dir)/third_party/ots/ots.gyp:ots',
                                 '<(chromium_src_dir)/third_party/zlib/zlib.gyp:zlib',
-                                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                                 # We must not add webkit_support here because of cyclic dependency.
                             ],
                             'direct_dependent_settings': {
@@ -718,7 +718,7 @@
                             },
                             'export_dependent_settings': [
                                 '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
-                                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                             ],
                             'include_dirs': [
                                 # WARNING: Do not view this particular case as a precedent for
Index: exp/src/webkit/tools/test_shell/test_shell_webkit_init.cc
===================================================================
--- exp.orig/src/webkit/tools/test_shell/test_shell_webkit_init.cc	2012-05-19 16:54:06.000000000 +0200
+++ exp/src/webkit/tools/test_shell/test_shell_webkit_init.cc	2012-05-19 19:23:29.334671559 +0200
@@ -14,7 +14,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScriptController.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebSecurityPolicy.h"
 #include "ui/gfx/gl/gl_bindings_skia_in_process.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/plugins/npapi/plugin_list.h"
 #include "webkit/plugins/webplugininfo.h"
 #include "webkit/tools/test_shell/simple_socket_stream_bridge.h"
Index: exp/src/content/content_renderer.gypi
===================================================================
--- exp.orig/src/content/content_renderer.gypi	2012-05-19 16:54:14.000000000 +0200
+++ exp/src/content/content_renderer.gypi	2012-05-19 19:30:49.966676288 +0200
@@ -17,7 +17,7 @@
     '../third_party/npapi/npapi.gyp:npapi',
     '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
     '../ui/surface/surface.gyp:surface',
-    '../v8/tools/gyp/v8.gyp:v8',
+    '../build/linux/system.gyp:v8',
     '../webkit/support/webkit_support.gyp:webkit_media',
     '../webkit/support/webkit_support.gyp:webkit_gpu',
   ],
Index: exp/src/third_party/WebKit/Source/ThirdParty/gyp/samples/samples
===================================================================
--- exp.orig/src/third_party/WebKit/Source/ThirdParty/gyp/samples/samples	2012-05-19 17:00:22.000000000 +0200
+++ exp/src/third_party/WebKit/Source/ThirdParty/gyp/samples/samples	2012-05-19 19:23:29.334671559 +0200
@@ -38,7 +38,7 @@
     'third_party/npapi/npapi.gyp',
     'third_party/sqlite/sqlite.gyp',
     'third_party/zlib/zlib.gyp',
-    'v8/tools/gyp/v8.gyp',
+    'build/linux/system.gyp:v8',
     'webkit/activex_shim/activex_shim.gyp',
     'webkit/activex_shim_dll/activex_shim_dll.gyp',
     'webkit/build/action_csspropertynames.py',
Index: exp/src/webkit/tools/test_shell/test_shell.gypi
===================================================================
--- exp.orig/src/webkit/tools/test_shell/test_shell.gypi	2012-05-19 16:54:06.000000000 +0200
+++ exp/src/webkit/tools/test_shell/test_shell.gypi	2012-05-19 19:23:29.334671559 +0200
@@ -46,7 +46,7 @@
         '<(DEPTH)/testing/gtest.gyp:gtest',
         '<(DEPTH)/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:inspector_resources',
         '<(DEPTH)/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
-        '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+        '<(DEPTH)/build/linux/system.gyp:v8',
         '<(DEPTH)/webkit/support/webkit_support.gyp:appcache',
         '<(DEPTH)/webkit/support/webkit_support.gyp:blob',
         '<(DEPTH)/webkit/support/webkit_support.gyp:database',
@@ -366,7 +366,7 @@
         '<(DEPTH)/testing/gmock.gyp:gmock',
         '<(DEPTH)/testing/gtest.gyp:gtest',
         '<(DEPTH)/third_party/leveldatabase/leveldatabase.gyp:leveldatabase',
-        '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
+        '<(DEPTH)/build/linux/system.gyp:v8',
         '<(DEPTH)/webkit/support/webkit_support.gyp:webkit_user_agent',
       ],
       'sources': [
Index: exp/src/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp
===================================================================
--- exp.orig/src/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp	2012-05-19 16:56:48.000000000 +0200
+++ exp/src/third_party/WebKit/Source/WebKit/chromium/src/WebTestingSupport.cpp	2012-05-19 19:23:29.334671559 +0200
@@ -28,7 +28,7 @@
 
 #include "WebCoreTestSupport.h"
 #include "WebFrame.h"
-#include <v8/include/v8.h>
+#include <v8.h>
 
 namespace WebKit {
 
Index: exp/src/chrome/renderer/chrome_render_view_observer.cc
===================================================================
--- exp.orig/src/chrome/renderer/chrome_render_view_observer.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/chrome_render_view_observer.cc	2012-05-19 19:23:29.334671559 +0200
@@ -52,7 +52,7 @@
 #include "webkit/glue/image_decoder.h"
 #include "webkit/glue/image_resource_fetcher.h"
 #include "webkit/glue/webkit_glue.h"
-#include "v8/include/v8-testing.h"
+#include <v8-testing.h>
 
 using WebKit::WebAccessibilityObject;
 using WebKit::WebCString;
Index: exp/src/chrome/renderer/extensions/event_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/event_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/event_bindings.cc	2012-05-19 19:23:29.334671559 +0200
@@ -29,7 +29,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURLRequest.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using WebKit::WebSecurityOrigin;
Index: exp/src/chrome/test/base/v8_unit_test.h
===================================================================
--- exp.orig/src/chrome/test/base/v8_unit_test.h	2012-05-19 16:55:30.000000000 +0200
+++ exp/src/chrome/test/base/v8_unit_test.h	2012-05-19 19:23:29.334671559 +0200
@@ -12,7 +12,7 @@
 #include "base/file_path.h"
 #include "base/string_piece.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // A superclass for unit tests that involve running JavaScript.  This class
 // sets up V8 context and has methods that make it easy to execute scripts in
Index: exp/src/content/renderer/mock_content_renderer_client.cc
===================================================================
--- exp.orig/src/content/renderer/mock_content_renderer_client.cc	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/renderer/mock_content_renderer_client.cc	2012-05-19 19:23:29.334671559 +0200
@@ -5,7 +5,7 @@
 #include "content/renderer/mock_content_renderer_client.h"
 
 #include <string>
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace content {
 
Index: exp/src/third_party/v8-i18n/build/all.gyp
===================================================================
--- exp.orig/src/third_party/v8-i18n/build/all.gyp	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/build/all.gyp	2012-05-19 19:23:29.334671559 +0200
@@ -50,7 +50,7 @@
         '<(v8_path)',
       ],
       'dependencies': [
-        '<(v8_path)/v8/tools/gyp/v8.gyp:v8',
+        '<(v8_path)/build/linux/system.gyp:v8',
         '<(icu_path)/icu.gyp:*',
         'api2c#host',
       ],
Index: exp/src/third_party/v8-i18n/include/extension.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/include/extension.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/include/extension.h	2012-05-19 19:23:29.338671560 +0200
@@ -15,7 +15,7 @@
 #ifndef V8_I18N_INCLUDE_EXTENSION_H_
 #define V8_I18N_INCLUDE_EXTENSION_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace v8_i18n {
 
Index: exp/src/third_party/v8-i18n/src/break-iterator.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/src/break-iterator.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/src/break-iterator.h	2012-05-19 19:23:29.338671560 +0200
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_BREAK_ITERATOR_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class BreakIterator;
Index: exp/src/third_party/v8-i18n/src/collator.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/src/collator.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/src/collator.h	2012-05-19 19:23:29.338671560 +0200
@@ -29,7 +29,7 @@
 #define V8_I18N_SRC_COLLATOR_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class Collator;
Index: exp/src/third_party/v8-i18n/src/datetime-format.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/src/datetime-format.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/src/datetime-format.h	2012-05-19 19:23:29.338671560 +0200
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_DATETIME_FORMAT_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class SimpleDateFormat;
Index: exp/src/third_party/v8-i18n/src/language-matcher.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/src/language-matcher.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/src/language-matcher.h	2012-05-19 19:23:29.338671560 +0200
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_LANGUAGE_MATCHER_H_
 
 #include "unicode/uloc.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace v8_i18n {
 
Index: exp/src/third_party/v8-i18n/src/number-format.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/src/number-format.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/src/number-format.h	2012-05-19 19:23:29.338671560 +0200
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_NUMBER_FORMAT_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class DecimalFormat;
Index: exp/src/third_party/v8-i18n/src/utils.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/src/utils.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/src/utils.h	2012-05-19 19:23:29.338671560 +0200
@@ -16,7 +16,7 @@
 #define V8_I18N_SRC_UTILS_H_
 
 #include "unicode/uversion.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace U_ICU_NAMESPACE {
 class UnicodeString;
Index: exp/src/webkit/glue/webkitplatformsupport_impl.cc
===================================================================
--- exp.orig/src/webkit/glue/webkitplatformsupport_impl.cc	2012-05-19 16:54:05.000000000 +0200
+++ exp/src/webkit/glue/webkitplatformsupport_impl.cc	2012-05-19 19:23:29.338671560 +0200
@@ -47,7 +47,7 @@
 #include "webkit/plugins/webplugininfo.h"
 
 #if defined(OS_LINUX)
-#include "v8/include/v8.h"
+#include <v8.h>
 #endif
 
 using WebKit::WebAudioBus;
Index: exp/src/webkit/support/test_webkit_platform_support.cc
===================================================================
--- exp.orig/src/webkit/support/test_webkit_platform_support.cc	2012-05-19 16:54:06.000000000 +0200
+++ exp/src/webkit/support/test_webkit_platform_support.cc	2012-05-19 19:23:29.338671560 +0200
@@ -32,7 +32,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebStorageNamespace.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/appcache/web_application_cache_host_impl.h"
 #include "webkit/database/vfs_backend.h"
 #include "webkit/glue/simple_webmimeregistry_impl.h"
Index: exp/src/chrome/chrome_browser.gypi
===================================================================
--- exp.orig/src/chrome/chrome_browser.gypi	2012-05-19 16:55:35.000000000 +0200
+++ exp/src/chrome/chrome_browser.gypi	2012-05-19 19:23:29.338671560 +0200
@@ -58,7 +58,7 @@
         '../ui/ui.gyp:ui_resources',
         '../ui/ui.gyp:ui_resources_2x',
         '../ui/ui.gyp:ui_resources_standard',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         '../webkit/support/webkit_support.gyp:appcache',
         '../webkit/support/webkit_support.gyp:blob',
         '../webkit/support/webkit_support.gyp:database',
Index: exp/src/content/content_shell.gypi
===================================================================
--- exp.orig/src/content/content_shell.gypi	2012-05-19 16:54:14.000000000 +0200
+++ exp/src/content/content_shell.gypi	2012-05-19 19:23:29.338671560 +0200
@@ -35,7 +35,7 @@
         '../skia/skia.gyp:skia',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
         '../ui/ui.gyp:ui',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         '../webkit/support/webkit_support.gyp:appcache',
         '../webkit/support/webkit_support.gyp:database',
         '../webkit/support/webkit_support.gyp:fileapi',
Index: exp/src/content/content_tests.gypi
===================================================================
--- exp.orig/src/content/content_tests.gypi	2012-05-19 16:54:14.000000000 +0200
+++ exp/src/content/content_tests.gypi	2012-05-19 19:23:29.338671560 +0200
@@ -189,7 +189,7 @@
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
         '../ui/gfx/gl/gl.gyp:gl',
         '../ui/ui.gyp:ui',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         '../webkit/support/webkit_support.gyp:database',
         '../webkit/support/webkit_support.gyp:glue',
         '../webkit/support/webkit_support.gyp:quota',
Index: exp/src/third_party/WebKit/Tools/DumpRenderTree/DumpRenderTree.gyp/DumpRenderTree.gyp
===================================================================
--- exp.orig/src/third_party/WebKit/Tools/DumpRenderTree/DumpRenderTree.gyp/DumpRenderTree.gyp	2012-05-19 16:56:08.000000000 +0200
+++ exp/src/third_party/WebKit/Tools/DumpRenderTree/DumpRenderTree.gyp/DumpRenderTree.gyp	2012-05-19 19:31:51.750676947 +0200
@@ -83,7 +83,7 @@
                 '<(source_dir)/WTF/WTF.gyp/WTF.gyp:wtf',
                 '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
                 '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                 '<(chromium_src_dir)/base/base.gyp:test_support_base',
                 '<(chromium_src_dir)/webkit/support/webkit_support.gyp:blob',
                 '<(chromium_src_dir)/webkit/support/webkit_support.gyp:webkit_support',
Index: exp/src/v8/src/d8.gyp
===================================================================
--- exp.orig/src/v8/src/d8.gyp	2012-05-19 16:57:28.000000000 +0200
+++ exp/src/v8/src/d8.gyp	2012-05-19 19:23:29.338671560 +0200
@@ -35,7 +35,7 @@
       'target_name': 'd8',
       'type': 'executable',
       'dependencies': [
-        '../tools/gyp/v8.gyp:v8',
+        '../../build/linux/system.gyp:v8',
       ],
       # Generated source files need this explicitly:
       'include_dirs+': [
Index: exp/src/chrome/renderer/benchmarking_extension.cc
===================================================================
--- exp.orig/src/chrome/renderer/benchmarking_extension.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/benchmarking_extension.cc	2012-05-19 19:23:29.338671560 +0200
@@ -11,7 +11,7 @@
 #include "content/public/common/content_switches.h"
 #include "content/public/renderer/render_thread.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebCache.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebCache;
 
Index: exp/src/chrome/renderer/extensions/chrome_v8_extension_handler.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/chrome_v8_extension_handler.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/chrome_v8_extension_handler.h	2012-05-19 19:23:29.338671560 +0200
@@ -10,7 +10,7 @@
 
 #include "base/basictypes.h"
 #include "ipc/ipc_channel.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class ChromeV8Context;
 
Index: exp/src/chrome/renderer/extensions/page_capture_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/page_capture_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/page_capture_custom_bindings.cc	2012-05-19 19:23:29.338671560 +0200
@@ -9,7 +9,7 @@
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebBlob.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/web_request_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/web_request_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/web_request_custom_bindings.cc	2012-05-19 19:23:29.338671560 +0200
@@ -8,7 +8,7 @@
 
 #include "base/string_number_conversions.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/content/shell/layout_test_controller_bindings.h
===================================================================
--- exp.orig/src/content/shell/layout_test_controller_bindings.h	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/shell/layout_test_controller_bindings.h	2012-05-19 19:23:29.338671560 +0200
@@ -8,7 +8,7 @@
 
 #include "base/basictypes.h"
 #include "base/compiler_specific.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace content {
 
Index: exp/src/content/shell/shell_content_renderer_client.cc
===================================================================
--- exp.orig/src/content/shell/shell_content_renderer_client.cc	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/shell/shell_content_renderer_client.cc	2012-05-19 19:23:29.338671560 +0200
@@ -6,7 +6,7 @@
 
 #include "content/shell/shell_render_process_observer.h"
 #include "content/shell/shell_render_view_observer.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace content {
 
Index: exp/src/chrome/browser/ui/webui/about_ui.cc
===================================================================
--- exp.orig/src/chrome/browser/ui/webui/about_ui.cc	2012-05-19 16:54:57.000000000 +0200
+++ exp/src/chrome/browser/ui/webui/about_ui.cc	2012-05-19 19:23:29.338671560 +0200
@@ -65,7 +65,7 @@
 #include "net/base/net_util.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/user_agent.h"
 #include "webkit/glue/webkit_glue.h"
 #include "webkit/plugins/webplugininfo.h"
Index: exp/src/chrome/browser/ui/webui/help/help_handler.cc
===================================================================
--- exp.orig/src/chrome/browser/ui/webui/help/help_handler.cc	2012-05-19 16:54:56.000000000 +0200
+++ exp/src/chrome/browser/ui/webui/help/help_handler.cc	2012-05-19 19:23:29.342671561 +0200
@@ -29,7 +29,7 @@
 #include "grit/google_chrome_strings.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/resource/resource_bundle.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/user_agent.h"
 #include "webkit/glue/webkit_glue.h"
 
Index: exp/src/chrome/common/profiling.cc
===================================================================
--- exp.orig/src/chrome/common/profiling.cc	2012-05-19 16:55:34.000000000 +0200
+++ exp/src/chrome/common/profiling.cc	2012-05-19 19:23:29.342671561 +0200
@@ -13,7 +13,7 @@
 #include "base/string_util.h"
 #include "base/threading/thread.h"
 #include "chrome/common/chrome_switches.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 std::string GetProfileName() {
Index: exp/src/chrome/renderer/extensions/app_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/app_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/app_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -21,7 +21,7 @@
 #include "grit/renderer_resources.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebFrame;
 using content::V8ValueConverter;
Index: exp/src/chrome/renderer/extensions/chrome_v8_context.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/chrome_v8_context.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/chrome_v8_context.cc	2012-05-19 19:23:29.342671561 +0200
@@ -15,7 +15,7 @@
 #include "content/public/renderer/render_view.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using extensions::Feature;
 
Index: exp/src/chrome/renderer/extensions/chrome_v8_context.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/chrome_v8_context.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/chrome_v8_context.h	2012-05-19 19:23:29.342671561 +0200
@@ -11,7 +11,7 @@
 #include "base/basictypes.h"
 #include "chrome/common/extensions/feature.h"
 #include "chrome/renderer/module_system.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace WebKit {
 class WebFrame;
Index: exp/src/chrome/renderer/extensions/chrome_v8_context_set.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/chrome_v8_context_set.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/chrome_v8_context_set.cc	2012-05-19 19:23:29.342671561 +0200
@@ -13,7 +13,7 @@
 #include "content/public/renderer/render_thread.h"
 #include "content/public/renderer/v8_value_converter.h"
 #include "content/public/renderer/render_view.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebDocument.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
Index: exp/src/chrome/renderer/extensions/chrome_v8_context_set.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/chrome_v8_context_set.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/chrome_v8_context_set.h	2012-05-19 19:23:29.342671561 +0200
@@ -10,7 +10,7 @@
 #include <string>
 
 #include "base/basictypes.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class ChromeV8Context;
 class GURL;
Index: exp/src/chrome/renderer/extensions/chrome_v8_context_set_unittest.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/chrome_v8_context_set_unittest.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/chrome_v8_context_set_unittest.cc	2012-05-19 19:23:29.342671561 +0200
@@ -9,7 +9,7 @@
 #include "chrome/renderer/extensions/chrome_v8_context_set.h"
 #include "testing/gtest/include/gtest/gtest.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 TEST(ChromeV8ContextSet, Lifecycle) {
   MessageLoop loop;
Index: exp/src/chrome/renderer/extensions/chrome_v8_extension.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/chrome_v8_extension.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/chrome_v8_extension.h	2012-05-19 19:23:29.342671561 +0200
@@ -11,7 +11,7 @@
 #include "base/string_piece.h"
 #include "chrome/renderer/extensions/chrome_v8_extension_handler.h"
 #include "chrome/renderer/native_handler.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <map>
 #include <set>
Index: exp/src/chrome/renderer/extensions/context_menus_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/context_menus_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/context_menus_custom_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -7,7 +7,7 @@
 #include "chrome/common/extensions/extension_messages.h"
 #include "content/public/renderer/render_thread.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 
Index: exp/src/chrome/renderer/extensions/experimental.socket_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/experimental.socket_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/experimental.socket_custom_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -9,7 +9,7 @@
 #include "chrome/common/extensions/extension_action.h"
 #include "chrome/renderer/extensions/extension_dispatcher.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 
Index: exp/src/chrome/renderer/extensions/extension_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/extension_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/extension_custom_bindings.cc	2012-05-19 19:35:36.774679368 +0200
@@ -18,7 +18,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebFrame.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "webkit/glue/webkit_glue.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/extension_dispatcher.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/extension_dispatcher.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/extension_dispatcher.h	2012-05-19 19:23:29.342671561 +0200
@@ -19,7 +19,7 @@
 #include "chrome/renderer/extensions/chrome_v8_context_set.h"
 #include "chrome/renderer/extensions/v8_schema_registry.h"
 #include "chrome/renderer/resource_bundle_source_map.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class ExtensionRequestSender;
 class GURL;
Index: exp/src/chrome/renderer/extensions/i18n_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/i18n_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/i18n_custom_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -8,7 +8,7 @@
 #include "chrome/common/extensions/extension_message_bundle.h"
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/miscellaneous_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/miscellaneous_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/miscellaneous_bindings.cc	2012-05-19 19:34:15.342678488 +0200
@@ -21,7 +21,7 @@
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebScopedMicrotaskSuppression.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // Message passing API example (in a content script):
 // var extension =
Index: exp/src/chrome/renderer/extensions/page_actions_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/page_actions_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/page_actions_custom_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -9,7 +9,7 @@
 #include "chrome/common/extensions/extension_action.h"
 #include "chrome/renderer/extensions/extension_dispatcher.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/tabs_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/tabs_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/tabs_custom_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -11,7 +11,7 @@
 #include "chrome/renderer/extensions/extension_dispatcher.h"
 #include "content/public/renderer/render_view.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/tts_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/tts_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/tts_custom_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -9,7 +9,7 @@
 #include "chrome/common/extensions/extension_action.h"
 #include "chrome/renderer/extensions/extension_dispatcher.h"
 #include "grit/renderer_resources.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/v8_schema_registry.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/v8_schema_registry.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/v8_schema_registry.h	2012-05-19 19:23:29.342671561 +0200
@@ -11,7 +11,7 @@
 #include <string>
 
 #include "base/basictypes.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/webstore_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/webstore_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/webstore_bindings.cc	2012-05-19 19:23:29.342671561 +0200
@@ -15,7 +15,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebElement.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNode.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNodeList.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 using WebKit::WebDocument;
 using WebKit::WebElement;
Index: exp/src/chrome/renderer/module_system.h
===================================================================
--- exp.orig/src/chrome/renderer/module_system.h	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/module_system.h	2012-05-19 19:23:29.342671561 +0200
@@ -10,7 +10,7 @@
 #include "base/memory/linked_ptr.h"
 #include "base/memory/scoped_ptr.h"
 #include "chrome/renderer/native_handler.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <map>
 #include <string>
Index: exp/src/chrome/renderer/native_handler.cc
===================================================================
--- exp.orig/src/chrome/renderer/native_handler.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/native_handler.cc	2012-05-19 19:37:42.606680716 +0200
@@ -7,7 +7,7 @@
 #include "base/memory/linked_ptr.h"
 #include "base/logging.h"
 #include "chrome/renderer/module_system.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 NativeHandler::NativeHandler()
     : object_template_(
Index: exp/src/chrome/renderer/native_handler.h
===================================================================
--- exp.orig/src/chrome/renderer/native_handler.h	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/native_handler.h	2012-05-19 19:23:29.342671561 +0200
@@ -8,7 +8,7 @@
 
 #include "base/bind.h"
 #include "base/memory/linked_ptr.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <string>
 #include <vector>
Index: exp/src/chrome/renderer/resource_bundle_source_map.h
===================================================================
--- exp.orig/src/chrome/renderer/resource_bundle_source_map.h	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/resource_bundle_source_map.h	2012-05-19 19:23:29.342671561 +0200
@@ -11,7 +11,7 @@
 #include "base/string_piece.h"
 #include "chrome/renderer/module_system.h"
 #include "chrome/renderer/static_v8_external_string_resource.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 #include <map>
 #include <string>
Index: exp/src/chrome/renderer/static_v8_external_string_resource.h
===================================================================
--- exp.orig/src/chrome/renderer/static_v8_external_string_resource.h	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/static_v8_external_string_resource.h	2012-05-19 19:23:29.342671561 +0200
@@ -8,7 +8,7 @@
 
 #include "base/compiler_specific.h"
 #include "base/string_piece.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 // A very simple implementation of v8::ExternalAsciiStringResource that just
 // wraps a buffer. The buffer must outlive the v8 runtime instance this resource
Index: exp/src/chrome/test/reliability/page_load_test.cc
===================================================================
--- exp.orig/src/chrome/test/reliability/page_load_test.cc	2012-05-19 16:55:30.000000000 +0200
+++ exp/src/chrome/test/reliability/page_load_test.cc	2012-05-19 19:23:29.342671561 +0200
@@ -66,7 +66,7 @@
 #include "chrome/test/ui/ui_test.h"
 #include "net/base/net_util.h"
 #include "ui/base/keycodes/keyboard_codes.h"
-#include "v8/include/v8-testing.h"
+#include <v8-testing.h>
 
 namespace {
 
Index: exp/src/content/public/renderer/v8_value_converter.h
===================================================================
--- exp.orig/src/content/public/renderer/v8_value_converter.h	2012-05-19 16:54:14.000000000 +0200
+++ exp/src/content/public/renderer/v8_value_converter.h	2012-05-19 19:23:29.342671561 +0200
@@ -6,7 +6,7 @@
 #define CONTENT_PUBLIC_RENDERER_V8_VALUE_CONVERTER_H_
 
 #include "content/common/content_export.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace base {
 class Value;
Index: exp/src/content/renderer/render_thread_impl.cc
===================================================================
--- exp.orig/src/content/renderer/render_thread_impl.cc	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/renderer/render_thread_impl.cc	2012-05-19 19:36:41.902680063 +0200
@@ -80,7 +80,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "ui/base/ui_base_switches.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/webkit_glue.h"
 
 // TODO(port)
Index: exp/src/content/renderer/render_view_impl.cc
===================================================================
--- exp.orig/src/content/renderer/render_view_impl.cc	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/renderer/render_view_impl.cc	2012-05-19 19:23:29.346671563 +0200
@@ -157,7 +157,7 @@
 #include "ui/gfx/native_widget_types.h"
 #include "ui/gfx/point.h"
 #include "ui/gfx/rect.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/appcache/web_application_cache_host_impl.h"
 #include "webkit/dom_storage/dom_storage_types.h"
 #include "webkit/forms/form_data.h"
Index: exp/src/content/renderer/v8_value_converter_impl.cc
===================================================================
--- exp.orig/src/content/renderer/v8_value_converter_impl.cc	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/renderer/v8_value_converter_impl.cc	2012-05-19 19:23:29.346671563 +0200
@@ -9,7 +9,7 @@
 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
 #include "base/values.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace content {
 
Index: exp/src/content/renderer/v8_value_converter_impl_unittest.cc
===================================================================
--- exp.orig/src/content/renderer/v8_value_converter_impl_unittest.cc	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/renderer/v8_value_converter_impl_unittest.cc	2012-05-19 19:23:29.346671563 +0200
@@ -8,7 +8,7 @@
 #include "base/values.h"
 #include "content/renderer/v8_value_converter_impl.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class V8ValueConverterImplTest : public testing::Test {
  protected:
Index: exp/src/content/renderer/web_intents_host.cc
===================================================================
--- exp.orig/src/content/renderer/web_intents_host.cc	2012-05-19 16:54:13.000000000 +0200
+++ exp/src/content/renderer/web_intents_host.cc	2012-05-19 19:23:29.346671563 +0200
@@ -17,7 +17,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebView.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebCString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebSerializedScriptValue.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/glue/cpp_bound_class.h"
 
 using WebKit::WebBindings;
Index: exp/src/third_party/WebKit/Tools/DumpRenderTree/chromium/LayoutTestController.cpp
===================================================================
--- exp.orig/src/third_party/WebKit/Tools/DumpRenderTree/chromium/LayoutTestController.cpp	2012-05-19 16:56:15.000000000 +0200
+++ exp/src/third_party/WebKit/Tools/DumpRenderTree/chromium/LayoutTestController.cpp	2012-05-19 19:23:29.346671563 +0200
@@ -61,7 +61,7 @@
 #include "platform/WebURL.h"
 #include "WebView.h"
 #include "WebViewHost.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/support/webkit_support.h"
 #include <algorithm>
 #include <cctype>
Index: exp/src/webkit/media/webmediaplayer_impl.cc
===================================================================
--- exp.orig/src/webkit/media/webmediaplayer_impl.cc	2012-05-19 16:54:05.000000000 +0200
+++ exp/src/webkit/media/webmediaplayer_impl.cc	2012-05-19 19:23:29.346671563 +0200
@@ -29,7 +29,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebSize.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebString.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebURL.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/media/buffered_data_source.h"
 #include "webkit/media/filter_helpers.h"
 #include "webkit/media/key_systems.h"
Index: exp/src/webkit/plugins/ppapi/message_channel.cc
===================================================================
--- exp.orig/src/webkit/plugins/ppapi/message_channel.cc	2012-05-19 16:54:05.000000000 +0200
+++ exp/src/webkit/plugins/ppapi/message_channel.cc	2012-05-19 19:23:29.346671563 +0200
@@ -21,7 +21,7 @@
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebNode.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/WebPluginContainer.h"
 #include "third_party/WebKit/Source/WebKit/chromium/public/platform/WebSerializedScriptValue.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 #include "webkit/plugins/ppapi/host_array_buffer_var.h"
 #include "webkit/plugins/ppapi/npapi_glue.h"
 #include "webkit/plugins/ppapi/ppapi_plugin_instance.h"
Index: exp/src/chrome/renderer/extensions/api_definitions_natives.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/api_definitions_natives.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/api_definitions_natives.h	2012-05-19 19:33:35.886678071 +0200
@@ -9,7 +9,7 @@
 #include "chrome/renderer/extensions/chrome_v8_extension.h"
 #include "chrome/renderer/extensions/extension_dispatcher.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace extensions {
 
Index: exp/src/chrome/renderer/extensions/experimental.usb_custom_bindings.cc
===================================================================
--- exp.orig/src/chrome/renderer/extensions/experimental.usb_custom_bindings.cc	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/experimental.usb_custom_bindings.cc	2012-05-19 19:33:53.406678252 +0200
@@ -5,7 +5,7 @@
 #include "chrome/renderer/extensions/experimental.usb_custom_bindings.h"
 
 #include "base/logging.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace {
 
Index: exp/src/chrome/renderer/extensions/extension_request_sender.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/extension_request_sender.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/extension_request_sender.h	2012-05-19 19:35:11.870679095 +0200
@@ -10,7 +10,7 @@
 #include <map>
 
 #include "base/memory/linked_ptr.h"
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class ChromeV8ContextSet;
 class ExtensionDispatcher;
Index: exp/src/chrome/renderer/extensions/send_request_natives.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/send_request_natives.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/send_request_natives.h	2012-05-19 19:34:44.198678803 +0200
@@ -9,7 +9,7 @@
 #include "chrome/renderer/extensions/chrome_v8_extension.h"
 #include "chrome/renderer/extensions/extension_dispatcher.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class ExtensionRequestSender;
 
Index: exp/src/chrome/renderer/extensions/set_icon_natives.h
===================================================================
--- exp.orig/src/chrome/renderer/extensions/set_icon_natives.h	2012-05-19 16:55:00.000000000 +0200
+++ exp/src/chrome/renderer/extensions/set_icon_natives.h	2012-05-19 19:34:28.678678632 +0200
@@ -9,7 +9,7 @@
 #include "chrome/renderer/extensions/chrome_v8_extension.h"
 #include "chrome/renderer/extensions/extension_dispatcher.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 class ExtensionRequestSender;
 
Index: exp/src/chrome/renderer/playback_extension.cc
===================================================================
--- exp.orig/src/chrome/renderer/playback_extension.cc	2012-05-19 16:55:01.000000000 +0200
+++ exp/src/chrome/renderer/playback_extension.cc	2012-05-19 19:36:59.774680256 +0200
@@ -4,7 +4,7 @@
 
 #include "chrome/renderer/playback_extension.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 const char kPlaybackExtensionName[] = "v8/PlaybackMode";
 
Index: exp/src/third_party/WebKit/Source/WebKit/chromium/WebKitUnitTests.gyp
===================================================================
--- exp.orig/src/third_party/WebKit/Source/WebKit/chromium/WebKitUnitTests.gyp	2012-05-19 16:56:54.000000000 +0200
+++ exp/src/third_party/WebKit/Source/WebKit/chromium/WebKitUnitTests.gyp	2012-05-19 19:31:13.006676553 +0200
@@ -56,7 +56,7 @@
             'dependencies': [
                 'WebKit.gyp:webkit',
                 '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                 '<(chromium_src_dir)/testing/gtest.gyp:gtest',
                 '<(chromium_src_dir)/testing/gmock.gyp:gmock',
                 '<(chromium_src_dir)/base/base.gyp:base',
Index: exp/src/third_party/WebKit/Tools/TestWebKitAPI/TestWebKitAPI.gyp/TestWebKitAPI.gyp
===================================================================
--- exp.orig/src/third_party/WebKit/Tools/TestWebKitAPI/TestWebKitAPI.gyp/TestWebKitAPI.gyp	2012-05-19 16:56:20.000000000 +0200
+++ exp/src/third_party/WebKit/Tools/TestWebKitAPI/TestWebKitAPI.gyp/TestWebKitAPI.gyp	2012-05-19 19:31:33.734676760 +0200
@@ -57,7 +57,7 @@
                 '<(source_dir)/WebKit/chromium/WebKit.gyp:webkit', 
                 '<(source_dir)/WTF/WTF.gyp/WTF.gyp:wtf', 
                 '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
-                '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+                '<(chromium_src_dir)/build/linux/system.gyp:v8',
                 '<(chromium_src_dir)/base/base.gyp:test_support_base', 
                 '<(chromium_src_dir)/testing/gtest.gyp:gtest', 
                 '<(chromium_src_dir)/testing/gmock.gyp:gmock', 
Index: exp/src/third_party/v8-i18n/src/locale.h
===================================================================
--- exp.orig/src/third_party/v8-i18n/src/locale.h	2012-05-19 16:56:57.000000000 +0200
+++ exp/src/third_party/v8-i18n/src/locale.h	2012-05-19 19:38:02.998680935 +0200
@@ -15,7 +15,7 @@
 #ifndef V8_I18N_SRC_LOCALE_H_
 #define V8_I18N_SRC_LOCALE_H_
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 namespace v8_i18n {
 
Index: exp/src/webkit/support/gc_extension.cc
===================================================================
--- exp.orig/src/webkit/support/gc_extension.cc	2012-05-19 16:54:06.000000000 +0200
+++ exp/src/webkit/support/gc_extension.cc	2012-05-19 19:36:26.990679900 +0200
@@ -4,7 +4,7 @@
 
 #include "webkit/support/gc_extension.h"
 
-#include "v8/include/v8.h"
+#include <v8.h>
 
 const char kGCExtensionName[] = "v8/GCController";
 
