Use system speex
--- a/src/build/linux/system.gyp
+++ b/src/build/linux/system.gyp
@@ -294,6 +294,26 @@
       }]]
     },
     {
+      'target_name': 'speex',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(pkg-config --cflags speex)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other speex)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l speex)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome-keyring',
       'type': 'settings',
       'conditions': [
--- a/src/chrome/chrome_browser.gypi
+++ b/src/chrome/chrome_browser.gypi
@@ -40,7 +40,7 @@
         '../third_party/libxml/libxml.gyp:libxml',
         '../third_party/npapi/npapi.gyp:npapi',
         '../third_party/hunspell/hunspell.gyp:hunspell',
-        '../third_party/speex/speex.gyp:libspeex',
+        '../build/linux/system.gyp:speex',
         '../webkit/support/webkit_support.gyp:appcache',
         '../webkit/support/webkit_support.gyp:blob',
         '../webkit/support/webkit_support.gyp:database',
--- a/src/chrome/browser/speech/audio_encoder.cc
+++ b/src/chrome/browser/speech/audio_encoder.cc
@@ -10,7 +10,7 @@
 #include "base/stl_util-inl.h"
 #include "base/string_number_conversions.h"
 #include "third_party/flac/flac.h"
-#include "third_party/speex/speex.h"
+#include <speex/speex.h>
 
 using std::string;
 
