Index: sid/src/remoting/remoting.gyp
===================================================================
--- sid.orig/src/remoting/remoting.gyp	2012-08-06 10:27:44.000000000 +0200
+++ sid/src/remoting/remoting.gyp	2012-08-06 12:15:16.629549120 +0200
@@ -1031,7 +1031,7 @@
         '../ui/ui.gyp:ui',
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
-        '../third_party/libvpx/libvpx.gyp:libvpx',
+        '../third_party/libvpx/libvpx.gyp:libvpx_include',
         '../third_party/protobuf/protobuf.gyp:protobuf_lite',
         '../third_party/zlib/zlib.gyp:zlib',
         '../media/media.gyp:yuv_convert',
Index: sid/src/third_party/libvpx/libvpx.gyp
===================================================================
--- sid.orig/src/third_party/libvpx/libvpx.gyp	2012-08-06 10:30:05.000000000 +0200
+++ sid/src/third_party/libvpx/libvpx.gyp	2012-08-06 12:15:16.665549120 +0200
@@ -281,11 +281,17 @@
     {
       'target_name': 'libvpx_include',
       'type': 'none',
-      'direct_dependent_settings': {
-        'include_dirs': [
-          'source/libvpx',
-        ],
-      },
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other vpx)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l vpx)',
+            ],
+          },
+      }]]
     },
     {
       'target_name': 'libvpx_lib',
