--- a/src/remoting/remoting.gyp
+++ b/src/remoting/remoting.gyp
@@ -427,7 +427,7 @@
         '../ui/ui.gyp:ui',
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
-        '../third_party/libvpx/libvpx.gyp:libvpx',
+        '../third_party/libvpx/libvpx.gyp:libvpx_include',
         '../third_party/protobuf/protobuf.gyp:protobuf_lite',
         '../third_party/zlib/zlib.gyp:zlib',
         '../media/media.gyp:yuv_convert',
--- a/src/third_party/ffmpeg/ffmpeg.gyp
+++ b/src/third_party/ffmpeg/ffmpeg.gyp
@@ -513,7 +513,7 @@
             ['OS == "linux" or OS == "solaris"', {
               'link_settings': {
                 'libraries': [
-                  '-ldl',
+                  '-ldl', '-lvpx',
                 ],
               },
             }],
