Index: sid/src/remoting/remoting.gyp
===================================================================
--- sid.orig/src/remoting/remoting.gyp	2012-09-26 11:12:57.802698559 +0200
+++ sid/src/remoting/remoting.gyp	2012-09-26 16:58:35.194257270 +0200
@@ -1139,7 +1139,7 @@
         '../ui/ui.gyp:ui',
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
-        '../third_party/libvpx/libvpx.gyp:libvpx',
+        '../third_party/libvpx/libvpx.gyp:libvpx_include',
         '../third_party/protobuf/protobuf.gyp:protobuf_lite',
         '../third_party/zlib/zlib.gyp:zlib',
         '../media/media.gyp:yuv_convert',
