Index: exp/src/remoting/remoting.gyp
===================================================================
--- exp.orig/src/remoting/remoting.gyp	2012-05-28 11:15:29.000000000 +0200
+++ exp/src/remoting/remoting.gyp	2012-05-28 11:16:36.962675697 +0200
@@ -1029,7 +1029,7 @@
         '../ui/ui.gyp:ui',
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
-        '../third_party/libvpx/libvpx.gyp:libvpx',
+        '../third_party/libvpx/libvpx.gyp:libvpx_include',
         '../build/linux/system.gyp:protobuf',
         '../third_party/zlib/zlib.gyp:zlib',
         '../media/media.gyp:yuv_convert',
Index: exp/src/third_party/ffmpeg/ffmpeg.gyp
===================================================================
--- exp.orig/src/third_party/ffmpeg/ffmpeg.gyp	2012-05-28 11:06:45.000000000 +0200
+++ exp/src/third_party/ffmpeg/ffmpeg.gyp	2012-05-28 11:15:48.230675556 +0200
@@ -504,7 +504,7 @@
             ['OS == "linux" or OS == "solaris"', {
               'link_settings': {
                 'libraries': [
-                  '-ldl',
+                  '-ldl', '-lvpx',
                 ],
               },
             }],
