description: debian-specific version parsing (for about dialog)
Index: chromium-browser/src/webkit/build/webkit_version.py
===================================================================
--- chromium-browser.orig/src/webkit/build/webkit_version.py	2012-03-28 21:07:06.182218195 -0400
+++ chromium-browser/src/webkit/build/webkit_version.py	2012-03-29 13:57:39.203509514 -0400
@@ -61,6 +61,15 @@
       default_lastchange=None,
       directory=os.path.join(webkit_dir, version_file_dir),
       directory_regex_prior_to_src_url='webkit')
+  if version_info.url is None or version_info.url == 'unknown':
+    # parse the DEPS file
+    data = open("../../DEPS", "rb").read()
+    x = re.search(r'"webkit_revision": "(\d+)",', data)
+    rev = x.group(1) if x else "n-a"
+    x = re.search(r'"webkit_trunk": "(.*?)",', data)
+    branch = os.path.basename(x.group(1)) if x else "n-a"
+    return "%s@%s" % (branch, rev)
+
 
   if version_info.url == None:
     version_info.url = 'Unknown URL'
