description: debian-specific version parsing (for about dialog)
Index: sid/src/webkit/build/webkit_version.py
===================================================================
--- sid.orig/src/webkit/build/webkit_version.py	2013-01-29 17:08:05.000000000 +0100
+++ sid/src/webkit/build/webkit_version.py	2013-02-06 13:36:27.753890448 +0100
@@ -61,6 +61,15 @@
       default_lastchange=None,
       directory=os.path.join(webkit_dir, version_file_dir),
       directory_regex_prior_to_src_url='webkit')
+  if version_info.url is None or version_info.url == 'unknown':
+    # parse the DEPS file
+    data = open("../../DEPS", "rb").read()
+    x = re.search(r'"webkit_revision": "(\d+)",', data)
+    rev = x.group(1) if x else "n-a"
+    x = re.search(r'"webkit_trunk": "(.*?)",', data)
+    branch = os.path.basename(x.group(1)) if x else "n-a"
+    return "%s@%s" % (branch, rev)
+
 
   if version_info.url == None:
     version_info.url = 'Unknown URL'
