--- a/src/webkit/build/webkit_version.py
+++ b/src/webkit/build/webkit_version.py
@@ -51,6 +51,15 @@ def GetWebKitRevision(webkit_dir, versio
       default_lastchange=None,
       directory=os.path.join(webkit_dir, version_file_dir))
 
+  if version_info.url is None or version_info.url == 'unknown':
+    # parse the DEPS file
+    data = open("../../DEPS", "rb").read()
+    x = re.search(r'"webkit_revision": "(\d+)",', data)
+    rev = x.group(1) if x else "n-a"
+    x = re.search(r'"webkit_trunk": "(.*?)",', data)
+    branch = os.path.basename(x.group(1)) if x else "n-a"
+    return "%s@%s" % (branch, rev)
+
   if (version_info.url.startswith(version_info.root) and
       version_info.url.endswith(version_file_dir)):
     # Now compute the real WebKit URL by stripping off the version file
