Subject: Prevent windows from jumping to front for new tabs
From: Tony Houghton <h@realh.co.uk>
Bug: https://code.google.com/p/chromium/issues/detail?id=297679
Bug-Debian: http://bugs.debian.org/725350
Index: sid/src/chrome/browser/ui/gtk/browser_window_gtk.cc
===================================================================
--- sid.orig/src/chrome/browser/ui/gtk/browser_window_gtk.cc	2014-02-02 19:28:17.496401871 +0100
+++ sid/src/chrome/browser/ui/gtk/browser_window_gtk.cc	2014-02-02 19:28:17.492401871 +0100
@@ -579,7 +579,7 @@
   // browser instead if we don't explicitly set it here.
   BrowserList::SetLastActive(browser());
 
-  gtk_window_present(window_);
+  gtk_widget_show(GTK_WIDGET(window_));
   if (show_state_after_show_ == ui::SHOW_STATE_MAXIMIZED) {
     gtk_window_maximize(window_);
     show_state_after_show_ = ui::SHOW_STATE_NORMAL;
