Subject: Prevent windows from jumping to front for new tabs
From: Tony Houghton <h@realh.co.uk>
Bug: https://code.google.com/p/chromium/issues/detail?id=297679
Bug-Debian: http://bugs.debian.org/725350
Index: chromium/src/chrome/browser/ui/gtk/browser_window_gtk.cc
===================================================================
--- chromium.orig/src/chrome/browser/ui/gtk/browser_window_gtk.cc	2014-04-11 01:43:31.348278927 +0000
+++ chromium/src/chrome/browser/ui/gtk/browser_window_gtk.cc	2014-04-11 01:43:31.344278927 +0000
@@ -577,7 +577,7 @@
   // browser instead if we don't explicitly set it here.
   BrowserList::SetLastActive(browser());
 
-  gtk_window_present(window_);
+  gtk_widget_show(GTK_WIDGET(window_));
   if (show_state_after_show_ == ui::SHOW_STATE_MAXIMIZED) {
     gtk_window_maximize(window_);
     show_state_after_show_ = ui::SHOW_STATE_NORMAL;
