--- a/base/compiler_specific.h
+++ b/base/compiler_specific.h
@@ -366,7 +366,7 @@ inline constexpr bool AnalyzerAssumeTrue
 #endif  // defined(__clang_analyzer__)
 
 // Use nomerge attribute to disable optimization of merging multiple same calls.
-#if defined(__clang__) && __has_attribute(nomerge)
+#if defined(__clang__) && (__clang_major__ >= 12) && __has_attribute(nomerge)
 #define NOMERGE [[clang::nomerge]]
 #else
 #define NOMERGE
