description: add constexpr to methods where it is required
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/core/animation/animation_time_delta.h
+++ b/third_party/blink/renderer/core/animation/animation_time_delta.h
@@ -53,7 +53,7 @@ class CORE_EXPORT AnimationTimeDelta {
     return AnimationTimeDelta(std::numeric_limits<double>::infinity());
   }
 
-  double InSecondsF() const { return delta_; }
+  constexpr double InSecondsF() const { return delta_; }
   double InMillisecondsF() const { return delta_ * 1000; }
 
   bool is_max() const {
--- a/third_party/blink/renderer/core/layout/ng/inline/ng_baseline.cc
+++ b/third_party/blink/renderer/core/layout/ng/inline/ng_baseline.cc
@@ -11,7 +11,7 @@
 namespace blink {
 
 const unsigned NGBaselineRequest::kTypeIdCount;
-const LayoutUnit NGBaselineList::kEmptyOffset;
+constexpr LayoutUnit NGBaselineList::kEmptyOffset;
 
 bool NGBaselineRequest::operator==(const NGBaselineRequest& other) const {
   return algorithm_type_ == other.algorithm_type_ &&
--- a/device/gamepad/public/cpp/gamepad.cc
+++ b/device/gamepad/public/cpp/gamepad.cc
@@ -6,12 +6,12 @@
 
 namespace device {
 
-const float GamepadButton::kDefaultButtonPressedThreshold;
-const double GamepadHapticActuator::kMaxEffectDurationMillis;
-const size_t Gamepad::kIdLengthCap;
-const size_t Gamepad::kMappingLengthCap;
-const size_t Gamepad::kAxesLengthCap;
-const size_t Gamepad::kButtonsLengthCap;
+constexpr float GamepadButton::kDefaultButtonPressedThreshold;
+constexpr double GamepadHapticActuator::kMaxEffectDurationMillis;
+constexpr size_t Gamepad::kIdLengthCap;
+constexpr size_t Gamepad::kMappingLengthCap;
+constexpr size_t Gamepad::kAxesLengthCap;
+constexpr size_t Gamepad::kButtonsLengthCap;
 
 Gamepad::Gamepad()
     : connected(false),
