description: gcc 6 won't compile 'extern "C"' as a macro
author: Michael Gilbert <mgilbert@debian.org>

--- a/base/allocator/allocator_shim_override_cpp_symbols.h
+++ b/base/allocator/allocator_shim_override_cpp_symbols.h
@@ -36,7 +36,7 @@
 #define ALIGN_DEL_ARR_NOTHROW operator delete[]
 #else
 #define ALIGN_VAL_T size_t
-#define ALIGN_LINKAGE extern "C"
+#define ALIGN_LINKAGE
 #if defined(OS_MACOSX) || defined(OS_WIN)
 #error "Mangling is different on these platforms."
 #else
@@ -96,6 +96,8 @@ SHIM_ALWAYS_EXPORT void operator delete[
   ShimCppDelete(p);
 }
 
+extern "C" {
+
 ALIGN_LINKAGE SHIM_ALWAYS_EXPORT void* ALIGN_NEW(std::size_t size,
                                                  ALIGN_VAL_T alignment) {
   return ShimCppAlignedNew(size, static_cast<size_t>(alignment));
@@ -150,3 +152,5 @@ ALIGN_LINKAGE SHIM_ALWAYS_EXPORT void
 ALIGN_DEL_ARR_NOTHROW(void* p, ALIGN_VAL_T, const std::nothrow_t&) __THROW {
   ShimCppDelete(p);
 }
+
+}
