description: gcc 6 refuses to compile methods redeclared with noexcept
author: Michael Gilbert <mgilbert@debian.org>

--- a/media/learning/common/labelled_example.cc
+++ b/media/learning/common/labelled_example.cc
@@ -21,7 +21,7 @@ LabelledExample::LabelledExample(std::in
 
 LabelledExample::LabelledExample(const LabelledExample& rhs) = default;
 
-LabelledExample::LabelledExample(LabelledExample&& rhs) noexcept = default;
+LabelledExample::LabelledExample(LabelledExample&& rhs) = default;
 
 LabelledExample::~LabelledExample() = default;
 
--- a/components/omnibox/browser/autocomplete_match.cc
+++ b/components/omnibox/browser/autocomplete_match.cc
@@ -153,7 +153,7 @@ AutocompleteMatch::AutocompleteMatch(con
       additional_info(match.additional_info),
       duplicate_matches(match.duplicate_matches) { pedal = match.pedal; }
 
-AutocompleteMatch::AutocompleteMatch(AutocompleteMatch&& match) noexcept =
+AutocompleteMatch::AutocompleteMatch(AutocompleteMatch&& match) =
     default;
 
 AutocompleteMatch::~AutocompleteMatch() {
