description: gcc 6 does not support copy construction from a nullptr
author: Michael Gilbert <mgilbert@debian.org>

--- a/components/omnibox/browser/autocomplete_match.cc
+++ b/components/omnibox/browser/autocomplete_match.cc
@@ -144,7 +144,6 @@ AutocompleteMatch::AutocompleteMatch(con
                              : nullptr),
       keyword(match.keyword),
       from_keyword(match.from_keyword),
-      pedal(match.pedal),
       from_previous(match.from_previous),
       search_terms_args(
           match.search_terms_args
@@ -154,7 +153,7 @@ AutocompleteMatch::AutocompleteMatch(con
                        ? new TemplateURLRef::PostContent(*match.post_content)
                        : nullptr),
       additional_info(match.additional_info),
-      duplicate_matches(match.duplicate_matches) {}
+      duplicate_matches(match.duplicate_matches) { pedal = match.pedal; }
 
 AutocompleteMatch::AutocompleteMatch(AutocompleteMatch&& match) noexcept =
     default;
