description: gcc 6 handles Optional<flat_map<K,M,C>> incorrectly
origin: derived from a similar fix for WTF::Vector in wtf/vector.h
author: Michael Gilbert <mgilbert@debian.org>

--- a/base/containers/flat_map.h
+++ b/base/containers/flat_map.h
@@ -358,6 +358,13 @@ void flat_map<Key, Mapped, Compare>::swa
   tree::swap(other);
 }
 
+#if defined(__GNUC__) && !defined(__clang__) && __GNUC__ <= 7
+// Workaround for g++7 and earlier family.
+// Due to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80654.
+template <class K, class M, class C>
+struct is_trivially_copy_constructible<flat_map<K,M,C>> : std::false_type {};
+#endif
+
 }  // namespace base
 
 #endif  // BASE_CONTAINERS_FLAT_MAP_H_
