description: rules for building breakpad logging class are missing
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/breakpad/BUILD.gn
+++ b/third_party/breakpad/BUILD.gn
@@ -569,6 +569,8 @@ if (is_linux || is_android) {
       "breakpad/src/client/linux/dump_writer_common/thread_info.h",
       "breakpad/src/client/linux/dump_writer_common/ucontext_reader.cc",
       "breakpad/src/client/linux/dump_writer_common/ucontext_reader.h",
+      "breakpad/src/client/linux/log/log.cc",
+      "breakpad/src/client/linux/log/log.h",
       "breakpad/src/client/linux/handler/exception_handler.cc",
       "breakpad/src/client/linux/handler/exception_handler.h",
       "breakpad/src/client/linux/handler/minidump_descriptor.cc",
