description: backport from upstream glibc 2.27 / gcc 7.3 buildfixes form ARM
author: Riku Voipio

Index: chromium-browser-67.0.3396.62/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
===================================================================
--- chromium-browser-67.0.3396.62.orig/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
+++ chromium-browser-67.0.3396.62/third_party/crashpad/crashpad/compat/linux/sys/ptrace.h
@@ -34,7 +34,7 @@ static constexpr __ptrace_request PTRACE
 #endif  // !PTRACE_GET_THREAD_AREA && !PT_GET_THREAD_AREA && defined(__GLIBC__)
 
 // https://sourceware.org/bugzilla/show_bug.cgi?id=22433
-#if !defined(PTRACE_GETVFPREGS) && \
+#if !defined(PTRACE_GETVFPREGS) && !defined(PT_GETVFPREGS) && \
     defined(__GLIBC__) && (defined(__arm__) || defined(__arm64__))
 static constexpr __ptrace_request PTRACE_GETVFPREGS =
     static_cast<__ptrace_request>(27);
Index: chromium-browser-67.0.3396.62/third_party/crashpad/crashpad/util/misc/capture_context_linux.S
===================================================================
--- chromium-browser-67.0.3396.62.orig/third_party/crashpad/crashpad/util/misc/capture_context_linux.S
+++ chromium-browser-67.0.3396.62/third_party/crashpad/crashpad/util/misc/capture_context_linux.S
@@ -288,7 +288,7 @@ CAPTURECONTEXT_SYMBOL2:
 #elif defined(__aarch64__)
 
   // Zero out fault_address, which is unused.
-  str x31, [x0, #0xb0]  // context->uc_mcontext.fault_address
+  str xzr, [x0, #0xb0]  // context->uc_mcontext.fault_address
 
   // Save general purpose registers in context->uc_mcontext.regs[i].
   // The original x0 can't be recovered.
