description: need complete type for use in unique_ptr
author: Michael Gilbert <mgilbert@debian.org>

--- a/services/preferences/tracked/pref_hash_filter.h
+++ b/services/preferences/tracked/pref_hash_filter.h
@@ -22,8 +22,8 @@
 #include "services/preferences/tracked/hash_store_contents.h"
 #include "services/preferences/tracked/interceptable_pref_filter.h"
 #include "services/preferences/tracked/tracked_preference.h"
+#include "services/preferences/tracked/pref_hash_store.h"
 
-class PrefHashStore;
 class PrefService;
 
 namespace base {
