description: jumbo build has trouble with these namespaces
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/apps/platform_apps/api/sync_file_system/extension_sync_event_observer.h
+++ b/chrome/browser/apps/platform_apps/api/sync_file_system/extension_sync_event_observer.h
@@ -81,8 +81,10 @@ class ExtensionSyncEventObserver : publi
 }  // namespace api
 }  // namespace chrome_apps
 
+namespace extensions {
 template <>
-void extensions::BrowserContextKeyedAPIFactory<
+void BrowserContextKeyedAPIFactory<
     chrome_apps::api::ExtensionSyncEventObserver>::DeclareFactoryDependencies();
+}
 
 #endif  // CHROME_BROWSER_APPS_PLATFORM_APPS_API_SYNC_FILE_SYSTEM_EXTENSION_SYNC_EVENT_OBSERVER_H_
--- a/gin/v8_platform.cc
+++ b/gin/v8_platform.cc
@@ -273,10 +273,13 @@ base::LazyInstance<PageAllocator>::Leaky
 
 }  // namespace gin
 
+namespace base {
+namespace trace_event {
+
 // Allow std::unique_ptr<v8::ConvertableToTraceFormat> to be a valid
 // initialization value for trace macros.
 template <>
-struct base::trace_event::TraceValue::Helper<
+struct TraceValue::Helper<
     std::unique_ptr<v8::ConvertableToTraceFormat>> {
   static constexpr unsigned char kType = TRACE_VALUE_TYPE_CONVERTABLE;
   static inline void SetValue(
@@ -289,6 +292,9 @@ struct base::trace_event::TraceValue::He
   }
 };
 
+} // namespace trace_event
+} // namespace base
+
 namespace gin {
 
 class V8Platform::TracingControllerImpl : public v8::TracingController {
