description: fix linking errors
author: Michael Gilbert <mgilbert@debian.org>

--- a/base/location.h
+++ b/base/location.h
@@ -31,6 +31,9 @@ namespace base {
 #define SUPPORTS_LOCATION_BUILTINS 0
 #endif
 
+#undef SUPPORTS_LOCATION_BUILTINS
+#define SUPPORTS_LOCATION_BUILTINS 0
+
 // Location provides basic info where of an object was constructed, or was
 // significantly brought to life.
 class BASE_EXPORT Location {
