description: add include statements that are missing
author: Michael Gilbert <mgilbert@debian.org>

--- a/cc/base/list_container_helper.cc
+++ b/cc/base/list_container_helper.cc
@@ -5,6 +5,7 @@
 #include "cc/base/list_container_helper.h"
 
 #include <stddef.h>
+#include <string.h>
 
 #include <algorithm>
 #include <vector>
--- a/third_party/crashpad/crashpad/util/net/http_multipart_builder.cc
+++ b/third_party/crashpad/crashpad/util/net/http_multipart_builder.cc
@@ -16,6 +16,7 @@
 
 #include <sys/types.h>
 
+#include <cstring>
 #include <utility>
 #include <vector>
 
--- a/third_party/crashpad/crashpad/snapshot/minidump/minidump_context_converter.cc
+++ b/third_party/crashpad/crashpad/snapshot/minidump/minidump_context_converter.cc
@@ -17,6 +17,8 @@
 #include "base/stl_util.h"
 #include "minidump/minidump_context.h"
 
+#include <cstring>
+
 namespace crashpad {
 namespace internal {
 
--- a/third_party/crashpad/crashpad/util/process/process_memory.cc
+++ b/third_party/crashpad/crashpad/util/process/process_memory.cc
@@ -14,6 +14,7 @@
 
 #include "util/process/process_memory.h"
 
+#include <cstring>
 #include <algorithm>
 
 #include "base/logging.h"
--- a/third_party/crashpad/crashpad/util/stream/log_output_stream.cc
+++ b/third_party/crashpad/crashpad/util/stream/log_output_stream.cc
@@ -14,6 +14,7 @@
 
 #include "util/stream/log_output_stream.h"
 
+#include <cstring>
 #include <algorithm>
 
 #include "base/logging.h"
--- a/third_party/crashpad/crashpad/util/net/http_transport_socket.cc
+++ b/third_party/crashpad/crashpad/util/net/http_transport_socket.cc
@@ -14,6 +14,7 @@
 
 #include "util/net/http_transport.h"
 
+#include <string.h>
 #include <fcntl.h>
 #include <netdb.h>
 #include <poll.h>
--- a/third_party/crashpad/crashpad/util/linux/ptrace_client.cc
+++ b/third_party/crashpad/crashpad/util/linux/ptrace_client.cc
@@ -16,6 +16,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <string.h>
 
 #include <string>
 
--- a/content/shell/browser/shell_platform_delegate.h
+++ b/content/shell/browser/shell_platform_delegate.h
@@ -11,6 +11,8 @@
 #include "ui/gfx/geometry/size.h"
 #include "ui/gfx/native_widget_types.h"
 
+#include <memory>
+
 #if defined(OS_MACOSX)
 #include "content/public/browser/native_web_keyboard_event.h"
 #endif
--- a/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h
+++ b/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h
@@ -12,6 +12,7 @@
 #define MODULES_AUDIO_PROCESSING_AEC3_CLOCKDRIFT_DETECTOR_H_
 
 #include <array>
+#include <cstddef>
 
 namespace webrtc {
 
--- a/chrome/browser/search/background/ntp_backgrounds.h
+++ b/chrome/browser/search/background/ntp_backgrounds.h
@@ -6,6 +6,7 @@
 #define CHROME_BROWSER_SEARCH_BACKGROUND_NTP_BACKGROUNDS_H_
 
 #include <array>
+#include <cstddef>
 
 class GURL;
 
