description: add include statements that are missing
author: Michael Gilbert <mgilbert@debian.org>

--- a/cc/base/list_container_helper.cc
+++ b/cc/base/list_container_helper.cc
@@ -5,6 +5,7 @@
 #include "cc/base/list_container_helper.h"
 
 #include <stddef.h>
+#include <string.h>
 
 #include <algorithm>
 #include <cstring>
--- a/third_party/crashpad/crashpad/util/net/http_multipart_builder.cc
+++ b/third_party/crashpad/crashpad/util/net/http_multipart_builder.cc
@@ -17,6 +17,7 @@
 #include <string.h>
 #include <sys/types.h>
 
+#include <cstring>
 #include <utility>
 #include <vector>
 
--- a/third_party/crashpad/crashpad/snapshot/minidump/minidump_context_converter.cc
+++ b/third_party/crashpad/crashpad/snapshot/minidump/minidump_context_converter.cc
@@ -20,6 +20,8 @@
 #include "base/stl_util.h"
 #include "minidump/minidump_context.h"
 
+#include <cstring>
+
 namespace crashpad {
 namespace internal {
 
--- a/third_party/crashpad/crashpad/util/process/process_memory.cc
+++ b/third_party/crashpad/crashpad/util/process/process_memory.cc
@@ -16,6 +16,7 @@
 
 #include <string.h>
 
+#include <cstring>
 #include <algorithm>
 
 #include "base/check_op.h"
--- a/third_party/crashpad/crashpad/util/stream/log_output_stream.cc
+++ b/third_party/crashpad/crashpad/util/stream/log_output_stream.cc
@@ -16,6 +16,7 @@
 
 #include <string.h>
 
+#include <cstring>
 #include <algorithm>
 
 #include "base/check.h"
--- a/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h
+++ b/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h
@@ -14,6 +14,7 @@
 #include <stddef.h>
 
 #include <array>
+#include <cstddef>
 
 namespace webrtc {
 
--- a/chrome/browser/search/background/ntp_backgrounds.h
+++ b/chrome/browser/search/background/ntp_backgrounds.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 
 #include <array>
+#include <cstddef>
 
 class GURL;
 
--- a/components/autofill/core/browser/pattern_provider/pattern_provider.h
+++ b/components/autofill/core/browser/pattern_provider/pattern_provider.h
@@ -6,6 +6,7 @@
 #define COMPONENTS_AUTOFILL_CORE_BROWSER_PATTERN_PROVIDER_PATTERN_PROVIDER_H_
 
 #include <string>
+#include <vector>
 
 #include "base/macros.h"
 #include "base/no_destructor.h"
@@ -55,4 +56,4 @@
   friend class base::NoDestructor<PatternProvider>;
 };
 }  // namespace autofill
-#endif  // COMPONENTS_AUTOFILL_CORE_BROWSER_PATTERN_PROVIDER_PATTERN_PROVIDER_H_
\ No newline at end of file
+#endif  // COMPONENTS_AUTOFILL_CORE_BROWSER_PATTERN_PROVIDER_PATTERN_PROVIDER_H_
--- a/components/autofill/core/browser/data_model/borrowed_transliterator.h
+++ b/components/autofill/core/browser/data_model/borrowed_transliterator.h
@@ -5,6 +5,8 @@
 #ifndef COMPONENTS_AUTOFILL_CORE_BROWSER_DATA_MODEL_BORROWED_TRANSLITERATOR_H_
 #define COMPONENTS_AUTOFILL_CORE_BROWSER_DATA_MODEL_BORROWED_TRANSLITERATOR_H_
 
+#include <memory>
+
 #include "base/i18n/unicodestring.h"
 #include "base/no_destructor.h"
 #include "base/synchronization/lock.h"
