--- a/chrome/common/string_matching/fuzzy_tokenized_string_match.cc
+++ b/chrome/common/string_matching/fuzzy_tokenized_string_match.cc
@@ -6,6 +6,7 @@
 
 #include <algorithm>
 #include <iterator>
+#include <cmath>
 
 #include "base/i18n/case_conversion.h"
 #include "base/metrics/field_trial_params.h"
--- a/content/public/test/back_forward_cache_util.h
+++ b/content/public/test/back_forward_cache_util.h
@@ -7,6 +7,8 @@
 
 #include "base/strings/string_piece.h"
 
+#include <memory>
+
 namespace content {
 class BackForwardCacheImpl;
 class RenderFrameHost;
