description: add include statements that are missing
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/public/test/back_forward_cache_util.h
+++ b/content/public/test/back_forward_cache_util.h
@@ -7,6 +7,8 @@
 
 #include "base/strings/string_piece.h"
 
+#include <memory>
+
 namespace content {
 class BackForwardCacheImpl;
 class RenderFrameHost;
--- a/base/test/icu_test_util.h
+++ b/base/test/icu_test_util.h
@@ -5,6 +5,7 @@
 #ifndef BASE_TEST_ICU_TEST_UTIL_H_
 #define BASE_TEST_ICU_TEST_UTIL_H_
 
+#include <memory>
 #include <string>
 
 #include "base/macros.h"
