--- a/third_party/blink/renderer/core/layout/ng/inline/ng_inline_node.cc
+++ b/third_party/blink/renderer/core/layout/ng/inline/ng_inline_node.cc
@@ -813,7 +813,7 @@ void NGInlineNode::SegmentText(NGInlineN
 
 // Segment NGInlineItem by script, Emoji, and orientation using RunSegmenter.
 void NGInlineNode::SegmentScriptRuns(NGInlineNodeData* data) {
-  DCHECK_EQ(data->segments, nullptr);
+  DCHECK_EQ(&data->segments, nullptr);
 
   String& text_content = data->text_content;
   if (text_content.IsEmpty()) {
