description: disable flags that are introduced after clang 9
author: Michael Gilbert <mgilbert@debian.org>

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1515,24 +1515,6 @@
           # TODO(thakis): Only for no_chromium_code? http://crbug.com/912662
           "-Wno-ignored-pragma-optimize",
 
-          # TODO(https://crbug.com/989932): Evaluate and possibly enable.
-          "-Wno-implicit-int-float-conversion",
-
-          # TODO(https://crbug.com/999886): Clean up, enable.
-          "-Wno-final-dtor-non-final-class",
-
-          # TODO(https://crbug.com/1016947) Clean up, enable.
-          "-Wno-bitwise-conditional-parentheses",
-
-          # TODO(https://crbug.com/1016945) Clean up, enable.
-          "-Wno-builtin-assume-aligned-alignment",
-
-          # TODO(https://crbug.com/1028110): Evaluate and possible enable.
-          "-Wno-deprecated-copy",
-
-          # TODO(https://crbug.com/1031169): Clean up and enable.
-          "-Wno-misleading-indentation",
-
           # TODO(https://crbug.com/1039697): Evaluate and possibly enable.
           "-Wno-range-loop-analysis",
         ]
--- a/third_party/blink/renderer/BUILD.gn	2020-04-02 23:11:35.000000000 -0500
+++ b/third_party/blink/renderer/BUILD.gn	2020-04-15 14:38:57.650905797 -0500
@@ -51,8 +51,6 @@
 
     if (!is_chromeos || default_toolchain != "//build/toolchain/cros:target") {
       cflags += [
-        "-Wno-enum-enum-conversion",
-        "-Wno-enum-float-conversion",
       ]
     }
   }
