--- b/debian/scripts/mk-origtargz
+++ a/debian/scripts/mk-origtargz
@@ -213,6 +213,11 @@
     }
 
     if ($deletecount or $self->config->force_repack) {
+        for my $filename (@to_delete) {
+            printf "%s\n", $filename;
+        }
+        return $self->status(0);
+
         $destfilebase = sprintf "%s_%s%s.%s.tar", $self->config->package,
           $self->config->version, $self->config->repack_suffix,
           $self->config->orig;
@@ -579,4 +584,4 @@
     return sprintf "%s.%s", $destfiletar, $comp;
 }
 
-1;
+exit Devscripts::MkOrigtargz->new->do;
