description: disable build commands for embedded fontconfig
author: Michael Gilbert <mgilbert@debian.org>

--- a/base/test/BUILD.gn
+++ b/base/test/BUILD.gn
@@ -210,7 +210,7 @@ static_library("test_support") {
     deps += [ "//third_party/fuchsia-sdk/sdk:zx" ]
   }
 
-  if (is_linux) {
+  if (false) {
     public_deps += [ ":fontconfig_util_linux" ]
     data_deps = [
       "//third_party/test_fonts",
@@ -396,7 +396,7 @@ if (is_linux) {
     ]
   }
 
-  if (current_toolchain == host_toolchain) {
+  if (false) {
     executable("generate_fontconfig_caches") {
       testonly = true
       sources = [
