description: use system harfbuzz
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/skia/src/pdf/SkPDFSubsetFont.cpp
+++ b/third_party/skia/src/pdf/SkPDFSubsetFont.cpp
@@ -7,6 +7,9 @@
 #include "SkLoadICU.h"
 #endif
 
+#undef SK_PDF_USE_SFNTLY
+#undef SK_PDF_USE_HARFBUZZ_SUBSET
+
 #if defined(SK_PDF_USE_HARFBUZZ_SUBSET)
 
 #include "include/private/SkTemplates.h"
--- a/third_party/harfbuzz-ng/BUILD.gn
+++ b/third_party/harfbuzz-ng/BUILD.gn
@@ -19,7 +19,6 @@ if (use_system_harfbuzz) {
     ]
     packages = [
       "harfbuzz",
-      "harfbuzz-subset",
     ]
   }
 } else {
--- a/components/paint_preview/common/subset_font.cc
+++ b/components/paint_preview/common/subset_font.cc
@@ -7,8 +7,7 @@
 #include <memory>
 #include <utility>
 
-#include <hb-subset.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 #include "base/bind.h"
 #include "base/callback.h"
--- a/components/paint_preview/common/serial_utils.cc
+++ b/components/paint_preview/common/serial_utils.cc
@@ -50,12 +50,8 @@ sk_sp<SkData> SerializeTypeface(SkTypefa
   if (usage_it == context->usage->end())
     return typeface->serialize(SkTypeface::SerializeBehavior::kDontIncludeData);
 
-  auto subset_data = SubsetFont(typeface, *usage_it->second);
-  // This will fail if the font cannot be subsetted properly. In such cases
-  // all typeface data should be added for portability.
-  if (!subset_data)
+  if (true)
     return typeface->serialize(SkTypeface::SerializeBehavior::kDoIncludeData);
-  return subset_data;
 }
 
 // Deserializies a clip rect for a subframe within the main SkPicture. These
--- a/components/paint_preview/common/BUILD.gn
+++ b/components/paint_preview/common/BUILD.gn
@@ -17,8 +17,6 @@ if (!is_ios) {
       "paint_preview_tracker.h",
       "serial_utils.cc",
       "serial_utils.h",
-      "subset_font.cc",
-      "subset_font.h",
     ]
 
     deps = [
--- a/third_party/skia/BUILD.gn
+++ b/third_party/skia/BUILD.gn
@@ -680,7 +680,6 @@ optional("pdf") {
   sources_when_disabled = [ "src/pdf/SkDocument_PDF_None.cpp" ]
   if (skia_use_icu && skia_use_harfbuzz && skia_pdf_subset_harfbuzz) {
     deps += [ "//third_party/harfbuzz" ]
-    defines = [ "SK_PDF_USE_HARFBUZZ_SUBSET" ]
   } else if (skia_use_icu && skia_use_sfntly) {
     deps += [ "//third_party/sfntly" ]
     defines = [ "SK_PDF_USE_SFNTLY" ]
