description: use system harfbuzz
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/skia/src/pdf/SkPDFSubsetFont.cpp
+++ b/third_party/skia/src/pdf/SkPDFSubsetFont.cpp
@@ -7,7 +7,7 @@
 #include "SkLoadICU.h"
 #endif
 
-#if defined(SK_PDF_USE_HARFBUZZ_SUBSET)
+#if 0
 
 #include "include/private/SkTemplates.h"
 #include "include/private/SkTo.h"
@@ -135,7 +135,7 @@ static sk_sp<SkData> subset_sfntly(sk_sp
 
 ////////////////////////////////////////////////////////////////////////////////
 
-#if defined(SK_PDF_USE_SFNTLY) && defined(SK_PDF_USE_HARFBUZZ_SUBSET)
+#if 0
 
 sk_sp<SkData> SkPDFSubsetFont(sk_sp<SkData> fontData,
                               const SkPDFGlyphUse& glyphUsage,
--- a/third_party/harfbuzz-ng/BUILD.gn
+++ b/third_party/harfbuzz-ng/BUILD.gn
@@ -18,7 +18,6 @@ if (use_system_harfbuzz) {
     ]
     packages = [
       "harfbuzz",
-      "harfbuzz-subset",
     ]
   }
 } else {
--- a/components/paint_preview/common/subset_font.cc
+++ b/components/paint_preview/common/subset_font.cc
@@ -7,8 +7,7 @@
 #include <memory>
 #include <utility>
 
-#include <hb-subset.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 #include "base/bind.h"
 #include "base/callback.h"
--- a/components/paint_preview/common/serial_utils.cc
+++ b/components/paint_preview/common/serial_utils.cc
@@ -50,12 +50,8 @@ sk_sp<SkData> SerializeTypeface(SkTypefa
   if (usage_it == context->usage->end())
     return typeface->serialize(SkTypeface::SerializeBehavior::kDontIncludeData);
 
-  auto subset_data = SubsetFont(typeface, *usage_it->second);
-  // This will fail if the font cannot be subsetted properly. In such cases
-  // all typeface data should be added for portability.
-  if (!subset_data)
+  if (true)
     return typeface->serialize(SkTypeface::SerializeBehavior::kDoIncludeData);
-  return subset_data;
 }
 
 // Deserializies a clip rect for a subframe within the main SkPicture. These
--- a/components/paint_preview/common/BUILD.gn
+++ b/components/paint_preview/common/BUILD.gn
@@ -15,8 +15,6 @@ if (!is_ios) {
       "paint_preview_tracker.h",
       "serial_utils.cc",
       "serial_utils.h",
-      "subset_font.cc",
-      "subset_font.h",
     ]
 
     deps = [
