description: system xml
author: Michael Gilbert <mgilbert@debian.org>

--- a/build/linux/unbundle/libxml.gn
+++ b/build/linux/unbundle/libxml.gn
@@ -16,3 +16,31 @@ static_library("libxml") {
 
   public_configs = [ ":system_libxml" ]
 }
+
+static_library("libxml_utils") {
+  deps = [
+    ":libxml",
+  ]
+}
+
+static_library("xml_reader") {
+  sources = [
+    "chromium/xml_reader.cc",
+    "chromium/xml_reader.h",
+  ]
+
+  deps = [
+    ":libxml",
+  ]
+}
+
+static_library("xml_writer") {
+  sources = [
+    "chromium/xml_writer.cc",
+    "chromium/xml_writer.h",
+  ]
+
+  deps = [
+    ":libxml",
+  ]
+}
--- a/third_party/libxml/chromium/libxml_utils.h
+++ b/third_party/libxml/chromium/libxml_utils.h
@@ -7,7 +7,7 @@
 
 #include <string>
 
-#include "third_party/libxml/src/include/libxml/xmlreader.h"
+#include <libxml/xmlerror.h>
 
 // libxml uses a global error function pointer for reporting errors.
 // A ScopedXmlErrorFunc object lets you change the global error pointer
--- a/third_party/libxml/chromium/xml_reader.cc
+++ b/third_party/libxml/chromium/xml_reader.cc
@@ -7,7 +7,7 @@
 #include <vector>
 
 #include "third_party/libxml/chromium/libxml_utils.h"
-#include "third_party/libxml/src/include/libxml/xmlreader.h"
+#include <libxml/xmlreader.h>
 
 using internal::XmlStringToStdString;
 
--- a/third_party/libxml/chromium/xml_writer.cc
+++ b/third_party/libxml/chromium/xml_writer.cc
@@ -5,7 +5,7 @@
 #include "third_party/libxml/chromium/xml_writer.h"
 
 #include "third_party/libxml/chromium/libxml_utils.h"
-#include "third_party/libxml/src/include/libxml/xmlwriter.h"
+#include <libxml/xmlwriter.h>
 
 XmlWriter::XmlWriter() : writer_(nullptr), buffer_(nullptr) {}
 
--- a/base/test/BUILD.gn
+++ b/base/test/BUILD.gn
@@ -209,6 +209,8 @@ static_library("test_support") {
 
   configs += [ "//build/config:precompiled_headers" ]
 
+  public_configs = [ "//third_party/libxml:system_libxml" ]
+
   data = [
     # The isolate needs this script for setting up the test. It's not actually
     # needed to run this target locally.
