description: build using system openjpeg
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/pdfium/core/fpdfapi/render/BUILD.gn
+++ b/third_party/pdfium/core/fpdfapi/render/BUILD.gn
@@ -46,6 +46,7 @@ source_set("render") {
   ]
   configs += [ "../../../:pdfium_core_config" ]
   deps = [
+    "../../../third_party:libopenjpeg2",
     "../../../constants",
     "../../fxcodec",
     "../../fxcrt",
--- a/third_party/pdfium/core/fxcodec/jpx/cjpx_decoder.h
+++ b/third_party/pdfium/core/fxcodec/jpx/cjpx_decoder.h
@@ -12,8 +12,8 @@
 #include "core/fxcrt/unowned_ptr.h"
 #include "third_party/base/span.h"
 
-#if defined(USE_SYSTEM_LIBOPENJPEG2)
-#include <openjpeg.h>
+#if 1
+#include <openjpeg-2.3/openjpeg.h>
 #else
 #include "third_party/libopenjpeg20/openjpeg.h"
 #endif
