description: build using system openjpeg
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/pdfium/core/fpdfapi/render/BUILD.gn
+++ b/third_party/pdfium/core/fpdfapi/render/BUILD.gn
@@ -51,6 +51,7 @@ jumbo_source_set("render") {
   ]
   configs += [ "../../../:pdfium_core_config" ]
   deps = [
+    "../../../third_party:libopenjpeg2",
     "../../../constants",
     "../../fxcodec",
     "../../fxcrt",
