description: system vpx
author: Michael Gilbert <mgilbert@debian.org>

--- a/media/base/supported_types.cc
+++ b/media/base/supported_types.cc
@@ -20,8 +20,8 @@
 #if BUILDFLAG(ENABLE_LIBVPX)
 // TODO(dalecurtis): This technically should not be allowed in media/base. See
 // TODO below about moving outside of base.
-#include "third_party/libvpx/source/libvpx/vpx/vp8dx.h"
-#include "third_party/libvpx/source/libvpx/vpx/vpx_codec.h"
+#include <vpx/vp8dx.h>
+#include <vpx/vpx_codec.h>
 #endif
 
 #if defined(OS_ANDROID)
--- a/media/gpu/vaapi/BUILD.gn
+++ b/media/gpu/vaapi/BUILD.gn
@@ -71,8 +71,6 @@ source_set("vaapi") {
     "vp8_vaapi_video_decoder_delegate.h",
     "vp9_encoder.cc",
     "vp9_encoder.h",
-    "vp9_rate_control.cc",
-    "vp9_rate_control.h",
     "vp9_temporal_layers.cc",
     "vp9_temporal_layers.h",
     "vp9_vaapi_video_decoder_delegate.cc",
@@ -87,7 +85,6 @@ source_set("vaapi") {
 
   configs += [
     "//build/config/linux/libva",
-    "//third_party/libvpx:libvpx_config",
   ]
 
   deps = [
@@ -102,7 +99,6 @@ source_set("vaapi") {
     "//media/gpu/chromeos:common",
     "//media/parsers",
     "//mojo/public/cpp/bindings",
-    "//third_party/libvpx:libvp9rc",
     "//third_party/libyuv",
     "//ui/gfx",
     "//ui/gfx/geometry",
@@ -227,7 +223,6 @@ source_set("unit_test") {
     "vp9_encoder_unittest.cc",
     "vp9_temporal_layers_unittest.cc",
   ]
-  configs += [ "//third_party/libvpx:libvpx_config" ]
   deps = [
     ":common",
     ":vaapi",
@@ -239,7 +234,6 @@ source_set("unit_test") {
     "//mojo/core/embedder",
     "//testing/gmock",
     "//testing/gtest",
-    "//third_party/libvpx:libvp9rc",
     "//ui/gfx:test_support",
     "//ui/gfx/geometry",
   ]
--- a/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.h
+++ b/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.h
@@ -15,10 +15,11 @@
 
 #include <memory>
 
-#include "vpx/vp8cx.h"
-#include "vpx/vpx_codec.h"
-#include "vpx/vpx_encoder.h"
-#include "vpx/vpx_image.h"
+#include <vpx/vp8cx.h>
+#include <vpx/vpx_codec.h>
+#include <vpx/vpx_encoder.h>
+#include <vpx/vpx_image.h>
+#include "vpx_ext_ratectrl.h"
 
 namespace webrtc {
 
--- a/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/interface/libvpx_interface.cc
@@ -170,11 +170,6 @@ class LibvpxFacade : public LibvpxInterface {
         return vpx_codec_control(ctx, VP9E_SET_COLOR_RANGE, param);
       case VP9E_SET_DELTA_Q_UV:
         return vpx_codec_control(ctx, VP9E_SET_DELTA_Q_UV, param);
-      case VP9E_SET_DISABLE_OVERSHOOT_MAXQ_CBR:
-        return vpx_codec_control(ctx, VP9E_SET_DISABLE_OVERSHOOT_MAXQ_CBR,
-                                 param);
-      case VP9E_SET_DISABLE_LOOPFILTER:
-        return vpx_codec_control(ctx, VP9E_SET_DISABLE_LOOPFILTER, param);
 
       default:
         if (param >= 0) {
@@ -328,8 +323,6 @@ class LibvpxFacade : public LibvpxInterface {
                                 vp8e_enc_control_id ctrl_id,
                                 vpx_rc_funcs_t* param) const override {
     switch (ctrl_id) {
-      case VP9E_SET_EXTERNAL_RATE_CONTROL:
-        return vpx_codec_control_(ctx, VP9E_SET_EXTERNAL_RATE_CONTROL, param);
       default:
         RTC_NOTREACHED() << "Unsupported libvpx ctrl_id: " << ctrl_id;
     }
--- /dev/null
+++ b/third_party/webrtc/modules/video_coding/codecs/interface/vpx_ext_ratectrl.h
@@ -0,0 +1,360 @@
+/*
+ *  Copyright (c) 2020 The WebM project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree. An additional intellectual property rights grant can be found
+ *  in the file PATENTS.  All contributing project authors may
+ *  be found in the AUTHORS file in the root of the source tree.
+ */
+
+#ifndef VPX_VPX_VPX_EXT_RATECTRL_H_
+#define VPX_VPX_VPX_EXT_RATECTRL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <vpx/vpx_integer.h>
+
+/*!\brief Current ABI version number
+ *
+ * \internal
+ * If this file is altered in any way that changes the ABI, this value
+ * must be bumped. Examples include, but are not limited to, changing
+ * types, removing or reassigning enums, adding/removing/rearranging
+ * fields to structures.
+ */
+#define VPX_EXT_RATECTRL_ABI_VERSION (1)
+
+/*!\brief Abstract rate control model handler
+ *
+ * The encoder will receive the model handler from create_model() defined in
+ * vpx_rc_funcs_t.
+ */
+typedef void *vpx_rc_model_t;
+
+/*!\brief Encode frame decision made by the external rate control model
+ *
+ * The encoder will receive the decision from the external rate control model
+ * through get_encodeframe_decision() defined in vpx_rc_funcs_t.
+ *
+ * If max_frame_size = 0, the encoding ignores max frame size limit.
+ * If max_frame_size = -1, the encoding uses VP9's max frame size as the limit.
+ * If the encoded frame size is larger than max_frame_size, the frame is
+ * recoded to meet the size limit, following VP9's recoding principles.
+ */
+typedef struct vpx_rc_encodeframe_decision {
+  int q_index;        /**< Quantizer step index [0..255]*/
+  int max_frame_size; /**< Maximal frame size allowed to encode a frame*/
+} vpx_rc_encodeframe_decision_t;
+
+/*!\brief Information for the frame to be encoded.
+ *
+ * The encoder will send the information to external rate control model through
+ * get_encodeframe_decision() defined in vpx_rc_funcs_t.
+ *
+ */
+typedef struct vpx_rc_encodeframe_info {
+  /*!
+   * 0: Key frame
+   * 1: Inter frame
+   * 2: Alternate reference frame
+   * 3: Overlay frame
+   * 4: Golden frame
+   */
+  int frame_type;
+  int show_index;   /**< display index, starts from zero*/
+  int coding_index; /**< coding index, starts from zero*/
+  /*!
+   * index in group of picture, starts from zero.
+   */
+  int gop_index;
+  int ref_frame_coding_indexes[3]; /**< three reference frames' coding indices*/
+  /*!
+   * The validity of the three reference frames.
+   * 0: Invalid
+   * 1: Valid
+   */
+  int ref_frame_valid_list[3];
+} vpx_rc_encodeframe_info_t;
+
+/*!\brief Frame coding result
+ *
+ * The encoder will send the result to the external rate control model through
+ * update_encodeframe_result() defined in vpx_rc_funcs_t.
+ */
+typedef struct vpx_rc_encodeframe_result {
+  int64_t sse;         /**< sum of squared error of the reconstructed frame */
+  int64_t bit_count;   /**< number of bits spent on coding the frame*/
+  int64_t pixel_count; /**< number of pixels in YUV planes of the frame*/
+  int actual_encoding_qindex; /**< the actual qindex used to encode the frame*/
+} vpx_rc_encodeframe_result_t;
+
+/*!\brief Status returned by rate control callback functions.
+ */
+typedef enum vpx_rc_status {
+  VPX_RC_OK = 0,
+  VPX_RC_ERROR = 1,
+} vpx_rc_status_t;
+
+/*!\brief First pass frame stats
+ * This is a mirror of vp9's FIRSTPASS_STATS except that spatial_layer_id is
+ * omitted
+ */
+typedef struct vpx_rc_frame_stats {
+  /*!
+   * Frame number in display order, if stats are for a single frame.
+   * No real meaning for a collection of frames.
+   */
+  double frame;
+  /*!
+   * Weight assigned to this frame (or total weight for the collection of
+   * frames) currently based on intra factor and brightness factor. This is used
+   * to distribute bits between easier and harder frames.
+   */
+  double weight;
+  /*!
+   * Intra prediction error.
+   */
+  double intra_error;
+  /*!
+   * Best of intra pred error and inter pred error using last frame as ref.
+   */
+  double coded_error;
+  /*!
+   * Best of intra pred error and inter pred error using golden frame as ref.
+   */
+  double sr_coded_error;
+  /*!
+   * Estimate the noise energy of the current frame.
+   */
+  double frame_noise_energy;
+  /*!
+   * Percentage of blocks with inter pred error < intra pred error.
+   */
+  double pcnt_inter;
+  /*!
+   * Percentage of blocks using (inter prediction and) non-zero motion vectors.
+   */
+  double pcnt_motion;
+  /*!
+   * Percentage of blocks where golden frame was better than last or intra:
+   * inter pred error using golden frame < inter pred error using last frame and
+   * inter pred error using golden frame < intra pred error
+   */
+  double pcnt_second_ref;
+  /*!
+   * Percentage of blocks where intra and inter prediction errors were very
+   * close.
+   */
+  double pcnt_neutral;
+  /*!
+   * Percentage of blocks that have intra error < inter error and inter error <
+   * LOW_I_THRESH
+   * - bit_depth 8: LOW_I_THRESH = 24000
+   * - bit_depth 10: LOW_I_THRESH = 24000 << 4
+   * - bit_depth 12: LOW_I_THRESH = 24000 << 8
+   */
+  double pcnt_intra_low;
+  /*!
+   * Percentage of blocks that have intra error < inter error and intra error <
+   * LOW_I_THRESH but inter error >= LOW_I_THRESH LOW_I_THRESH
+   * - bit_depth 8: LOW_I_THRESH = 24000
+   * - bit_depth 10: LOW_I_THRESH = 24000 << 4
+   * - bit_depth 12: LOW_I_THRESH = 24000 << 8
+   */
+  double pcnt_intra_high;
+  /*!
+   * Percentage of blocks that have almost no intra error residual
+   * (i.e. are in effect completely flat and untextured in the intra
+   * domain). In natural videos this is uncommon, but it is much more
+   * common in animations, graphics and screen content, so may be used
+   * as a signal to detect these types of content.
+   */
+  double intra_skip_pct;
+  /*!
+   * Percentage of blocks that have intra error < SMOOTH_INTRA_THRESH
+   * - bit_depth 8:  SMOOTH_INTRA_THRESH = 4000
+   * - bit_depth 10: SMOOTH_INTRA_THRESH = 4000 << 4
+   * - bit_depth 12: SMOOTH_INTRA_THRESH = 4000 << 8
+   */
+  double intra_smooth_pct;
+  /*!
+   * Image mask rows top and bottom.
+   */
+  double inactive_zone_rows;
+  /*!
+   * Image mask columns at left and right edges.
+   */
+  double inactive_zone_cols;
+  /*!
+   * Mean of row motion vectors.
+   */
+  double MVr;
+  /*!
+   * Mean of absolute value of row motion vectors.
+   */
+  double mvr_abs;
+  /*!
+   * Mean of column motion vectors.
+   */
+  double MVc;
+  /*!
+   * Mean of absolute value of column motion vectors.
+   */
+  double mvc_abs;
+  /*!
+   * Variance of row motion vectors.
+   */
+  double MVrv;
+  /*!
+   * Variance of column motion vectors.
+   */
+  double MVcv;
+  /*!
+   * Value in range [-1,1] indicating fraction of row and column motion vectors
+   * that point inwards (negative MV value) or outwards (positive MV value).
+   * For example, value of 1 indicates, all row/column MVs are inwards.
+   */
+  double mv_in_out_count;
+  /*!
+   * Duration of the frame / collection of frames.
+   */
+  double duration;
+  /*!
+   * 1.0 if stats are for a single frame, or
+   * number of frames whose stats are accumulated.
+   */
+  double count;
+} vpx_rc_frame_stats_t;
+
+/*!\brief Collection of first pass frame stats
+ */
+typedef struct vpx_rc_firstpass_stats {
+  /*!
+   * Pointer to first pass frame stats.
+   * The pointed array of vpx_rc_frame_stats_t should have length equal to
+   * number of show frames in the video.
+   */
+  vpx_rc_frame_stats_t *frame_stats;
+  /*!
+   * Number of show frames in the video.
+   */
+  int num_frames;
+} vpx_rc_firstpass_stats_t;
+
+/*!\brief Encode config sent to external rate control model
+ */
+typedef struct vpx_rc_config {
+  int frame_width;      /**< frame width */
+  int frame_height;     /**< frame height */
+  int show_frame_count; /**< number of visible frames in the video */
+  /*!
+   * Target bitrate in kilobytes per second
+   */
+  int target_bitrate_kbps;
+  int frame_rate_num; /**< numerator of frame rate */
+  int frame_rate_den; /**< denominator of frame rate */
+} vpx_rc_config_t;
+
+/*!\brief Create an external rate control model callback prototype
+ *
+ * This callback is invoked by the encoder to create an external rate control
+ * model.
+ *
+ * \param[in]  priv               Callback's private data
+ * \param[in]  ratectrl_config    Pointer to vpx_rc_config_t
+ * \param[out] rate_ctrl_model_pt Pointer to vpx_rc_model_t
+ */
+typedef vpx_rc_status_t (*vpx_rc_create_model_cb_fn_t)(
+    void *priv, const vpx_rc_config_t *ratectrl_config,
+    vpx_rc_model_t *rate_ctrl_model_pt);
+
+/*!\brief Send first pass stats to the external rate control model callback
+ * prototype
+ *
+ * This callback is invoked by the encoder to send first pass stats to the
+ * external rate control model.
+ *
+ * \param[in]  rate_ctrl_model    rate control model
+ * \param[in]  first_pass_stats   first pass stats
+ */
+typedef vpx_rc_status_t (*vpx_rc_send_firstpass_stats_cb_fn_t)(
+    vpx_rc_model_t rate_ctrl_model,
+    const vpx_rc_firstpass_stats_t *first_pass_stats);
+
+/*!\brief Receive encode frame decision callback prototype
+ *
+ * This callback is invoked by the encoder to receive encode frame decision from
+ * the external rate control model.
+ *
+ * \param[in]  rate_ctrl_model    rate control model
+ * \param[in]  encode_frame_info  information of the coding frame
+ * \param[out] frame_decision     encode decision of the coding frame
+ */
+typedef vpx_rc_status_t (*vpx_rc_get_encodeframe_decision_cb_fn_t)(
+    vpx_rc_model_t rate_ctrl_model,
+    const vpx_rc_encodeframe_info_t *encode_frame_info,
+    vpx_rc_encodeframe_decision_t *frame_decision);
+
+/*!\brief Update encode frame result callback prototype
+ *
+ * This callback is invoked by the encoder to update encode frame result to the
+ * external rate control model.
+ *
+ * \param[in]  rate_ctrl_model     rate control model
+ * \param[out] encode_frame_result encode result of the coding frame
+ */
+typedef vpx_rc_status_t (*vpx_rc_update_encodeframe_result_cb_fn_t)(
+    vpx_rc_model_t rate_ctrl_model,
+    const vpx_rc_encodeframe_result_t *encode_frame_result);
+
+/*!\brief Delete the external rate control model callback prototype
+ *
+ * This callback is invoked by the encoder to delete the external rate control
+ * model.
+ *
+ * \param[in]  rate_ctrl_model     rate control model
+ */
+typedef vpx_rc_status_t (*vpx_rc_delete_model_cb_fn_t)(
+    vpx_rc_model_t rate_ctrl_model);
+
+/*!\brief Callback function set for external rate control.
+ *
+ * The user can enable external rate control by registering
+ * a set of callback functions with the codec control flag
+ * VP9E_SET_EXTERNAL_RATE_CONTROL.
+ */
+typedef struct vpx_rc_funcs {
+  /*!
+   * Create an external rate control model.
+   */
+  vpx_rc_create_model_cb_fn_t create_model;
+  /*!
+   * Send first pass stats to the external rate control model.
+   */
+  vpx_rc_send_firstpass_stats_cb_fn_t send_firstpass_stats;
+  /*!
+   * Get encodeframe decision from the external rate control model.
+   */
+  vpx_rc_get_encodeframe_decision_cb_fn_t get_encodeframe_decision;
+  /*!
+   * Update encodeframe result to the external rate control model.
+   */
+  vpx_rc_update_encodeframe_result_cb_fn_t update_encodeframe_result;
+  /*!
+   * Delete the external rate control model.
+   */
+  vpx_rc_delete_model_cb_fn_t delete_model;
+  /*!
+   * Private data for the external rate control model.
+   */
+  void *priv;
+} vpx_rc_funcs_t;
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif
+
+#endif  // VPX_VPX_VPX_EXT_RATECTRL_H_
--- a/media/filters/vpx_video_decoder.cc
+++ b/media/filters/vpx_video_decoder.cc
@@ -265,13 +265,6 @@ bool VpxVideoDecoder::ConfigureDecoder(c
       return false;
     }
 
-    vpx_codec_err_t status =
-        vpx_codec_control(vpx_codec_.get(), VP9D_SET_LOOP_FILTER_OPT, 1);
-    if (status != VPX_CODEC_OK) {
-      DLOG(ERROR) << "Failed to enable VP9D_SET_LOOP_FILTER_OPT. "
-                  << vpx_codec_error(vpx_codec_.get());
-      return false;
-    }
   }
 
   if (config.alpha_mode() == VideoDecoderConfig::AlphaMode::kIsOpaque)
--- a/media/gpu/vaapi/vp9_encoder.cc
+++ b/media/gpu/vaapi/vp9_encoder.cc
@@ -12,7 +12,6 @@
 #include "media/gpu/macros.h"
 #include "media/gpu/vaapi/vp9_rate_control.h"
 #include "media/gpu/vaapi/vp9_temporal_layers.h"
-#include "third_party/libvpx/source/libvpx/vp9/ratectrl_rtc.h"
 
 namespace media {
 
@@ -114,47 +113,6 @@ VideoBitrateAllocation GetDefaultVideoBi
   return bitrate_allocation;
 }
 
-libvpx::VP9RateControlRtcConfig CreateRateControlConfig(
-    const gfx::Size encode_size,
-    const VP9Encoder::EncodeParams& encode_params,
-    const VideoBitrateAllocation& bitrate_allocation,
-    const size_t num_temporal_layers) {
-  libvpx::VP9RateControlRtcConfig rc_cfg{};
-  rc_cfg.width = encode_size.width();
-  rc_cfg.height = encode_size.height();
-  rc_cfg.max_quantizer =
-      QindexToQuantizer(encode_params.scaling_settings.max_qp);
-  rc_cfg.min_quantizer =
-      QindexToQuantizer(encode_params.scaling_settings.min_qp);
-  // libvpx::VP9RateControlRtcConfig is kbps.
-  rc_cfg.target_bandwidth = encode_params.bitrate_allocation.GetSumBps() / 1000;
-  // These default values come from
-  // //third_party/webrtc/modules/video_coding/codecs/vp9/vp9_impl.cc.
-  rc_cfg.buf_initial_sz = 500;
-  rc_cfg.buf_optimal_sz = 600;
-  rc_cfg.buf_sz = 1000;
-  rc_cfg.undershoot_pct = 50;
-  rc_cfg.overshoot_pct = 50;
-  rc_cfg.max_intra_bitrate_pct = MaxSizeOfKeyframeAsPercentage(
-      rc_cfg.buf_optimal_sz, encode_params.framerate);
-  rc_cfg.framerate = encode_params.framerate;
-
-  // Spatial layers variables.
-  rc_cfg.ss_number_layers = 1;
-  rc_cfg.scaling_factor_num[0] = 1;
-  rc_cfg.scaling_factor_den[0] = 1;
-  // Fill temporal layers variables.
-  rc_cfg.ts_number_layers = num_temporal_layers;
-  int bitrate_sum = 0;
-  for (size_t ti = 0; ti < num_temporal_layers; ti++) {
-    rc_cfg.max_quantizers[ti] = rc_cfg.max_quantizer;
-    rc_cfg.min_quantizers[ti] = rc_cfg.min_quantizer;
-    bitrate_sum += bitrate_allocation.GetBitrateBps(0, ti);
-    rc_cfg.layer_target_bitrate[ti] = bitrate_sum / 1000;
-    rc_cfg.ts_rate_decimator[ti] = 1u << (num_temporal_layers - ti - 1);
-  }
-  return rc_cfg;
-}
 }  // namespace
 
 VP9Encoder::EncodeParams::EncodeParams()
@@ -166,11 +124,6 @@ VP9Encoder::EncodeParams::EncodeParams()
       scaling_settings(kMinQP, kMaxQP),
       error_resilient_mode(false) {}
 
-void VP9Encoder::set_rate_ctrl_for_testing(
-    std::unique_ptr<VP9RateControl> rate_ctrl) {
-  rate_ctrl_ = std::move(rate_ctrl);
-}
-
 void VP9Encoder::Reset() {
   current_params_ = EncodeParams();
   reference_frames_.Clear();
@@ -226,22 +179,12 @@ bool VP9Encoder::Initialize(const VideoE
     }
     current_params_.scaling_settings.max_qp = kMaxQPForSoftwareRateCtrl;
 
-    // |rate_ctrl_| might be injected for tests.
-    if (!rate_ctrl_) {
-      rate_ctrl_ = VP9RateControl::Create(CreateRateControlConfig(
-          visible_size_, current_params_, initial_bitrate_allocation,
-          num_temporal_layers));
-    }
-    if (!rate_ctrl_)
-      return false;
   } else {
     if (config.HasTemporalLayer()) {
       DVLOGF(1) << "Temporal layer encoding works only when in "
                 << "kConstantQuantizationParameter";
       return false;
     }
-    DCHECK(!rate_ctrl_) << "|rate_ctrl_| should only be configured when in "
-                           "kConstantQuantizationParameter";
   }
 
   return UpdateRates(initial_bitrate_allocation,
@@ -309,14 +252,12 @@ BitstreamBufferMetadata VP9Encoder::GetM
 
 void VP9Encoder::BitrateControlUpdate(uint64_t encoded_chunk_size_bytes) {
   if (accelerator_->bitrate_control() !=
-          BitrateControl::kConstantQuantizationParameter ||
-      !rate_ctrl_) {
+          BitrateControl::kConstantQuantizationParameter) {
     DLOG(ERROR) << __func__ << "() is called when no bitrate controller exists";
     return;
   }
 
   DVLOGF(4) << "|encoded_chunk_size_bytes|=" << encoded_chunk_size_bytes;
-  rate_ctrl_->PostEncodeUpdate(encoded_chunk_size_bytes);
 }
 
 bool VP9Encoder::UpdateRates(const VideoBitrateAllocation& bitrate_allocation,
@@ -340,13 +281,8 @@ bool VP9Encoder::UpdateRates(const Video
       current_params_.bitrate_allocation.GetSumBps() *
       current_params_.cpb_window_size_ms / 1000;
 
-  if (!rate_ctrl_)
-    return true;
-
   const size_t num_temporal_layers =
       temporal_layers_ ? temporal_layers_->num_layers() : 1u;
-  rate_ctrl_->UpdateRateControl(CreateRateControlConfig(
-      visible_size_, current_params_, bitrate_allocation, num_temporal_layers));
   return true;
 }
 
@@ -397,21 +333,6 @@ void VP9Encoder::SetFrameHeader(
     }
   }
 
-  if (!rate_ctrl_)
-    return;
-
-  libvpx::VP9FrameParamsQpRTC frame_params{};
-  frame_params.frame_type =
-      keyframe ? FRAME_TYPE::KEY_FRAME : FRAME_TYPE::INTER_FRAME;
-  if (picture->metadata_for_encoding) {
-    frame_params.temporal_layer_id =
-        picture->metadata_for_encoding->temporal_idx;
-  }
-  rate_ctrl_->ComputeQP(frame_params);
-  picture->frame_hdr->quant_params.base_q_idx = rate_ctrl_->GetQP();
-  picture->frame_hdr->loop_filter.level = rate_ctrl_->GetLoopfilterLevel();
-  DVLOGF(4) << "qp=" << rate_ctrl_->GetQP()
-            << ", filter_level=" << rate_ctrl_->GetLoopfilterLevel();
 }
 
 void VP9Encoder::UpdateReferenceFrames(scoped_refptr<VP9Picture> picture) {
--- a/media/gpu/vaapi/vp9_encoder.h
+++ b/media/gpu/vaapi/vp9_encoder.h
@@ -20,7 +20,6 @@
 
 namespace media {
 class VP9TemporalLayers;
-class VP9RateControl;
 
 class VP9Encoder : public AcceleratedVideoEncoder {
  public:
@@ -101,8 +100,6 @@ class VP9Encoder : public AcceleratedVid
  private:
   friend class VP9EncoderTest;
 
-  void set_rate_ctrl_for_testing(std::unique_ptr<VP9RateControl> rate_ctrl);
-
   Vp9FrameHeader GetDefaultFrameHeader(const bool keyframe) const;
   void SetFrameHeader(bool keyframe,
                       VP9Picture* picture,
@@ -122,7 +119,6 @@ class VP9Encoder : public AcceleratedVid
   Vp9ReferenceFrameVector reference_frames_;
   std::unique_ptr<VP9TemporalLayers> temporal_layers_;
 
-  std::unique_ptr<VP9RateControl> rate_ctrl_;
   const std::unique_ptr<Accelerator> accelerator_;
 
   SEQUENCE_CHECKER(sequence_checker_);
--- a/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/vp9/libvpx_vp9_encoder.cc
@@ -743,8 +743,6 @@
     for (int si = 0; si < num_spatial_layers_; ++si) {
       svc_params_.speed_per_layer[si] =
           performance_flags_by_spatial_index_[si].base_layer_speed;
-      svc_params_.loopfilter_ctrl[si] =
-          performance_flags_by_spatial_index_[si].deblock_mode;
     }
   }
 
